\documentclass{article}
\usepackage{fullpage}
\usepackage{amsmath,amssymb}
\usepackage{natbib}
\usepackage{marginnote}
\usepackage{graphicx}
\usepackage{color}
\usepackage{hyperref}
\usepackage{latexml} % for \iflatexml

% \newcommand{\var}{\mathop{\mbox{Var}}}
% \newcommand{\Exp}{\mathop{\mbox{Exp}}}
\DeclareMathOperator{\var}{Var}
\DeclareMathOperator{\Exp}{Exp}
\renewcommand{\P}{\mathbb{P}}
\newcommand{\E}{\mathbb{E}}
\newcommand{\R}{\mathbb{R}}
\newcommand{\calE}{\mathcal{E}}
\newcommand{\calC}{\mathcal{C}}
\newcommand{\dconv}{\xrightarrow{d}}
\newcommand{\deq}{\stackrel{\scriptscriptstyle{d}}{=}}

\newcommand{\migrate}{\lambda_\text{mig}}
\newcommand{\mutrate}{\lambda_\text{mut}}

\newcommand{\gc}[1]{{\it\color{green}(#1)} }
\newcommand{\plr}[1]{{\it\color{blue}(#1)}}

\bibliographystyle{plain}

\begin{document}

\section{Introduction}

The covergent evolution of similar phenotypes in response to shared
selection pressures is a testiment to the power of selection to
repeatedly sculpt phenotypic variation. 
In some cases this convergence extends to the molecular level with
dispate taxa converging to the same phenotype through parallel genetic
changes in the same pathway, genes, 
or even by precisely the same genetic changes. Such precise
convergence points to the conservation of function of the genes
underlying adaptive traits, 
sometimes over deep time scales \citep{deephomologypapers}. Convergence on a molecular level also suggests that the path taken by adaptation can sometimes be relatively constained as few changes could have resulted in the adaptive phenotype, or that relatively few changes that can confer the advantagous phenotype of interest were sufficiently free of deleterious pleiotropic consequences to contribute to adaptation. 

%For highly polygenic traits 

Convergent adaptation can also occur within species, with individuals within a species 
adapted to the same environment in parallel through different large effect changes. 
There are a growing number of examples of this in a range of well studied organisms and phenotypes REFs.
All else being equal the evolution of convergent phenotypes is more
likely with a 
higher mutational input, i.e.\ when the mutational target and population sizes are larger \citep{}. 
Standing variation for a trait can also increase the probability of convergence, 
as multiple pre-existing alleles can spread from low frequency, 
if the newly beneficial alleles were previously not too deleterious \citep{Orr,Hermission}. 
The geographic distribution of populations can also shape the probability of parallel mutation within a species.
A widespread species is more likely to adapt by multiple mutations in parallel if dispersal 
is geographically limited, as subpopulations will adapt via new mutations rather than waiting for migration \citep{RalphCoop}. 

%When multiple rare changes sweep up from low frequency in parallel within a population this has been termed a soft sweep.\\

Intuitively, convergence is also more likely 
when geographically separated populations adapt to ecologically similar conditions. 
The probability that convergent adaptations arise independently
before adaptations spread between the populations by migration
will be enhanced if these adaptive alleles are maladapted in intervening environments,
since such adverse conditions can strongly spatially restrict the spread of locally adapted alleles \citep{slatkin}.
 
For example, a variety of plant species have repeatedly adapted to patches of soil with high concentrations of heavy metals
(e.g.\ serpentine outcrops and mine tailings) \citep{turnerarabidopsis,mimulus};
the alleles conferring heavy metal tolerance are thought to be disfavored off of these patches. 
Similarly, across the American Southwest, a variety of species of animals have developed locally adaptive cryptic coloration,
to particular substrates, e.g.\ dark rock outcrops or white sand dunes \citep{benson}.
One of the best-known examples is the rock pocket mouse (\textit{Chaetodipus intermedius}),
which is primarily dark on a number of black lava flows separated by much lighter soil \citep{dice}.
Strong selection appears to favour such crypsis \citep{owls},
so perhaps as a result of this strong selection against migrants, 
at least two distinct genetic changes are responsible from the dark pigmentation adaptation on different outcrops \citep{nachman}. 
Similar situations have been demonstrated in other mouse systems \citep{beachmice,peromyscus}.

%http://www.nature.com/hdy/journal/v94/n2/full/6800600a.html

%In \citep{RalphCoop} we formulated a simple model of parallel adaptation in a geographic setting
%where a continuous population was subjected to a selection pressure
%that was constant across the entire environment. 


%and that a single mutational change was sufficient to adapt populations to the novel environment. 
%We assumed that there was no standing variation for the adaptive allele, such that parallel mutation must be due to multiple mutations after the environmental shift. In that setting we found that there was a characteristic geographic scale, over which we would expect multiple instances of our adaptive allele to have arisen in parallel. This characteristic length could be expressed in terms of a simple compound parameter determined by our parameters of interest.
%Recall questions of origin of adaptations

%References on parallel adapation, standing variation, etc.:
 % coop,
 % pennings and hermisson.
%Note pennings and hermisson found standing deleterious variation was very important.  

%recall peromyscus examples with patchy environments

%Summarize previous paper

%Question assumptions of no standing variation and homogenous selection


%%%%%%%%%%%%%
\section{Methods}

%%%
\subsection{Introduction to a patchy model}
\label{ss:patchyspace}

Consider a population spread uniformly across a landscape. Within this environment there are patches of
habitat that individuals are initially poorly adapted to, surrounded by large areas that the population is well adapted to. 

Suppose it takes only a single mutational change to create an allele ($B$) that adapts an individual to the
poor habitat type. This mutational change occurs at a low rate of
$\mu$ per chromosome per generation. We assume that this initially rare new
allele has fitness $1+s_B$ relative to the unmutated type ($b$) in the ``poor'' habitat patches,
and assume that it has fitness $1-s_m$ when rare in the intervening areas, with
$s_m>0$. To simplify matters we assume that the disadvantage $s_m$ 
is sufficiently large that, on the relevant timescale,
the allele is very unlikely to fix locally in the regions where it is at a disadvantage.
(In particular, we require $s_m\neq 0$.)
\plr{NOTE: changed over to $s_m>0$.}

We are interested in the establishment of mutations in the ``poor'' patches by either
migration or mutation, and so we are only interested in whether the allele
can escape initial loss by drift when rare. Thus we do not have to
specify the the fitness of the homozygote, only that its
fitness is such the dynamics of the  allele when it is rare are
determined by the fitness of the heterozygote. 
More general dominance will only make small corrections,
with the exception of the recessive case, which we will omit.
Mostly, we follow the literature in treating the diploid model as essentially haploid.
% This allows us to treat a diploid model as being essentially haploid for our purposes. \gc{TRUE?}  \plr{would change e.g. freqs at migration-selection balance}

For the sake of simplicity we will assume a constant haploid population density $\rho$, 
across both types of habitat (we will return to discuss this equal
density point later). We further assume that the
variance in offspring number is $\xi^2$, 
and that the mean squared distance between parent and child is $\sigma^2$ (the dispersal variance).

\paragraph{Previous work} 
We will make use of a number previous results from the literature. 
\citet{slatkin1973geneflow}, working in one dimensional deterministic framework, 
showed that if the physical width of the patch is less than $2 \tan^{-1} (\sqrt{s_m/s_B})$, 
then there is not a stable equilibrium with both $b$ and $B$ present,
so that $B$ cannot establish within the patch due to migrational swamping \citep[see also][ for a review]{Lenormand}.
\citet{barton1987establishment}, again working in one dimension,
showed that this critical patch size also held in a stochastic framework \citep[see also the work of][]{Polk}. 
For patches above this critical size, \citet{barton1987establishment}
found that the probability of establishment of a new mutant that appears at distance $x$ from the patch
decays exponentially with distance from the patch, with the scale given by $\sigma/\sqrt{s_m}$.
% mutations appearing further than a few multiples of $\sigma/\sqrt{2s_m}$ away from such a patch 
% have vanishingly small probability of becoming established within the patch. 
\citet{barton1987establishment} also showed that mutations appearing within the patch have probability of establishment
less than the probability for a panimictic population 
\citep[which we denote $p_e$, and often approximate by $2 s_B / \xi^2$][]{haldane,fisher};
and that the probability approaches $p_e$ as the size of the patch increases.
This result holds quite generally for a geographically spread population that experiences a uniform selection
pressure \citep{Maruyama,cherry}. 
\plr{wait, what result? that fixation is hindered by geographic structure?}

%Maruyama: On the fixation probability of mutant genes in a subdivided population*
%should we cite Cherry http://www.ncbi.nlm.nih.gov/pmc/articles/PMC1462507/

%To make things tractable, we assume that the distances between the patches are large relative to migration distance $\sigma$,
%and that the disadvantage $s_m$ ensures that on the relevant timescale,
%the allele is very unlikely to fix locally in the regions where it is at a disadvantage.


%equal deme sizes?  general migration rates?


%%%%%
\subsection{Time to the establishment of a locally adaptive allele
  due to mutational influx}
\label{ss:patchymutation}

Consider first a single, isolated poor habitat patch of area $A$ in which no $B$ allele has yet become established. 
As we are interested in patches where local adaptation can occur,
we will assume that our patch is larger than the cutoff for local establishment, 
i.e.\ wider than $2 \tan^{-1} (\sqrt{s_m/s_B})$.

Let $p(x)$ be the probability that a new mutant $B$ allele arising at location $x$ 
relative to the patch fixes within the poor habitat patch.
The function $p(x)$ can be found in one dimension in terms of Jacobi elliptic functions,
XXX as shown in the Appendix, but the expressions are not particularly helpful.
The total successful mutational influx per generation is then given by the integral of $p(x)$ over the entire species range $\int \rho \mu p(x) dx$,
and depends in a complicated way on the relationship of patch width and selection coefficients,
but still scales linearly with the mutational influx density $\rho \mu$.
If the width of the patch is large relative to $\sigma/\sqrt{2s_m}$, 
then a reasonable approximation is to assume that $p(x) = p_e$ within the patch, and $p(x) = 0$ otherwise,
for a patch of area $A$ approximating $\int p(x) dx \approx 2 s_B A / \xi^2$. 
This holds for large patches, in which the number of mutations arising within the patch is larger 
than those arising sufficiently nearby;
we examine this more generally via simulation in Appendix XXX.
\plr{In Appendix: we simulated fixation prob in a patch with gradual transition.  We have convincing figures.}

The rate at which mutations arise and colonize a patch of area $A$ is therefore
\begin{align} \label{eqn:mutrate}
  \mutrate = \rho \mu \int p(x) dx  \approx 2 s_B \rho A \mu / \xi^2.
\end{align}
If this rate is low,  then the number of generations until a mutation arises that
will become locally established within the patch is exponentially distributed with rate $\mutrate$.  
Assuming that once a mutation becomes established it quickly reaches its equilibrium frequency across the patch, 
the expected time it takes for our patch to become colonized by the $B$ allele is approximately $1/\mutrate$.


%%%%
\subsection{Time to the establishment of a locally adaptive allele due to migrational influx}
\label{ss:patchymigration}
Now suppose that there are two patches, each of area $A$, of the poor
habitat type separated by distance $R$. 
We will consider the case where the $B$ allele has arisen and become established in the first patch, but has not yet appeared in the second,
for the purpose of comparison ignoring the possibility that a $B$ allele arises independently by mutation in the first patch.
As above, any copy of the new allele appears at location $x$ relative to the second, unadapted patch 
has probability approximately $p(x)$ of fixing locally in this patch,
whether the allele is a new mutation or is descended from the occupants of the first patch.

Due to migration--selection balance, 
there will be a small number of $B$ alleles descended from those of the first patch
present in the regions where they are disadvantageous;
the expected frequency at distance $r$ may again be expressed in terms of Jacobi elliptic functions \citep[Appendix XXX][]{barton},
although the equations are again not particularly helpful.
This expected frequency is the \emph{time-averaged} occupation frequency,
i.e.\ the average frequency of the allele at each location over a long time period,
below we study the fluctuations about this mean.
More helpful than the exact expressions,
we know that if $q(r)$ is the expected proportion present at migration--selection balance at distance $r$ from the first patch
in the absence of the second patch, 
then for large $r$ this proportion decays exponentially, 
\begin{align} \label{eqn:eqfreq}
  q(r) \approx C \exp( -\sqrt{2 s_m} r / \sigma),
\end{align}
where $C$ is a constant depending on the geographic shape of the populations and the selection coefficients. 
(Probably $C=1/2$ is close enough for applications; see Appendix XXX.)
\plr{same appendix with sims as above}
\gc{Should we also use this approx. to get the approx. rate of establishing mutations
arising outside of the patch, above. Feel like in the mutation rate
part we could pull $s_B$ out front of the rate as well.} 
\plr{huh?}

This allows us to get a good sense of the time scale of adaptation by migration between separated patches, as follows.
First imagine that the new allele is not advantageous in the second patch,
so that the expected number of alleles present in a small area $\Delta$ at distance $r$ from the already established patch
is $\Delta \rho q(r)$.
As these copies are disadvantageous, they form a subcritical branching process, which die out fairly soon.
Now add to this local selective advantage by stipulating that any copy of the allele in the new patch
has ``extra'' offspring that are marked in some way, on average $(1+s_B)-(1-s_m)=s_B+s_m$ additional offspring per generation.
If the probability of establishment \emph{in the new patch} is $p(x)$ as above,
then each of these new, marked copies appearing at location $x$ has probability $p(x)$ of establishing.
(This general relation is stated formally in Appendix XXX.)
Therefore, the total rate of establishment 
(i.e.\ the mean number of successfully establishing migrant lineages per generation)
is $\rho (s_B+s_m) \int q(|x|) p(x) dx$.
Using a similar approximation as in section \ref{ss:patchymutation},
we can approximate this, for a patch of area $A$ at distance $R$ from an already occupied patch, as
\begin{align}
  \migrate(R) = \rho A \frac{s_B}{\xi^2} (s_B+s_m) \exp\left( -\frac{ \sqrt{2 s_m} R}{\sigma} \right),
\end{align}
where we have set $C=1/2$ in equation \eqref{eqn:eqfreq}.
Using different methods, we arrive at the same functional form below in section \ref{ss:haplotype_length}.
(Note that if the size of the new patch is large relative to $\sqrt{s_m}/\sigma$,
then the integral of $q(x)$ over the patch will not be well approximated by $A \times q(R)$,
but the adjustment is easy to make.)

If this establishment rate is low, the time it takes
a mutation to arise that will become established 
will be approximately exponentially distributed with rate $\migrate$.
Assuming that the time it takes for a allele from the first patch to
become established in the second patch is short,
then  the expected waiting time until some migrant allele fixes in the
second patch will be $1/\migrate$.


%%%
\subsection{The probability of parallel adaptation between patches.} 
\label{ss:probparallel}

We now turn to the question of whether the separated populations adapt by parallel genetic changes 
or by the spread of a migrant allele between the patches. 
% These two scenerios are assumed to be distinguishable, 
% either because in the case of parallelism the mutations occur at different positions 
% or if the mutations occur at the same position they can be distinguished due to their haplotypic background. 
% Assuming that the geographically spread population initally contains no copies of the $B$ allele, 
% one or other of the patches of different habitat 
% will be the first to adapt via the establishment of a $B$ allele;
% so the question is then the relative timescale of new mutation and migration to the other patch.
As only a single mutation is required to adapt individuals to the
poor habitat patch, subsequent mutations that arise after an allele becomes established on the patch gain no selective benefit. 
Similarly, an allele introduced into a patch by migration will not be favored by selection to spread, 
if the patch has already been colonized. 
Therefore, mutations selectively exclude each other from patches, over short time scales, 
and will only slowly mix together over longer time scales by drift. 

We will assume that once a $B$ allele is introduced (by migration or mutation), 
it becomes established in the poor habitat patch rapidly. 
Under this approximation, and our selective exclusion assumption, 
after one of the patches becomes colonized by a single $B$ allele, 
the other patch will become colonized by migration or mutation, but not by both. 
As such, the question of whether the
population adapts to the second poor habitat patch in parallel, 
simply comes down to whether the a new mutation or a migrant allele is the first to become established in the second patch. 

Taking expressions \eqref{eqn:mutrate} and \eqref{eqn:migrate} above,
we see that if the patches each have sufficiently large area,
the rate of migration and mutation will be on the same order if 
$\mu \approx ((s_B+s_m)/2) \exp(-\sqrt{2 s_m} R / \sigma)$,
where $R$ is the distance between the patches.
(Note that in both cases, $A$ is the area of the not-yet-adapted patch.)

If we take $\mu = 10^{-8}$, and $s_B=s_m=s$, then this is satisfied if $R/\sigma \approx (18+\log(s))/\sqrt{s}$,
so if the selective pressure is fairly strong (say, $s=.05$),
then the distance between patches must be about 70 times the dispersal distance (i.e.\ $R/\sigma \approx 68$);
while if $s$ is much smaller, say $s = .001$, 
then the patches must be separated by many hundreds of times the dispersal distance (i.e.\ $R/\sigma \approx 368$).
The situation does not change much if $\mu$ is larger -- even if $\mu = 10^{-6}$, 
the required separation between patches is only reduced by $5/\sqrt{s_m}$.
If the separation between patches is much larger than this value -- certainly, by a factor of ten -- 
then patches are effectively isolated -- most adaptation will occur independent between patches.
On the other hand, if the separation is much smaller, then a single
mutation is likely to fix in both patches.

If we are willing to take these approximations at face value, 
and assuming both rates are small, 
the time till the first patch is colonized by the $B$ allele will be approximately exponentially distributed with rate $2 \mutrate$.
Following this, the time till the second patch is subsequently colonized 
(via either migration or new mutation) 
will be exponentially distributed with rate $\mutrate+\migrate$.
Both these times scale linearly with $\xi^2/(\rho A s_{B})$, so that 
the overall time to adaptation is robustly predicted to increase with offspring variance ($\xi^2$)
and decrease with patch population size and benefical seleciton coefficient.
Furthermore, the probability that the second adaptation is a new mutation,
i.e.\ the probability of parallel adapatation, is
\begin{equation}
  \frac{\mutrate}{\mutrate+\migrate} = \frac{2\mu/(s_B+s_m)}{2\mu/(s_B+s_m) + \exp\left(- \sqrt{2 s_m} R / \sigma \right) },
\end{equation}
so that the probability of parallel mutation should increase
approximately logistically with the distance between the patches at rate $\sqrt{s_m} /\sigma$. 



%%%%
\subsection{Multiple patches}


We can extend this reasoning to multiple patches of unadapted habitat 
in which the $B$ allele can facilitate local adaptation. 
Under our assumptions of mutational exclusion and rapid establishment, 
each patch will adapt through establishment of a single copy of the $B$ allele, 
either by migration or mutation.
Since we assume that the times between successive establishments is the result of many nearly-independent attempts
with small probability of success,
the process of occupation is well approximated by a discrete-state Markov chain,
whose state is the labeling of patches according to the colonizing allele
(or none, for those that haven't yet adapted).

If not yet adapted,
a patch of area $A$ will acquire the adaptation by new mutation at rate $2 s_B \mu \rho A/\xi^2$.
Without loss of generality, assume that patches $1, \ldots, k$ are already adapted,
and these patches are at distances $R_1, \ldots, R_k$ away from this unadapted patch, 
then the total rate of adaptation through migration is
\begin{equation}
  \frac{ s_B (s_B+s_m) \rho A}{\xi^2} \sum_{i=1}^{k} \exp\left(- \sqrt{2 s_m} R_i/\sigma\right),
\end{equation}
and the probability that the adaptation is from patch $i$ is $\exp\left(- \sqrt{2 s_m} R_i/\sigma\right)/\sum_{j=1}^{k} \exp\left(- \sqrt{2 s_m} R_j/\sigma\right)$.
These specify the transition rates of the process.

Since the compound parameter $s_B \rho / \xi^2$ is common to both rates,
it functions only as a time scaling of the process, 
and therefore has no effect on the final configuration of the pattern of which patches have adapted
in parallel versus have shared genetic basis to their adaptive response to the poor environment.

Also note that we can rescale time by the typical patch size, and introduce a parameter, say $a_k = A_k/\bar A$,
making the properties (other that the time scaling) of the discrete model independent of the \emph{numerical sizes} of the demes themselves.
This is complementary to the results of \cite{softsweepsII}, who showed that multiple mutations are likely to arise \emph{within} a panmictic deme
if the population-scaled mutation rate $2 N \mu$ is greater than $1$.

\plr{conclusion?}

%%%%%%%%%
\subsection{Length of the hitchhiking haplotype}
\label{ss:haplotype_length}

If a patch adapts through new mutation, the genomic region surrounding the selected site will hitchhike \citep{hitchhiking} along with it,
so that at least initially, all adapted individuals share a fairly long stretch of haplotype.
This association gets slowly whittled down by recombination during interbreeding at the edge of the patch;
but there will always be longer LD nearby to the selected site.
\plr{Could work out on what time scale it gets whittled down and the shared length at stationarity.}

On the other hand, if a an already adapted patch colonizes another through migration,
then the newly colonized patch will inherit a long piece of haplotype around the selected site from the originating patch.
The length of this haplotype is inversely proportional to the number of generations the allele spends ``in transit'' --
this is because, following back the successful migrant lineage, 
each reproductive event in locations where the allele is at low frequency
whittles down the hitchhiking haplotype.
(In this case, the haplotype is literally hitchhiking across geography!)

Here is a simplified picture of the process of adaptation by migration.
There is a constant outflux of migrants from the patch where the allele is already established; 
any descendants of these carrying the allele are locally maladapted, so are mostly doomed to extinction.
(We do not require that migration be blind to the environment --
although adaptive migration would reduce the outflux of migrants --
but do assume that the focal allele does not affect migration once outside the patch.)
\plr{or maybe we should discuss this later?}
However, eventually the descendants of one lucky migrant will reach the new patch and colonize it.
(If the new patch is at some distance, it will almost certainly \plr{a.c.} be colonized first by only one family.)
We will estimate the total outflux of migrants, and assume that their fates are independent of each other,
so that we need only calculate the probability that offspring carrying the focal allele of a given migrant colonize the new patch,
and the distribution of the transit time, given that this occurs.
In doing so, we only follow the offspring of the migrant that carry the focal allele,
and approximate these as a branching process, 
i.e.\ assume that the number of offspring each produces is independent of each other.

Since we are concerned with the rare migrant families that transit quickly between patches, 
we should remove any backmigrants into the already-established patch;
we do this by assuming rather that the allele is deleterious \emph{everywhere} --
backmigrants will contribute little, since they are going the wrong direction,
and we deal with arrival at the new patch later.
Spatial motion is now independent of reproduction,
so the chance that offspring of the migrant first arrive in the new patch after $\tau$ generations 
is the probability of survival of the family until time $\tau$
multiplied by the probability that some of the family are in the new patch, given survival.
The first probability comes from well-known properties of branching processes,
and the second from Brownian motion.
Then, the chance that the migrant family establishes in the new patch 
depends on the size of the family that arrives in the patch,
and the probability that each establishes.

If each individual has on average $(1-s_m)$ offspring, then after $t$ generations,
the family has on average $(1-s_m)^t$ members.
Let $p_e(t)$ be the chance of extinction by time $t$, and let $K_t$ be the (random) family size conditioned on nonextinction,
so that $(1-s_m)^t = (1-p_e(t))\E[K_t]$.
It turns out that by assuming the fate of each offspring is independent (i.e.\ that this forms a branching process),
and that the distribution of offspring numbers is not too heavy-tailed \citep[see][for details]{jaegers1976branching},
that $K_t$ has a limiting distribution: $K_t \dconv K$ 
and that $(1-s_m)^{t}/(1-p_e(t)) \to \E[K]$ as $t \to \infty$.
In other words, the chance of survival to time $t$ is a constant multiplied by $(1-s_m)^t$,
and conditional on survival, the family size is fixed.

We can understand this quite concretely, by a method described rigorously by \citet{geiger1999elementary}.
When we condition on survival, we condition on existence of at least one lineage from time $0$ to time $t$.
Once given this lineage, nothing distinguishes any of the other reproduction events --
then each individual in this lineage must give birth to at least one offspring,
and all other reproduction events are unconditioned.
In other words, the genealogy of a family that survives until $t$
can be decomposed into a ``trunk'' that lasts until $t$
anmd independent ``branches'' growing from each intermediate time $0\le s \le t$ 
that may or may not survive until $t$.
This process is depicted in Figure~XA.
Concretely, if we write $Z_t$ for the number of offspring in the branching process alive at time $t$
(so $\P\{Z_t = 0\} = p_e(t)$),
then
\begin{align}
  Z_t \; \vert \; Z_t>0 \deq \sum_{s=0}^t \sum_{k=1}^{W_s-1} Z^{(s,k)}_{t-s},
\end{align}
where each $Z^{(s,k)}$ is an independent copy of $Z$,
and each $W_s$ is an independent copy of the offspring distribution, conditioned on $W_s \ge 1$.

\begin{figure}[ht!!]
  \begin{center}
  \input{branching-concept.pdf_tex}
  \end{center}
\caption{Depiction of the decomposition described in the text.
In red is the central lineage (whose spatial motion we follow);
in black are the surviving branches off of this lineage, 
and in grey are offspring of the central lineage that did not survive until $t$.
\plr{Label $W_s$ in this figure?}
\plr{Add as part B a simulation figure.}
\label{fig:branching_decomp}
}
\end{figure}

Now we can add in the spatial motion, by following only the motion of the ``trunk''
(i.e.\ the red line in figure \ref{fig:branching_decomp}).
We will first work in one dimension.
If the variance of the parent-offspring distance is $\sigma^2$ (and the distribution is not too heavy-tailed),
then the motion of this is approximately Brownian, with variance $\sigma^2$ per generation.
Suppose that the new patch is at distance $x$.
The probability density for the first time such a Brownian motion $B_s$ hits $x$ is
\citep[XXX]{feller}
% note $\sigma B_t \deq B_{\sigma^2 t}$ so $\P\{ \sup_{s\le t} \sigma B_s \le x \} = \P\{ \sup_{s \le \sigma^2 t} B_s \le x \}
\begin{align} \label{eqn:hitting_dist}
  \P\{ B_t=x \;\mbox{and} \; B_u<x \;\mbox{for all}\; u<t\} =  \frac{x}{\sigma^3 t^{3/2}\sqrt{2\pi}} \exp\left(-\frac{x^2}{2t\sigma^2}\right) .
\end{align}
The chance that the descendants of a given migrant arrives at the new patch at time $t$
is now just equation~\eqref{eqn:hitting_dist} multiplied by the chance that 
the migrant family is still alive at time $t$.
For convenience, we define $\alpha = - \log(1-s_m) \approx s_m$,
so that $1-p_e(t) \simeq (1/\mu_K) \exp(-\alpha t)$,
and define $\mu_K$ to be the limiting mean family size conditioned on survival.
By integrating over $t$, we get the chance that descendants of the migrant ever arrive at the new patch is
\begin{align} 
  & \int_0^\infty \frac{(1-s_m)^t}{\mu_K} \frac{x}{\sigma^3 t^{3/2}\sqrt{2\pi}} \exp\left(-\frac{x^2}{2t\sigma^2}\right)  dt \\
  & \qquad = \frac{1}{\mu_K} \exp\left( - \frac{ x \sqrt{2 \alpha}}{\sigma} \right) .\label{eqn:estab_prob}
\end{align}
(Note that this is, up to the constant $\mu_K$, the chance that a Brownian motion killed at rate $\alpha$
ever hits $x$ \citep{feller}.
This approximation holds if $x$ is large enough that the approximation $1-p_e(t) \approx (1-s_m)^t/\mu_K$ holds.)

We now have the tools to compute the two quantities of interest:
the probability that the family of a given migrant will eventually establish in the new patch,
and the transit time conditioned on establishment.

The probability of establishment of a single migrant in a patch distance $x$ away
is equal to the probability that the migrant's family makes it to the patch at all,
multiplied but the chance that it establishes locally, given that it makes it there.
We have computed the first term in equation~\eqref{eqn:estab_prob},
and we make the small assumption that the second term is just the probability of local fixation
of a beneficial allele present in $K$ copies,
i.e.~$\E[1-(1-p_f(s_b))^K] \approx 2 s_b \mu_K/\xi^2$,
where $p_f(s_b) \approx 2s_b/\xi^2$ is the probability of fixation of a single mutant with additive benefit $s_b$.
Therefore, the probability of establishment is
\begin{align} \label{eqn:bp_prob_estab}
  F(x) &\approx \frac{2s_b}{\xi^2} \exp\left( - \frac{\sqrt{2 \alpha} x}{\sigma} \right) ,
\end{align}
which verifies the less intuitive calculation of $\migrate$ above.

Now, write $\tau$ for the first time the trunk lineage hits $x$, i.e.\ reaches the new patch,
with the convention that $\tau=\infty$ if it never does.
By the above arguments, and equation~\eqref{eqn:estab_prob}, we have that the conditional probability density is
\begin{align}
  \P\{\tau\in dt | \tau<\infty \} &= \frac{\P\{\tau \in dt\}}{\P\{\tau<\infty\}} \\
  &= \frac{x}{\sigma^3 t^{3/2}\sqrt{2\pi}} \exp\left(-\frac{x^2}{2t\sigma^2} -\alpha t + \sqrt{\alpha} \frac{x}{\sigma} \right) .
\end{align}
Although $\tau$ is interesting, we are more interested in the length of the hitchhiking segment,
since that is where our signal would come from.
Since each generation spent in transit provides an opportunity for recombination,
if recombination is Poisson, the length of the hitchhiking segment (in Morgans) on each side of the site will be exponentially distributed
with mean $\tau$, so that if $L$ is the length on, say, the right, then
\begin{align} \label{eqn:haplen_cdf}
\P\{L>\ell\} &= \E[e^{-\ell \tau}|\tau<\infty] \\
  &= \exp\left\{{-\frac{x}{\sigma}\left(\sqrt{2(\ell+\alpha)} - \sqrt{2\alpha}\right)}\right\} .
\end{align}
This is also the Laplace transform of $\tau$;
we can differentiate it to find that $\E[\tau] = x/(\sigma\sqrt{2\alpha})$
and $\var[\tau] = x/( 2^{3/2}\alpha^{3/2}\sigma)$.
Integrating~\eqref{eqn:haplen_cdf} gives that 
$\E[L] = \sigma \sqrt{2\alpha}/x + \sigma^2/x^2$ and $\var[L] = 2\alpha\sigma^2/x^2 + 4 \sigma^3 \sqrt{2\alpha}/x^3 + 5 \sigma^4 / x^4$;
an easier way to see this is to note that if $Y$ is an exponential random variable with rate $x\sqrt{2}/\sigma$,
then $L$ has the same distribution as $(Y + \sqrt{\alpha})^2 - \alpha$.

%%%
\subsubsection{Limiting distributions}

We are already assuming that $x$ is large relative to the migration-selection cline width $\sigma/\sqrt{s_m}$;
by taking the large-$x$ limit above we can obtain more intuitive descriptions of the process
via a central limit for $\tau$.
Define $\eta_x$ to be the deviation of $\tau$ from its mean, normalized by its standard deviation,
so that $\tau = x/\sigma(2\alpha)^{-1/2} + \eta_x \sqrt{x/\sigma} (2\alpha)^{-3/4}$.
Plugging this into equation~\eqref{eqn:haplen_cdf},
after some algebra (and using that $\sqrt{x+\epsilon} = \sqrt{x} + \frac{\epsilon}{2\sqrt{x}} - \frac{\epsilon^2}{2\sqrt{x}^3} + O(\epsilon^3)$) we obtain that
\begin{align}
  \E[e^{-\ell \eta}] &= \exp\left\{ \frac{\ell^2}{2} + O\left(\frac{1}{x}\right) \right\}.
\end{align}
By the properties of Laplace transforms \citep{durrett},
this implies that $\eta_x$ converges in distribution to a standard Gaussian random variable as $x \to \infty$.
In other words, we now know that the distribution of $\tau$ about its mean is approximately Gaussian.
Since the standard deviation of $\tau$ is of lower order than its mean,
a similar calculation shows that for large $x$, the haplotype halflength $L$ is approximately exponentially distributed, 
with mean $\sqrt{2\alpha} \sigma/x$.

% \begin{align}
%   \E[e^{-\ell \eta}] &= \E\left[ \exp\left\{-\ell\left(\tau - \frac{x}{\sigma\sqrt{2\alpha}}\right)\frac{(2\alpha)^{3/4}\sqrt{\sigma}}{\sqrt{x}} \right\} \right] \\
%   &= \E\left[ \exp\left\{ - \left(\frac{\ell (2\alpha)^{3/4} \sqrt{\sigma}}{\sqrt{x}}\right) \tau \right\} \right] \exp\left( (2\alpha)^{1/4} \sqrt{\frac{x}{\sigma}} \right) \\
% \end{align}


%%%%%%%%%%
\section{Applications} 
Hoekstra, Nachman, and colleagues have made an extensive study of coat color in the rock pocket mouse, \emph{Chaetodipus intermedius}, building on classic work by XXX. These mice live throughout the American SouthWest and into Mexico, and mostly live on the light colored rock outcrops, and have light pigmentation \citep[presumably an adaptation to visual-based predators][]{}. However, in some regions these mice live on dark colored substrates, including old lava flows, and have adapted to these patches in part through changes to a dark pigmentation. On one of these flows (Pinacate) an allele at the Mc1r locus is responsible for much of the change to a dark pelage \citep{Nachman:03}. This dark allele differs from the light allele by four amino acid changes - and has a dominant or partially dominant effect depending on the measure of coat color assessed. These lava flows are separated from each other by a range of distances, with some being separated from any other flow by hundreds of kilometers of light colored substrate. The Pinacate allele is not present in a number of other dark colored rock pocket mouse populations suggesting these populations have adapted in parallel. However, there is some evidence \citep{Hoekstra:05} that, elsewhere in the range, some nearby populations may share a dark phenotype whose genetic basis has been spread by migration. 

As can be seen from above a key parameter is the characteristic lengths $\sigma/\sqrt{s_m}$. \citep{Hoekstra:04} studied the frequency of the dark MC1R allele, and coat color phenotypes, at sites on the Pinacate lava flow and at two nearby light substrate sites.
 On the lava flow the dark MC1R allele is at $86\%$ frequency. At the Tule light substrate site, which is $\sim 12km$ from the flow, the dark MC1R allele was at $\sim 3\%$ frequency. 
At a greater distance from the lava ($XXX$km) the light colored substrate the dark MC1R was absent from a sample at the Christmas pass. 
While at the other light substrate site (O'Neill), which is $10$km from the flow on opposite side from Tule, the dark MC1R allele has a frequency of $\sim 34\%$. On the basis of these numbers we can gain a sense of a plausible range of characteristic lengths.
Assuming the allele is at $50\%$ frequency at the edge of the lava flow, 
we can fit the frequency found at each of the light substrate sites that are 
polymorphic for the dark allele under a simple allele frequency cline model \citep{XXX}.
Doing this for the Tule we obtain $\sigma/\sqrt{s_m} \approx 4$km and for the O'Neill site
$\sigma/\sqrt{s_m} \approx 30$km. \\

The mutational target size $\mu$ for the trait is unclear. 
While the Pinacate dark haplotype differs from the light haplotype at three amino-acid residues,
it is likely that not all of these changes are needed for a population to begin to  
adapt. Also there a number of genes, beyond MC1R, at which adaptative changes affecting 
pigmentation have been identified in closely related species and more
broadly across vertibrates \citep{}.To span a range of plausible
values, we use a low mutation rate 
We use a low mutation rate $\mu= 10^{-8}$, where the adaptive
mtation has to occur at a single base pair.
An intermediate mutation rate $\mu= 10^{-5}$, equivalent to
the mutation being able to arise at any of a thousand bases pairs,
e.g. any of the coding bases of a typical length gene. And a high
mutation rate $\mu = 10^{3}$, corresponding to the adaptive mutation
being able to arise in any of $100$ genes, e.g. anywhere in a large pathway. 


\section{Discussion} 

\plr{Note that optimal $s_m$ for migration makes migration harder to spot.}

if populations and mutation rates are large enough, parallel adaptation is likely, aided by geography. 
recall others' work. 

patchy selection helps this happen; 
effective migration rate looks like what; 
strength of (positive?) selection suprisingly does not affect results (?) 

standing variation will be important under these circumstances; 
migration rate (and so geographic structure) will not affect numbers of types if this is true; 
however migration always affcts spatial patterns. 

discussion of applications. 

discuss mixing of types; 
Graph: from simulation showing local proportions in a single patch showing initial fixation of a single type and later mixing converging toward more than one type.  (and eventual loss of one?)

\bibliography{standing_patches_refs}

\appendix

\section{Integrals appearing in the text}
    \label{apx:integrals}

In the text at several points appear integrals of the form
\begin{align}
  \int_0^\infty t^c \exp \left( - \alpha t^d - \beta t^{d+1} \right) dt 
\end{align}
where $c$ is a positive integer, $d$ is the dimension, and $\alpha$ and $\beta$ are positive real numbers.
This could be evaluated through standard numerical methods; below we describe a power series expansion.
Changing variables to $u = \beta t^{d+1}$, this becomes
\begin{align}
    \left( (d+1)^{-1} \beta^{ (1-c)/(d+1) } \right) \int_0^\infty u^{(c+1-d)/(d+1)} \exp\left( - \alpha \beta^{-d/(d+1)} u^{d/(d+1)} - u \right) du ,
\end{align}
so it suffices to evaluate the function
\begin{equation}
    G(a,b,x) := \int_0^\infty  t^a \exp\left( -x t^b - t \right) dt ,
\end{equation}
in the case that $a=(c-d+1)/(d+1)$, $b=d/(d+1)$, and $x$ is a function of the demographic paramters.
Since $a$ and $b$ only depend on the dimension and the quantity being computed,
we are interested in $G$ as a function of $x$.
At least in the case $b=d/(d+1)$ it is possible to express $G$ as a finite sum of gamma functions,
but we proceed with a simpler method.
Note that $\partial_x G(a,b,x) = -G(a+b,b,x)$,
and that at $x=0$, the function $G$ is the gamma function $G(a,b,0) = \Gamma(a+1)$.
Therefore, a Taylor series for $G$ would be
\[
    G(a,b,x) = \sum_{n \ge 0} \frac{(-x)^n}{n!} \Gamma(a+nb+1) .
\]
It is easy to check using Stirling's formula that $\limsup_{n \to \infty} ( x^n \Gamma(a+nb+1)/n! )^{1/n} = 0$
if $b<1$, so the sum converges.


%%%%%%%%%%
\section{The probability of establishment}
\label{ss:prob_estab}


In the main text we have made use of two decompositons of the spatial branching process
to approximate in two different ways the rate of establishment of migrant alleles.
Here we make the relationship more formal.
The decompositions use a similar idea to that found in \citep[section D.12]{athreyaney},
where a supercritical branching process is decomposed into individuals whose families eventually die out
and those having infinite lines of descent.
We also note that the approximations fall generally in the realm covered by \citet{aldouspoissonclumping}.

Consider a discrete-time branching process $Z_t$ with spatial motion and offspring distribution depending on spatial location.
Suppose that the offspring distribution for an individual at location $x$ is distributed as $X_x$.
We can record the state of the process simply as the list of locations of individuals:
$Z_t = (x_1, \ldots, x_{N(t)})$ means that there are $N(t)$ individuals alive at $t$, at locations $x_1, \ldots, x_{N(t)}$.
We can construct the process iteratively: given $Z_t$,
we sample first the number of offspring of each, and then the locations the offspring migrate to.
Concretely, given $Z_{t} = (x_1, \ldots, x_{N(t)})$,
first let $X_1(t),\ldots,X_{N(t)(t)}$ be independent random draws from the offspring distribution,
with $X_k(t)$ having distribution $X_{x_k}$.
The number of new offspring is $N(t+1) = \sum_{k=1}^{N(t)} X_k(t)$,
and we can sample their locations independently given the location of their parent to obtain $Z_{t+1}$.
Note that the genealogy is implicitly recorded:
the parent of the $k^\mathrm{th}$ individual at time $t$ is
the $a(k,t) = \min\{ \ell : \sum_{j=1}^\ell X_j(t-1) \ge k \}$-th individual at time $t-1$.
It will be useful to extend this to more distant ancestors:
let $a_1(k,t) = a(k,t)$ and recursively define $a_m(k,t) = a(a_{m-1}(k,t),t-m+1)$, as long as $m\le t$.

We want to find (approximations to) $p(x)$, the probability that a branching process beginning with a single individual at $x$ does not die out. 
(In other words, $p(x) = \P\{ N(t) = 0 \; \mbox{for some}\; t>0 \; \vert \; Z_0 = (x) \}$.)
We assume that the mean number of offspring $\E[X_x]$ is only greater than one for $x \in A$ (``the patch''),
and so the process can only establish if some offspring eventually get to the patch (with probability one).

The second decomposition we use is simpler, only conditioning on the number of individuals who ever reach the patch.  
Let $\calE = \{(t_i,k_i)\}$ be the times and indices of individuals who are in the patch but none of whose ancestors have been --
i.e.\ $(t,k) \in \calE$ iff $x_k(t) \in A$ and $a_m(k,t) \notin A$ for all $1\le m \le t$.
If we remove these individuals (and their offspring), the resulting branching process has mean number of offspring less than or equal to 1 everywhere,
so there are a finite number of such individuals with probability one (the process dies out).
Therefore, the probability that the process dies out is then the probability that the families descending from each individual in $\calE$ dies out.
By independence of different offspring, this is
\begin{align}
  1-p(x) = \E\left[ \prod_{(t,k) \in \calE} (1-p(x_k(t))) \; \vert \; Z_0 = (x) \right] .
\end{align}
Above, we make the approximation that $p(x_k(t)) \approx p_e$,
the probability of establishment ignoring the possibility of migrating outside the patch.
In this case,
\begin{align}
  p(x) \approx \E\left[ p_e^{|\calE|} \; \vert \; Z_0 = (x) \right] .
\end{align}
Here $|\calE|$ is the ``size of the family that reaches the patch'';
in section \ref{ss:haplotype_length} we estimated the probability that any of the family reaches the patch
and argued that the distribution of $|\calE|$ is given by $K$,
the size of the subcritical branching process conditioned on still being alive.


The first approximation uses a similar trick, but rather than treating all individuals migrating into the patch
as an opportunity for establishment,
we couple the branching process to a strictly subcritical branching process
that produces extra offspring, each of which provides an opportunity for establishment.
To this end, suppose that the offspring distribution off the patch is given by $X'$ (so $X_x \deq X'$ for $x \notin A$),
and the offspring distribution within the patch stochastically dominates $X$,
i.e.\ $\P\{X_x\ge n\} \ge \P\{X'\ge n\}$ for all $x\in A$ and $n \ge 0$.
Stochastic domination implies that we can \emph{couple} any instance of $X_x$ with one of $X'$ and a remainder --
i.e.\ jointly construct an $X'$ and a $Y_x$ such that $X_x = X' + Y_x$,
and that $Y_x$ also takes values in the nonnegative integers.
We can therefore modify the construction of the branching process above,
so that $X_k(t) = X_k'(t) + Y_k(t)$, where $X_k'(t) \deq X$ and $Y_k(t) \deq Y_{x_k(t)}$.
We can define a \emph{core} branching process $Z'$ using the $X'$, which will be finite (since $\E[X']<1$).
Formally, say that the first $X_k'(t)$ offspring of the $(t,k)^\mathrm{th}$ individual are ``core'' offspring,
denoted by $\calC_t = \{ (t,k) \; : \; 0 \le k - \sum_{j<\ell} X_j(t) \le X_\ell'(t) \; \mbox{for some}\; 1 \le \ell \le N(t) \}$;
and define $Z'$ as the core individuals whose ancestors are all ``core'' as well,
i.e.\ $Z'_t = \{ x_k \in Z_t \; : \;  a_m(k,t) \in \calC_{t-m} \; \mbox{for all}\; 0 \le m \le t\}$.
The boundary analogous to above in the first computation is then the non-core offspring of core individuals;
denote the times and indices of these by $\partial \calC$,
so that $(t,k) \in \partial \calC$ iff $(t,k) \notin \calC_t$ and $a(k,t) \in \calC_{t-1}$.
We can now use this as above to compute the probability of establishment beginning from a single individual:
\begin{align}
  1-p(x) = \E\left[ \prod_{(t,k) \in \partial\calC} (1-p(x_k(t))) \; \vert \; Z_0 = (x) \right] ,
\end{align}
but this is less helpful, as the distribution of $\calC$ is harder to get at.

However, recall the actual problem at hand.
Roughly speaking, we approximate the process of colonization of the new patch by migrants from an established patch
by imagining that the established patch sends out migrants, and that each of these acts as an independent branching process,
at least in the region of the new patch.
We can suppose that migrant families appear at some rate, say, midway between the patches, and far enough from the original one that we can ignore it,
and leave the precise distribution of their appearance unspecified
We can decompose each of these branching processes as above into ``core'' and ``extra'',
so that the point in time of ``establishment'' is well-defined:
the time $\tau$ of first appearance of an ``extra'' offspring with infinite line of descent.
Given that establishment has not yet occurred by $t$, 
and the combined state of the core process $Z'_t$,
then the mean number of establishment events occurring at time $t$ is $\sum_{x \in Z'_t} p(x) \E[Y_x]$.

If establishment events were independent and sufficiently rare,
the influx of branching processes begain long enough ago that $Z'$ was at stationarity
with $z_A := \E[ \# \{ (x,t) \in Z'_t : x \in A \} ]$,
and we again approximate $p(x) \approx p_e$,
then we would have that 
\begin{align}
  \P\{\tau>t\} \approx \exp\left\{ - t z_A \E[Y_x] p_e \right\} ;
\end{align}
in our case, $\E[Y_x] \approx (s_B+s_m)$ and $p_e \approx 2 s_B/\xi^2$,
so $\tau$ is approximately exponentially distributed with rate $ 2 z_A (s_B+s_m) s_B/\xi^2 $.
We can get $z_A$, the occupation density without the new patch, 
directly by integrating the expression~\eqref{eqn:eqfreq} over the patch.

In obtaining an exponential distribution, we have assumed effectively that establishment events
are independent of each other.
This will not be strictly true, since an establishment implies that a family of migrants are present in the patch,
and hence others may produce offspring with infinite lines of descent at a similar time.
The mean total family size of an allele with fitness $1-s_m$ (i.e.\ a subcritical branching process) is $\sum_{t \ge 0} (1-s_m)^t = 1/s_m$;
so the mean number of ``extra'' offspring is $1+s_B/s_m$.
Establishment events are correlated to the extent that single such families give rise to multiple establishment events,



% We will record the state of the branching process as a finite point measure,
% so that if there are $n$ individuals alive at $t$, at locations $x_1, \ldots, x_n$,
% then $Z_t = \sum_{k=1}^n \delta_{x_k}$, and we write $Z_t(A) = \sum_{k=1}^n \bone_{x_k \in A}$ for the number of individuals in a set $A$.


%%%%%%%%%%%
\section{Patchy selection} 
\label{ss:discretedemes}

We have been working in a continuous model of geographic space; 
in this section we move to a discrete model of isolated populations exchanging rare migrants,
but in section \ref{ss:patchyspace} discuss how a model with {\em patchy} selection can be approximated by such a discrete model.

Suppose now that we have a set of $L$ discrete populations of individuals that may reproduce or migrate,
and that migration and mutation are rare, so that the waiting times until either event is approximately exponential.
Suppose that for each pair of populations, labeled $i$ and $j$, the mean number of migrants that travel from $i$ to $j$ per generation
is $M_{ij}$ and that the mean number of new mutations appearing in population $i$ per generation is $\mu_i$.

{\tt Include cartoon here.}

If we suppose that fixation occurs on a faster time scale than mutation or migration,
so that it is very unlikely that two different migrants or mutants begin to fix locally in the same population,
then we can treat the process of extinction or local fixation as instantaneous 
(a limit previous studied by \cite{Slatkin:81}).
In this limit we may also ignore swamping effects of inmigration of nonmutant alleles.

In this case, we have the following picture.
The population at first starts out with no selected alleles. 
At some later time point suppose that there are $T$ different alleles present in the population, 
that the $k^\mathrm{th}$ allele is occupying demes $\{i^k_1, \ldots, i^k_{n_k}\}$, for $1\le k \le T$.
Let $I$ denote the set of all adapted demes, 
and let $J = \{j_1, \ldots, j_\ell\}$ denote the demes that have not yet adapted, with $\ell = L - \sum_k n_k$.
Then we are assuming that two types of event can happen which could change the state:
either at rate $\mu_j p_f$, an unadapted deme $j$ produces a new mutant, which fixes in this deme;
or at rate $M_{ij} p_f$, an adapted deme $i$ sends a migrant to unadapted deme $j$, which fixes in that deme.
Therefore, under these assumptions, the probability of fixation $p_f$ (and hence the selection coefficient)
only enter as a time scaling: for instance,
if we define $R = \sum_{j \in J} \left( \mu_j + \sum_{i \in I} M_{ij} \right)$,
then the total rate of such events is $p_f R$.
The probability that the next event results in, say, adaptation of deme $j$
is 
\[
 \left( \mu_j + \sum_{i \in I} M_{ij} \right) / R,
\]
and the probability that deme $j$ adapts through migration from deme $i$, given that deme $j$ is the next to adapt, is
$M_ij / R$,
while the probability that it adapts through a new mutation is $\mu_j / R$.

The main point here is that the final pattern,
and hence the likely extent of parallel adaptation,
is independent of the strength of selection.
This is in contrast to the continuous case, however, this observation is dependent on the assumption of weak migration
and fails to hold if different migrant lineages interact.
This can be viewed as a continuous-time Markov process XXX.
Explicit solutions are not available in the general case,
but if we specialize to a completely symmetric model (i.e.\ the ``island'' model),
then the mathematical picture is a pretty one.

\subsection{Symmetric patches}

Indeed, suppose that migration and mutation are symmetric: $M_{ij}=m$ and $\mu_j = \mu$ for all $j$ and $i\neq j$.
Suppose at some time the $k^\mathrm{th}$ allele is occupying $n_k$ demes, 
with $n = \sum_k n_k < L$.
Then at rate $(L-n) N \mu p_f + n (L-n) N m p_f$,
either a migration or mutation event happens.
With probability $\mu/(\mu + n m)$, it is a mutation,
and a new deme is randomly chosen and assigned a new allele.
With probability $n_k m / (\mu + n m)$,
the $k^\mathrm{th}$ type sends a successful migrant to a randomly chosen new island,
which is assigned the $k^\mathrm{th}$ allele.

This process is a continuous-time version of the ``Chinese restaurant process''
described in \citet{aldous1985exchangeability} and \citet{pitman1995partitions},
and so the final partition of types across demes has the Ewens sampling distribution with parameter $\mu/m$.
Again note that the selection coefficient, which is implicit in the probability $p_f$,
does not enter into the final distribution.

Now we can compute most properties we might want about the process.
For instance, the expected number of distinct mutational origins is
\begin{equation}
    \E\left[ \mbox{ \# of independent mutations } \right] = 
            1 + \frac{\mu}{m} \sum_{k=2}^L \frac{1}{k+(\mu/m)}. \label{discrete_expected}
\end{equation}
The probability that there is only a single successful mutation is
\begin{equation}
\P \left\{ \mbox{only one mutational origin} \right\} = 
            \prod_{k=1}^{L-1} \frac{ k }{ k+(\mu/m)} .
\end{equation}
More generally, suppose we sample a single deme, and are interested in the total number of demes $S$
(including the sampled one) that share the same fixed mutation as our sampled deme.
Then $S$ has distribution
\begin{equation} \label{eqn:Sdistrn}
\P\{ S=s \} = \frac{ (s-1)! (\mu/m)^{L-s} }{ \prod_{k=1}^{L-1} (k+(\mu/m)) } .
\end{equation}
Furthermore, if $L$ is large, then $S/L$ has probability density approximately
\begin{equation} \label{eqn:betadistrn}
   (\mu/m) (1-x)^{(\mu/m)-1}
\end{equation}
namely, is a $\mathrm{Beta}((\mu/m), 1)$ distribution--- see \cite{donnelly-joyce} and \cite{permanPitmanYor92}.

% In Figure \ref{fig:discbyratio} we plot mean number of types and size (number of demes occupied) of a sampled type,
% for a model with 20 interconnected demes, across different parameter values.  
% This indicates that we need the population-scaled mutation and migration rates to be within a factor of about 100 of each other for parallel adaptation to leave an interesting pattern. If migration is faster, then a single type is likely to take over, while if migration is weaker, each deme is likely to come up with its own type.

The high connectedness of the discrete deme island model means that the expected number of distinct alleles, (Equation \eqref{discrete_expected}), 
grows with the $\log$ of the number of demes. This strongly contrasts with the continuous spatial model where the local nature of dispersal means that doubling the species range will double the number of mutations expected. 
Furthermore, while in the continuous model the areas occupied by distinct mutations are similar in area, in this strong-selection discrete island model, 
only a few types tend to dominate even as the number of demes (and thus, distinct types) grows large (see Equations \eqref{eqn:Sdistrn} and \eqref{eqn:betadistrn}).

A more general model would allow migration only along edges of some graph connecting the demes.
In the low migration limit such a model still produces a partition distribution independent of the selection coefficient,
but it does not in general have the Ewens distribution.
Another extension would be to include the time alleles need to achieve an intermediate frequency,
along the lines of \citet{Navarro:03}, which would reintroduce dependence on the selection coefficient.

\texttt{XXX omit this? XXX}
We have treated the time during which the selected allele is at intermediate frequency in a deme (about $\log(\sigma^2 \rho)/s_b$ generations) as negligible, 
which ignores migration events that occur while the mutation is spreading through the population. 
we now sketch of a more realistic approximation.
For each occupied--unoccupied pair of demes, if the number of mutant alleles in the occupied deme is $n(t)>0$,
then at time $t$, migrants from the occupied deme pass the selected allele to the unoccupied deme at rate $2 n(t) m p_f(s_b)$,
while newly arisen mutants arise and fix in the unoccupied deme at constant rate $2 N \mu p_f(s_b)$.
The selected type should grow approximately as $n(t)=\exp(s(t-t_0))$.
Of course, to follow this model through, we will have to account for multiple types within a single deme and other complications,
but we can at least make a few observations.
One is that the final distribution of types is no longer independent of $s_b$, which enters through the dynamics $n(t)$.
However, it is clear that increasing the selection coefficient $s$ will decrease the number of types (independent mutational origins),
while on the other hand, increasing the deme size $N$ will increase the number of types,
and that the case presented above is at the extreme (see also \cite{Navarro:03} for deterministic approximations). 


\section{The equilibrium frequency}

One route to the ``equlibrium frequency'' of the allele outside the range where it is advantageous is as follows;
see \citet{slatkin} or \citet{barton,pollack} for other arguments in this case, or \citet{etheridge} or \citet{dawson} for general theory
Suppose that the population is composed of a finite number of small demes of equal size $N$ arranged in a regular grid,
and that selection (for or against) the allele is given by the function $s(x)$, with $x$ denoting the spatial location.
Then each individual at location $x$ reproduces after a random lifetime,
dying and giving birth in the same deme to a random number of offspring with distribution given by $X$;
who all migrate to a new location chosen as $x+R$
and replace randomly chosen individuals there.
If $x+R$ is outside of the range, then they perish.
Each individual's reproduction time is exponentially distributed, 
either with rate 1 if it carries the original allele, or with rate $1+s(x)$ if it carries the mutant allele.
Suppose that $R$ has mean zero and variance $\sigma^2$, and that $X$ has mean $\mu$ and variance $\xi^2$.
We also assume that the distribution of $X$ makes the resulting Markov chain irreducible and aperiodic.

Let $\Phi^N_t(x)$ be the number of mutant alleles present at location $x$ at time $t$,
and denote by $\delta_x$ a single unit at location $x$, so that e.g.~$\Phi^N_t + \delta_x/N$
is the configuration after a mutant allele has been added to location $x$.
For $0\le \phi \le 1$, we also denote by $\bar X_\phi$ the number of mutant alleles added if $X$ new offspring carrying mutant alleles
replace randomly chosen individuals in a deme where the mutant allele is at frequncy $\phi$ (i.e.~hypergeometric with parameters $(X,\phi)$);
similarly, $\tilde X_\phi$ is the number lost if the new offspring do not carry the allele (i.e.~hypergeometric with parameters $(X,1-\phi)$).
(We like to think of $\Phi^N_t$ as a measure, but it does not hurt to think of $\Phi^N$ as a vector;
we aren't providing the rigorous justification here.)
Then we know that for any sufficiently nice function $f(\phi)$ that
\begin{align} \label{eqn:discrete_generator}
  \begin{split} \frac{\partial}{\partial t} \E\left[ f(\Phi^N_t) \right] 
  &= N \sum_x \left\{ \E\left[ (1+s(x+R)) \Phi^N_t(x+R) \left( f\left(\Phi^N_t + \frac{\bar X_{\Phi_t(x+R)}}{N}\delta_{x}\right) - f(\Phi^N_t) \right) \right] \right. \\
     & \qquad  \qquad \left. {} + \E\left[ \left(1-\Phi^N_t(x+R)\right) \left( f\left(\Phi^N_t - \frac{\tilde X_{\Phi_t(x+R)}}{N}\delta_{x}\right) - f(\Phi^N_t) \right) \right] \right\}  \end{split} \\
     &= \mu \sum_x \E\left[ \left(\partial_{\phi(x)} f(\Phi_t) \right) \left\{ \Phi_t(x+R) - \Phi_t(x) + s(x+R) \Phi_t(x+R) (1-\Phi_t(x)) \right\} \right] + O\left(\frac{1}{N}\right).
\end{align}
This follows in part from $\E[\bar X_\phi] = \phi \mu$ and $\E[\tilde X_\phi] = (1-\phi)\mu$.
We can see two things from this:
First, since this is a first-order differential operator, the limiting of the stochastic process
is deterministic (check by applying to $f(\phi) = \phi(x)^2$ to find the variance).
Second, if we want to rescale space as well to get the usual differential equation, 
we need to choose $\var[R]=\sigma^2$ and $s(x)$ to be of the same, small, order, another way of seeing that $\sigma/\sqrt{s}$ is the relevant length scale.
More concretely, suppose that the grid size is $\epsilon \to 0$, that $\var[R] = \sigma^2 \epsilon$, and that $s(x)/\epsilon \to \gamma(x)$,
and suppose that $\Phi_t(x)$ is deterministic (and sufficiently differentiable) and let $\xi(t,x) = \Phi_{t/\epsilon}(x)$;
then the previous equation converges to the familiar form:
\begin{align}
  \partial_t \xi(t,x) = \mu \left( \frac{\sigma^2}{2} \partial_x^2 \xi(t,x) + \gamma(x) \xi(t,x) (1-\xi(t,x)) \right) .
\end{align}

The process $\Phi_t$ is an irreducible, aperiodic finite-state Markov chain with absorbing states at 0 and 1;
therefore, the inevitable outcome for finite $N$ is extinction of one type or another.
Setting the left-hand side to zero in \eqref{eqn:discrete_generator} gives us equations for the moments of the stationary distribution,
for instance, if $\phi(x) = \lim_{t \to \infty} \E[ \Phi_t(x) ]


\end{document} 
