\documentclass{article}
\usepackage{fullpage}
\usepackage{amsmath,amssymb}
\usepackage{marginnote}
\usepackage{graphicx}
\usepackage{color}
\usepackage[hidelinks]{hyperref}
\usepackage{etoolbox}

% make supp figures separate
\usepackage{caption}
\DeclareCaptionType{sfigure}[Figure]

\setcounter{secnumdepth}{0}

% turn this on to doublespace and put all figures at the very end.
\newif\ifsubmissionversion
% \submissionversiontrue
\submissionversionfalse

\newif\iflatexml\latexmlfalse %%%graham added; remove with latexml package

%%%%%
% Responses to reviews:
% set this to show line numbers and include responses to reviews or not
\newif\ifreviewresponses
\reviewresponsestrue  % include them
% \reviewresponsesfalse  % don't include them
\iflatexml\reviewresponsesfalse\fi  % they don't work so well in html
% \newcommand{\responsefile}{pbio-reviews-19sept12-responses.tex}  % name of the review responses file
\newcommand{\responsefile}{patchy-review-responses-second-round.tex}  % name of the review responses file

% override above choices (see makefile)
\ifdef{\submissionnofigs}{  % for submission, no figures
  \reviewresponsesfalse
  \submissionversiontrue
}{
}
\ifdef{\submissionfigs}{  % for reviewers, with figs
  \reviewresponsestrue
  \submissionversionfalse
}{
}
\ifdef{\arxiv}{  % for general use
  \reviewresponsesfalse
  \submissionversionfalse
}{
}

\ifreviewresponses
  \input{review-response-commands}
\else
  \newcommand{\linelabel}[1]{}
  \newcommand{\revpoint}[2]{}
  \newcommand{\includereviews}{}
\fi

% % Text layout
% \topmargin 0.0cm
% \oddsidemargin 0.5cm
% \evensidemargin 0.5cm
% \textwidth 16cm 
% \textheight 21cm

% Bold the 'Figure #' in the caption and separate it with a period
% Captions will be left justified
\usepackage[labelfont=bf,labelsep=period,justification=raggedright]{caption}

\ifsubmissionversion
    \usepackage[nolists,nomarkers,nohead]{endfloat}
    \usepackage{cite}
    \DeclareDelayedFloat{sfigure}[sss]{Supplementary Figure}
    % Use the PLoS provided bibtex style
    \bibliographystyle{plos2009}
    \newcommand{\citep}[1]{\cite{#1}}
    \newcommand{\citet}[1]{\cite{#1}}
\else
    \usepackage{natbib}
    \bibliographystyle{abbrvnat}
\fi



% Remove brackets from numbering in List of References
\makeatletter
\renewcommand{\@biblabel}[1]{\quad#1.}
\makeatother


% Leave date blank
\date{}

% \pagestyle{myheadings}
%% ** EDIT HERE **

\DeclareMathOperator{\var}{Var}
\DeclareMathOperator{\Exp}{Exp}
\DeclareMathOperator{\sn}{sn}
\DeclareMathOperator{\erf}{erf}
\DeclareMathOperator{\atan}{atan}
\renewcommand{\P}{\mathbb{P}}
\newcommand{\E}{\mathbb{E}}
\newcommand{\R}{\mathbb{R}}
\newcommand{\one}{\mathbf{1}}
\newcommand{\calE}{\mathcal{E}}
\newcommand{\calC}{\mathcal{C}}
\newcommand{\dconv}{\xrightarrow{d}}
\newcommand{\deq}{\stackrel{\scriptscriptstyle{d}}{=}}
\newcommand{\given}{\mid}
\newcommand{\st}{\,\colon\,}

\newcommand{\migrate}{\lambda_\text{mig}}
\newcommand{\mutrate}{\lambda_\text{mut}}
\newcommand{\Tmig}{T_\text{mig}}
\newcommand{\Tmut}{T_\text{mut}}

\newcommand{\gc}[1]{{\it\color{green}(#1)} }
\newcommand{\plr}[1]{{\it\color{blue}(#1)}}

% make references to named sections look better
\newcommand{\secref}[1]{{\emph{\nameref{#1}}}}

% short title:  Convergent Evolution on Patchy Landscapes
\title 
{
Convergent Evolution During Local Adaptation to Patchy Landscapes
}

\author{Peter L. Ralph$^1$ \\ email: pralph@usc.edu  \and Graham Coop$^2$ \\ email: gmcoop@ucdavis.edu }


\begin{document}

\maketitle
\begin{center}

$^1$ Computational Biology and Bioinformatics \\ 
University of Southern California, Los Angeles, CA \\
$^2$Center for Population Biology \& Department of Evolution and Ecology \\ 
University of California -- Davis, Davis, CA, 95616
\end{center}

% Keywords
% convergent evolution; parallel evolution; MC1r; local adaptation; landscape genetics; branching processes; reaction-diffusion equations

%%%%%%%%%%%
\section*{Abstract}
Species often encounter, and adapt to, many patches of locally similar environmental conditions across their range. 
Such adaptation can occur through convergent evolution if different alleles arise and spread in different patches, 
or through the spread of shared alleles by migration
acting to synchronize adaptation across the species.
The tension between the two reflects 
the degree of constraint imposed on evolution by the underlying genetic architecture 
versus how effectively selection and geographic isolation act to inhibit the geographic spread of locally adapted alleles.
This paper studies a model of the balance between these two routes to adaptation in continuous environments with patchy selection pressures.
We address the following questions: 
How long does it take for a novel, locally adapted allele to appear in a
patch of habitat where it is favored through mutation?
Or, through migration from another, already adapted patch?
Which is more likely to occur, as a function of distance between the patches?
How can we tell which has occurred, i.e., what population genetic
signal is left by the spread of migrant alleles?
To answer these questions we examine the family structure underlying migration--selection equilibrium
surrounding an already adapted patch,
in particular treating those rare families that reach new patches
as spatial branching processes.
This provides a way to understand the role of geographic separation between patches 
in promoting convergent adaptation and the genomic signals it leaves behind.
We illustrate these ideas using the convergent evolution of cryptic coloration in the rock pocket mouse, \textit{Chaetodipus intermedius}, as an empirical example.

\section*{Author Summary}
Often, a large species range will include patches where the species differs because it has adapted to locally differing conditions.
For instance, rock pocket mice are often found with a coat color that matches the rocks they live in,
these color differences are controlled genetically,
and mice that don't match the local rock color are more likely to be eaten by predators.
Sometimes, similar genetic changes have occurred independently in different patches,
suggesting that there were few accessible ways to evolve the locally adaptive form.
However, the genetic basis could also be shared if
migrants carry the locally beneficial genotypes between nearby patches,
despite being at a disadvantage between the patches.
We use a mathematical model of random migration to determine how quickly adaptation is expected to occur 
through new mutation and through migration from other patches,
and study in more detail what we would expect successful migrations between patches to look like.
The results are useful for determining whether similar adaptations in different locations are likely to have the same genetic basis or not,
and more generally in understanding how species adapt to patchy, heterogeneous landscapes.




%%%%%%%%%%% %%%%%%%%%%% 
\section*{Introduction}

% Such precise convergence points to the conservation of function of the genes underlying adaptive traits, 
% sometimes over deep time scales \citep{deephomologypapers}. 
% Convergence on a molecular level also suggests that the path 
% accessible to adaptation is sometimes relatively constrained \citep{stern2013genetic}, 
% as would be the case if only a few changes confer the advantageous phenotype 
% and are sufficiently free of deleterious pleiotropic consequences.

The convergent evolution of similar phenotypes 
in response to similar selection pressures is a testament to the power that selection 
has to sculpt phenotypic variation. 
In some cases this convergence extends to the molecular level, 
with disparate taxa converging to the same phenotype through parallel 
genetic changes in the same pathway, genes, 
or even by precisely the same genetic changes
\citep{stern2013genetic,zhen2012parallel,martin2013repeated}. 
Convergent adaptation also occurs within species, if different individuals
adapt to the same environment through different genetic changes. 
There are a growing number of examples of this in a range of well studied organisms and phenotypes \citep{Arendt:08}.
Such evolution of convergent phenotypes is favored by higher mutational input, 
i.e., higher total mutational rate and/or population size \citep{softsweepsII}.
The geographic distribution of populations can also affect the probability of parallel mutation within a species:
a widespread species is more likely to adapt by multiple, parallel mutations if dispersal is geographically limited, 
since subpopulations will adapt via new mutations before the adaptive allele arrives via migration \citep{ralph2010parallel}. 
Standing variation for a trait can also increase the probability of convergence, 
as this increases the probability that the selective sweep will be \emph{soft}
(beginning from a base of multiple copies),
which 
leads to genetic patterns similar to convergent adaptation \citep{orr2001sieve,softsweepsI}
whether or not the copies derive from the same mutation,

Intuitively, convergence is also more likely 
when geographically separated populations adapt to ecologically similar conditions. 
The probability that convergent adaptations arise independently
before adaptations spread between the populations by migration
will be larger if these adaptive alleles are maladapted in intervening environments,
since such adverse conditions can strongly restrict the spread of locally adapted alleles \citep{slatkin1973geneflow}.
 
%\citep{wright2013indirect}
One elegant set of such examples is provided by the assortment of plant species 
that have repeatedly adapted to patches of soil with high concentrations of heavy metals
(e.g., serpentine outcrops and mine tailings) \citep{kruckeberg1951intraspecific,macnair1991evolution,schat1996identical,turner2010serpentine};
the alleles conferring heavy metal tolerance are thought to be locally restricted 
because they incur a cost off of these patches.
Similarly, across the American Southwest, a variety of species of animals have developed locally adaptive cryptic coloration
to particular substrates, e.g., dark rock outcrops or white sand dunes
\citep{benson1933concealing}.
One of the best-known examples is the rock pocket mouse (\textit{Chaetodipus intermedius}),
which on a number of black lava flows has much darker pelage than on intervening areas of lighter rock \citep{DiceBlossom1937}.
Strong predator-mediated selection appears to favour such crypsis \citep{kaufman1974adaptive},
and, perhaps as a result of this strong selection against migrants, 
at least two distinct genetic changes are responsible for the dark pigmentation found on different outcrops \citep{hoekstra2003different}. 
Similar situations have been demonstrated in other small rodent systems \citep{dice1940ecologic,steiner2009genetic,kingsley2009melanism}
and in lizards \citep{rosenblum2010molecular}.

In this paper, we study this situation, namely,
when a set of alleles provide an adaptive benefit in geographically localized patches, 
separated by inhabited regions where the alleles are deleterious.
The main questions are:
Under what conditions is it likely that each patch adapts in parallel, 
i.e., convergently through new mutation,
and when is it likely that migration carries these alleles between patches?
How easy will it be to detect adaptive alleles that are shared by
migration, i.e., over what genomic scale will a population genetic
signal be visible?

We work in a model of continuous geography,
using a variety of known results and new methods.
In the section \secref{ss:patchymutation} we develop a simple approximation,
equation \eqref{eqn:mutrate}, for the rate at which patches become established by new mutations. 
The most important conceptual work of the paper occurs in the section \secref{ss:patchymigration},
where we develop an understanding of the process by which alleles move from an existing patch 
to colonize a novel patch, culminating in equation \eqref{eqn:migrate} for the rate at which this happens.
We combine these two results in the section \secref{ss:probparallel} 
to discuss the probability of parallel adaptation, equation \eqref{eqn:parallel_prob}.
To understand the genomic signal left by adaptations shared by migration, 
in the section \secref{ss:haplotype_length}, 
we study the time it will take for an allele to transit between patches,
(equation \eqref{eqn:mean_tau}),
and thus the length of haplotype that hitchhikes with it (equation \eqref{eqn:haplotype_length}). 
Finally, in the section \secref{ss:applications} we apply this work to understand the geographic scale of
convergent evolution in \textit{Chaetodipus intermedius}. 



%%%%%%%%%%%%%
\section*{Results}

%%%
\subsection{Model of a patchy landscape}
\label{ss:patchyspace}

Consider a population spread across a landscape to which it is generally well adapted,
but within which are patches of habitat to which individuals are (initially) poorly adapted.
(When we refer to ``patches'' it is to these pieces of poor habitat.)
Suppose it takes only a single mutational change to create an allele
($B$) that adapts an individual to the poor habitat type.
The required change could be as specific as a single base change, 
or as general as a knockout of a gene or one of a set of genes.
This change occurs at a (low) rate of $\mu$ per chromosome per generation,
and has fitness advantage $s_p$ relative to the unmutated type ($b$) in these ``poor'' habitat patches.
Outside of these patches the new allele has a fitness disadvantage of 
$s_m$ in the intervening areas, with $s_p$ and $s_m$ both positive.
(Here we define ``fitness'' to be the intrinsic growth rate of the type when rare.)
We assume that the disadvantage $s_m$ 
is sufficiently large that, on the relevant timescale,
the allele is very unlikely to fix locally in these intervening regions.
(The case where $s_m=0$ requires a different approach, which we do not treat here.)

We are interested in the establishment of mutations in the ``poor'' patches by either
migration or mutation, which depends on whether the allele
can escape initial loss by drift when rare. 
Therefore, we need not specify the fitness of the homozygote; 
only that the dynamics of the allele when rare 
are determined by the fitness of the heterozygote. 
More general dominance will only make small corrections to the dynamics until initial fixation,
with the exception of the recessive case, which we omit.
In other words,
we follow the literature in treating the diploid model as essentially haploid.

We also assume population density $\rho$ is constant across the range (even in the ``poor'' patches).
% If this is not the case, in most equations below, the population density that should be used is the density
% on the poor patches; see \citet{lenormand2002limits} for a more thorough approach.
The variance in offspring number is $\xi^2$, 
and that the mean squared distance between parent and child is $\sigma^2$
(i.e., $\sigma$ is the dispersal distance). 
We will deal with migration by immediately appealing to the central limit theorem,
treating the total distance traveled across $t$  dispersal events as Gaussian with variance $t \sigma^2$,
and do not discuss the (interesting) cases where rare, long-distance dispersal events
are more important (see e.g., \citet{levin2003ecology,ralph2010parallel,hallatschek2014acceleration} for discussion).  \linelabel{rr:longtailed}
%We also aim to describe the scenario where it is sensible to think of large, discrete patches;
%the case of a fine-scale heterogeneous environment is treated
%\citep{elsewhere}. \gc{IS THIS LAST SENTENCE NEEDED IF SO GIVE CITATION} \plr{haven't found it yet}

\paragraph{Past work on spatial models}
We will make use of several previous results from the literature on
migration in models of spatially varying selection.
\citet{haldane1948theory}, \citet{fisher1950frequencies}, and \citet{slatkin1973geneflow} first analyzed
deterministic models of of this sort.
\citet{nagylaki1975conditions} and \citet{conley1975application} showed that in one dimension, if the physical width of the patch is less than $(\sigma/\sqrt{2s_p}) \tan^{-1} (\sqrt{s_m/s_p})$, 
then there is no stable equilibrium with both $b$ and $B$ present,  % see p.603 in Nagylaki
so that migrational swamping prevents $B$ from establishing (see also \citet{lenormand2002limits} for a review).
\citet{barton1987establishment}, using general theory of \citet{pollak1966survival}, showed  
that patches must be at least this critical size so that a new mutation has positive probability of establishment
(in the large population density limit).
The same paper also showed that
this probability of establishment of a new mutation decays exponentially with distance from the patch, 
with the scale given by $\sigma/\sqrt{2s_m}$. 
Mutations appearing within the patch have probability of establishment
strictly less than the probability for a panmictic population,
but approaching this as the size of the patch increases.
This latter panmictic probability of establishment we denote $p_e$,
and often approximate $p_e$ by $2 s_p / \xi^2$ \citep{haldane1927mathematical,fisher1930genetical}.
This result holds quite generally for a geographically spread population that experiences a uniform selection
pressure \citep{maruyama1970fixation,cherry2003diffusion}. 
More general work on the mathematically equivalent problem of density-dependent population growth 
has obtained much more general criteria
under which a habitat configuration will support a stable polymorphic equilibrium \citep{cantrell1989diffusive},
and determined the habitat configuration that maximizes the probability of establishment \citep{lou2006minimization}.


%%%%%
\subsection[Establishment by Mutation]{Establishment of a locally adaptive allele due to mutational influx}
\label{ss:patchymutation}

We first compute the time scale on which a new $B$ mutations will appear and fix in 
a single, isolated poor habitat patch in which no $B$ allele has yet become established. 
As we are interested in patches where local adaptation can occur,
we will assume that the patch is larger than the cutoff for local establishment 
mentioned above.

Let $p(x)$ be the probability that a new mutant $B$ allele that arises at location $x$
relative to the center of the patch fixes within the poor habitat patch.
Under various assumptions, precise expressions can be found for $p(x)$ \citep{barton1987establishment},
% the function $p(x)$ can be found (in one dimension) in terms of Jacobi elliptic functions,
but results will be more interpretable if we proceed with a simple approximation.
The total influx per generation of mutations that successfully establish is 
the product of population density $\rho$, mutation rate $\mu$, and the
integral of $p(x)$ over the entire species range:
\begin{equation}
  \mutrate = \rho \mu \int p(x) dx \label{eqn:exactmutrate} .
\end{equation}
% $\int \rho \mu p(x) dx$,
This depends in a complicated way on the patch geometry and selection coefficients,
but still scales linearly with the mutational influx density $\rho \mu$.
If we consider a patch of area $A$, whose width is large relative to $\sigma/\sqrt{2s_m}$, 
then a reasonable approximation is to ignore migration, 
so that $p(x) = p_e \approx 2 s_p / \xi^2$ within the patch, and $p(x) = 0$ otherwise.
%for a patch of area $A$ approximating $\int p(x) dx \approx 2 s_p A / \xi^2$. 
This approximates the integrand $p(x)$ in \eqref{eqn:exactmutrate} by a step function,
which will be a good approximation if the patch is large relative the scale over which $p(x)$ goes from 0 to $p_e$,
or if $p(x)$ is close to $p_e$ at some point and is symmetric about the edge of the patch. \linelabel{rr:p_estab}
We examine this more generally via exact calculation of $p(x)$ in the section \secref{apx:establishment_sims}.

The rate at which mutations arise and colonize a patch of area $A$ is then
\begin{align} \label{eqn:mutrate} 
  \mutrate % = \rho \mu \int p(x) dx  
  \approx 2 s_p \rho A \mu / \xi^2,  
\end{align}
i.e., the product of mutational influx in the patch ($\rho A\mu$) and the probability of establishment ($p_e \approx 2s_p/\xi^2$).
If this rate is low, then the time (in generations) until a mutation arises that
will become locally established within the patch is exponentially distributed with mean $1/\mutrate$.  
Assuming that once a mutation becomes established it quickly reaches its equilibrium frequency across the patch, 
the time scale on which new patches become colonized by the $B$ allele from new mutation is therefore approximately $1/\mutrate$.


%%%%%%%
\subsection[Establishment by Migration]{Establishment of a locally adaptive allele due to migrational influx}
\label{ss:patchymigration}

Now suppose that there are two patches separated by distance $R$
(i.e., the shortest distance between the two is $R$). \revpoint{2}{8}
If the $B$ allele has arisen and become established in the first patch, but has not yet appeared in the second,
we would like to know the time scale on which copies of $B$ move between patches by migration
(supposing that no $B$ allele arises independently by mutation in the meantime).
To determine this, 
we study the fine-scale genealogy of alleles that transit between patches,
obtaining along the way other useful information about the genealogy of $B$ alleles. 
Doing this we arrive at equation \eqref{eqn:migrate} for the rate at which 
an allele established in one patch spreads to a neighboring one by migration.

Migration--selection balance ensures that 
there will be some $B$ alleles present in the regions where they are disadvantageous,
but only rarely, far away from the patch where $B$ is established.
Denote the expected frequency of allele $B$ at displacement $x$ relative to the patch by $q(x)$,
and assume that the shape of the patch is at least roughly circular.
Following \citet{haldane1948theory} or \citet{slatkin1973geneflow}, one can write down a differential equation to which $q(x)$ is the solution,
and show that $q(x)$ decays exponentially for large $|x|$,
with a square-root correction in two dimensions:
\begin{align} \label{eqn:eqfreq}
  q(x) \approx C \left( |x| \sqrt{2 s_m}/\sigma \right)^{-(d-1)/2} \exp( - |x| \sqrt{2 s_m} / \sigma) \qquad \text{for large $|x|$},
\end{align}
where $d$ is the dimension ($d=1$ or $2$), 
and $C$ is a constant depending on the geographic shape of the populations and the selection coefficients.
In applications we fit $C$ to the data;
to get concrete numbers from theory we take $C=1$ if necessary.

As J.\ Hermisson pointed out in comments on an earlier draft,
this functional form has a simple intuition:
local migration leads to the exponential decay,
since if each migrant at distance $x$ 
produces an average of $c$ descendants that make it to distance $x+1$, 
then in one dimension, the number of migrants must decay as $\exp(-cx)$;
and the square-root term in two dimensions 
enters because the available area grows with $x$.  
The ``renewal'' aspect of this same argument suggests that for equation~\eqref{eqn:eqfreq} to hold,
$|x|$ must be larger than a few multiples of $\sigma$.
\linelabel{rr:q_form}

These asymptotics fit simulations quite well, as shown in Figure~\ref{fig:sim_occupation_freqs}.
To be clear about the assumptions implicit here,
below we provide a derivation of \eqref{eqn:eqfreq} in section \secref{apx:eqfreq},  \revpoint{3}{5}
as well as justification for the asymptotics below in section \secref{apx:asymptotics}.
In one dimension, the equation can be solved to give $q(x)$ 
in terms of Jacobi elliptic functions; see the Supporting Information.

\begin{figure}[ht!]
  \begin{center}
    % made in patchy-sim-writeup-plots.R
    \includegraphics{sim-occupation-freqs}
  \end{center}
  \caption{
  \textbf{(A)} 
  Mean occupation frequencies of individual demes (grey dots)
  and averaged over demes at similar distances (black circles) from simulations, 
  on a one-dimensional grid of 201 demes, and
  a two-dimensional grid of $101\times 101$ demes.
  Superimposed in color is the decay predicted by \eqref{eqn:eqfreq},
  with $C$ chosen arbitrarily to give a reasonable fit: 0.85 and 0.11 respectively.
  (The disagreement at long distances is due to boundary effects near the edge of the range.)
  \textbf{(B)} 
  Fluctuations: colored lines are snapshots of allele frequencies at evenly spaced times
  from a one-dimensional grid of 151 demes;
  the solid line shows the cumulative mean occupation frequency across all generations.
  For both plots, the simulation was run for 1000 generations;
  the focal allele is beneficial in the central 10 demes, with $s_p=0.1$,
  and deleterious elsewhere, with $s_m=-0.02$; 
  demes had 1000 individuals in each;
  dispersal was to nearby demes with mean dispersal distances of 
  $\sigma = 0.95$ and 0.74 times the deme spacing, respectively.
  More details are given below in \secref{ss:simulations}.
   }   \label{fig:sim_occupation_freqs}
\end{figure}

This expected frequency $q(x)$ is the \emph{time-averaged} occupation frequency,
i.e., the total number of $B$ alleles found across $T$ generations near location $x$, per unit area, divided by $T$.
If, for instance, $q(x)=.01$ and the population density is $\rho=10$ individuals per unit area, 
then in a survey tract of unit area around $x$ we expect to find one individual every 10 generations
-- or, perhaps more likely, 10 individuals every 100 generations.
This points to an important fact: 
close to the original patch, the ``equilibrium frequency'' $q(x)$ describes well the density of the $B$ allele at most times,
but far away from the patch, 
the equilibrium is composed of rare excursions of families of $B$
alleles, interspersed by long periods of absence. 
An example of one of these rare excursions is shown in Figure~\ref{fig:sim_snapshots}.
The relevant time scale on which $B$ alleles migrate between patches is given by the rate of appearance of such families.
% To compute the rate of establishment by such migration,
% we argue that these families (suitably defined) can be approximated by
% independent branching processes whose spatial motion is described by Brownian motion.
% Properties of these, combined with equation \eqref{eqn:eqfreq},
% will lead us eventually to the expression \eqref{eqn:migrate} for the rate of effective migration,
% an approximation that is valid if the distance $R$ is large (at least several times $\sigma/\sqrt{2s_m}$).

\begin{figure}[ht!]
  \begin{center}
    \includegraphics{sim-snapshots}
  \end{center}
  \caption{
  A segment of the temporal dynamics of the same simulation as Figure~\ref{fig:sim_occupation_freqs}B.
  Space is again shown across the horizontal axis, with time on the vertical axis; 
  demes are colored darker red the greater their allele frequency at the corresponding time.
  The 50\% and 5\% frequency contour lines are in grey,
  and the genealogy of 50 individuals in a rare long-distance excursion is traced back with black lines.
  }   \label{fig:sim_snapshots}
\end{figure}

This suggests decomposing the genealogy of $B$ alleles into families in the following way.
First, consider the genealogy of all $B$ alleles that were alive at any time outside of the original patch.  
This is a collection of trees, each rooted at an allele living outside the patch whose parent was born inside the patch.
Next, fix some intermediate distance $r_0$ from the established patch,
and erase from the genealogy every allele that has no ancestors living further away than $r_0$ to the patch.
This fragments the original set of trees into a collection of smaller trees that relate to each other all $B$ alleles living outside of $r_0$ at any time,
and some living inside of $r_0$;
we refer to these trees as ``families''.
If $r_0$ is large enough that there are not too many families
and interactions between family members are weak,
then these families can be approximated 
by a collection of independent spatial branching processes
whose members are ignored if they enter the original patch,
illustrated in figure~\ref{fig:branching_decomp}.
(This can be made formal in the limit of large population density, also taking $r_0$ large enough that the chance of reaching the original patch is small.)
The opportunity for adaptation depends on how often these families of $B$ alleles encounter the new patch.
Suppose that the area occupied by the new patch is $S$.
We can learn about the rate of arrival of families at $S$ from
the time-averaged number of $B$ alleles expected in $S$ were it not a patch
(i.e., if $B$ was still maladaptive in $S$),
which from equation \eqref{eqn:eqfreq}
is 
\begin{align}
    \label{eqn:gestalt_q}
    \begin{split}
        \rho q(S) &= \text{ ( outflux of families ) } \times \text{ ( mean occupation in new patch per family ) } ,
\end{split}
\end{align}
where $q(S) = \int_S q(y) dy$.
The quantity we wish to compute,
the effective rate at which \emph{families of} migrant $B$ alleles establish in the new patch,
is
\begin{align}
    \label{eqn:gestalt_migrate}
    \begin{split}
        \migrate(S) &= \text{ ( outflux of families ) } \times \text{ ( probability a family establishes in new patch ) } .
    \end{split}
\end{align}
The quantities on the right-hand side depend implicitly on $r_0$,
but their product does not.



%%%% %%%%
\subsubsection{The genealogy of migrant families}
\label{ss:migrant_math}

To understand the process of adaptation by migration,
we therefore need to look more closely at the rare families of $B$ alleles 
far from the original patch.
It is reasonable to model each such family as a branching process,
i.e., assuming that each individual
migrates, then reproduces, independently of all other family members.
Migration takes the individual in a random direction
for a random distance whose mean square is $\sigma^2$,
and if the individual arrives to a patch where the $B$ allele is favored, 
we drop the individual from the family.
Since the $B$ allele is uniformly deleterious elsewhere,
we can decouple reproduction and migration
by first forming a tree in which each individual has a random number of offspring,
then assigning spatial locations to each individual in the tree,
and finally pruning any branches that wander into a patch.
This pruning removes both branches leading back into the already established patch (which are irrelevant)
and branches leading into the new patch (we will need to count these).
Since we are concerned with the rare migrant families that transit quickly between patches, 
we ignore back migrants to the already established patch. 
% will contribute little, since they are going the wrong direction;
% so to simplify the analysis we ignore this detail.  % but could add this detail in an appendix

Since each individual has on average $e^{-s_m}$ offspring,
the expected family size after time $t$ is $e^{- t s_m}$.
Therefore, if we define $k_e(t)$ to be the chance that the family
has gone extinct by time $t$, 
and let $K_t$ be the (random) family size conditioned on nonextinction,
then by Bayes' rule, $e^{-s_m t} = (1-k_e(t))\E[K_t]$.
It turns out that if the distribution of offspring numbers is not too heavy-tailed 
(see \citet{jagers1975branching} for details),
then $K_t$ has a limiting distribution: $K_t \dconv K$,
and 
\begin{align} \label{eqn:EK_limit}
    e^{-s_m t}/(1-k_e(t)) \to \E[K] \quad \text{as} \quad t \to \infty .
\end{align}
In other words, the chance of survival to time $t$ is asymptotically a constant multiplied by $e^{-s_m t}$,
and conditional on survival, the family size is given by a fixed distribution.

This suggests that families can be well-described by motion of a central ``trunk'',
which dies at a random time whose distribution is given by $k_e(t)$,
surrounded by ``branches'',
% and conditional on survival of this trunk, the family is composed of side ``branches'' off of this trunk,
as depicted in Figure~\ref{fig:branching_decomp}.
We can understand this with a decomposition described by \citet{geiger1999elementary}.
% (see also \citet{chauvin1991growing}). 
% (lyons1995conceptual deals with conditioned on eventual nonextinction)
When we condition on survival until $t$, we condition on survival of at least one trunk lineage
(the long red lineage in Figure \ref{fig:branching_decomp}).
Given this lineage, the other reproduction events are undistinguished --
% each individual along the trunk lineage must give birth to at least one offspring that survives until time $t$,
% and all other reproduction events are unconditioned.
% In other words, 
the genealogy of a family that survives until $t$
can be decomposed into a trunk that lasts until $t$
and that sprouts independent, unconditioned branches that may or may not survive until $t$.
Concretely, if we write $Z_t$ for the number of offspring in the branching process alive at time $t$,
% (so $\P\{Z_t = 0\} = k_e(t)$),
then
\begin{align} \label{eqn:Zt_decomposition}
  \left( Z_t \; \vert \; Z_t>0 \right) \deq 1 + \sum_{s=0}^t \sum_{k=1}^{W_s-1} Z^{(s,k)}_{t-s},
\end{align}
where each $Z^{(s,k)}$ is an independent copy of $Z$ 
that arose from the main branch in generation $s$,
and each $W_s$ is independent with distribution 
%% equation (2.3) in Geiger:
%% $\P\{ W_s = k\} = \sum_{j=k}^\infty \P\{ W = j \} k_e(t-s-1)^{j-k} (1-k_e(t-s-1))/(1-k_e(t-s))$, 
%% and equation (2.5) in Geiger:
that tends to $\P\{W_s=k\} \simeq \P\{W\ge k\}/\E[W]$ as $s \to \infty$,
where $W$ is the unconditioned offspring number distribution.  \linelabel{ll:branching_process_fix}
In particular, $\E[W_s-1] \simeq \E[W(W-1)]/\E[W]$;
if $W$ is Poisson then this is equal to $e^{-s_m}$.
Since each subfamily $Z^{(s,k)}$ is a branching process with mean offspring number less than one,
most of these die out quickly,
and so $Z_t$ is composed of a cluster of recently split subfamilies.
If we take the expectation of \eqref{eqn:Zt_decomposition},
since $\E[Z_t] = e^{-s_m t}$,
we get the limiting mean family size conditioned on nonextinction:
\begin{align} \label{eqn:mean_K}
    \E[K] 
    = 1 + \sum_{s=0}^\infty \E[W_s-1] e^{-s_m t}
    \approx \E[W(W-1)] / ( \E[W] s_m  )
\end{align}

Subfamilies typically live for $1/s_m$ generations,
thus spreading across an area of width $\sigma/\sqrt{s_m}$.
The oldest subfamily in a surviving migrant family 
is of order $\log(1/s_m)/s_m$ generations,
so the entire family is spread slightly wider.
\revpoint{2}{2}
% If $\P\{ T_k > k \} = e^{-sk}$
% then 
% \begin{align}
%   \P\{ max_{k \ge n} (T_k - k) \le 0 \} &\simeq \exp(-e^{-sn}/s) \\
%       &= \exp( - \exp( -s(n+\log(s)/s) ) ) \\
% \end{align}
% which is Gumbel with mean $(\log(s)+\gamma)/s$ and variance $\pi^2/(6s^2)$.

\begin{figure}[ht!!]
  \begin{center}
    \includegraphics{branching-concept}
  \end{center}
  \caption{
      \textbf{(a)} 
      Cartoon of the decomposition described in the text. 
    Migrants families that get further than $r_0$ from the already-established patch (grey circle)
    are each composed of a ``trunk'' lineage (in red), whose spatial motion we follow,
    embellished by transient branches.
    For the depicted long-lived family, these are further distinguished by those still alive at time $t$ (in black)
    and those that did not survive until $t$ (in grey).
    Figure~\ref{fig:sim_snapshots} has a depiction of such a family, from a simulation. 
    Note that the spatial motions of lineages are independent random walks
    (and hence should be much wigglier than in this cartoon, which confounds space and time).
      \textbf{(b)} 
      Diagram of the area $A'$ used in equation \eqref{eqn:migrate}:
      $R$ is the shortest distance from the old patch to the new,
      and $A'$ is the area of the portion of the new patch that is no more than distance $R+\sigma/\sqrt{2s_m}$
      from the old patch.
} \label{fig:branching_decomp}
\end{figure}


%%%% %%%%
\subsubsection{Heuristics}
\label{ss:heuristics}

The parallel form suggested by \eqref{eqn:gestalt_q} and \eqref{eqn:gestalt_migrate}
suggests that we can find $\migrate(S)$ by multiplying the mean density $q(S)$
by the ratio of the second terms in the two equations,
i.e.
\begin{align} \label{eqn:ratio_gestalt}
  \migrate(S) 
    &= \rho q(S) \times 
    \frac{
        \text{ ( probability a family establishes in $S$ ) }
    }{
        \text{ ( mean occupation density of a family in $S$ ) }
    } ,
\end{align}
where again the ``occupation density'' is computed \emph{without} the selective advantage.
To make this precise we need to understand what we mean by ``outflux of families'';
we do this below in section \secref{ss:hitting_occupation},
and provide a heuristic argument here.

Both numerator and denominator in \eqref{eqn:ratio_gestalt} 
count only families that have arrived at the new patch,
so we can consider a family conditioned on having successfully transited between the patches.
To obtain simply interpretable expressions,
we assume that once any member of the family arrives at the patch,
all other members are close by, and so have roughly equal opportunity
to establish in the new patch.

Suppose the family arrives with $K$ individuals.
The probability that at least one individual leaves a lineage that establishes in the patch
is $1-(1-p_e)^K$.
On the other hand, in the absence of the patch of new habitat,
each of the $K$ newly arrived individuals would leave behind a lineage with 
mean total size $\sum_{t=0}^\infty e^{-t s_m} \approx 1/s_m$,
roughly half of which would stay in the new patch.
Therefore, the expected contribution of the family to the occupation density is $K/2s_m$.
(For more careful consideration of the geometry of the patch
and the distinction between occupation density and number of individuals,
see section \secref{ss:hitting_occupation}, below.)

This suggests that 
\begin{align} \label{eqn:ratio_K}
  \migrate(S) 
    &= \rho q(S) \times 
    \frac{
        \E[1-(1-p_e)^K]
    }{
        \E[K]/2s_m
    } .
\end{align}
There are now two extremes to consider.
If $\P\{K>1/p_e\}$ is small, then each family has a small chance of establishment,
and the numerator is approximately $p_e \E[K]$.
On the other hand, if $p_e$ is not too small,
then the numerator would be close to 1, as any family that arrives will likely establish.

The remaining term is $q(S)$, the integral of equation \eqref{eqn:eqfreq} over the patch.
Since the patches are at distance $R$,
if the new patch has width smaller than $\sigma/\sqrt{2s_m}$, 
then $q(S)$ is approximately $q(R)$ multiplied by the area of the new patch.
If $S$ is larger, then we should use the area of the closest strip
of width $\sigma/\sqrt{2s_m}$ to the original patch;
see figure \ref{fig:branching_decomp}, and section \secref{apx:qS} for more details.
In either case, denote this area $A'$;
for numerical examples we compute $q(S)$ directly.


Combining these, and absorbing the term $\E[W(W-1)]/\E[W]$ from equation \eqref{eqn:mean_K} 
into the constant $C$,
\begin{align} \label{eqn:migrate}
  \migrate(R) &\approx C A' \rho s_m \min(s_m,p_e) \left( |R| \sqrt{2 s_m}/\sigma \right)^{-(d-1)/2} \exp( - |R| \sqrt{2 s_m} / \sigma)  \qquad \text{for } |R| > \sigma/\sqrt{2 s_m}  ,
\end{align}
where $A'$ is the area of the new patch no further than $R+\sigma/\sqrt{2s_m}$
from the old patch.
When we use this below, we set $p_e = 2 s_p / \xi^2$.
Note that the approximation has the undesirable property that it is
non-monotonic in $s_m$, when we expect the rate of adaptation via
migration to be a decreasing function of $s_m$. However, it only increases for small $s_m$, 
and is a decreasing function of $s_m$ for $s_m > |R|^2 / 2 \sigma^2$, 
which is the parameter region that we are restricted to by other assumptions.
\linelabel{rr:sm_condition}


%%%%%%%%%
\subsection{A review, and a test, of the assumptions}
\label{ss:assumptions}

Now that we have expressions for the mean rates of adaptation by new mutation, \eqref{eqn:mutrate},
and by migration from an already colonized patch, \eqref{eqn:migrate},
it seems helpful to step back and review the assumptions 
underlying the asymptotic results we have used, or will use below.
Our results should hold exactly in the limit of 
large, circular patches sufficiently far apart,
large population density,
and small selection coefficients of equal magnitude.

To be more precise, the mutation rate results most obviously apply if $s_m \approx s_p$,
if the patch diameter is large relative to $\sigma/\sqrt{\min(s_m,s_p)}$,
and the patch is not too far from circular.
If $s_m \ll s_p$ then a strip of width $\sigma/\sqrt{s_m}$
should be added to the outside of the patch in computing $A$ for equation \eqref{eqn:mutrate},
and if $s_m \gg s_p$, a strip of width $\sigma/\sqrt{s_p}$ should be subtracted from the patch. \revpoint{2}{6}
As for the migration rate,
we assume that 
each patch is large enough to support a stable population of $B$ alleles
($A^{1/d} > (\sigma/\sqrt{2s_p}) \tan^{-1}(\sqrt{s_m/s_p})$).
The geometry and size of the patch will also affect the approximation of equation \eqref{eqn:gA_1D}.
Next, in using equation \eqref{eqn:eqfreq}, we assume that 
the inter-patch distance is large relative to the characteristic length
($R > \sigma/\sqrt{s_m}$),
and that
local drift is not so strong that the equilibrium frequency is at least approximately attained
(using Wright's effective neighborhood size, $s_m \gg 1/(\rho \sigma^d)$).
The last requirement is necessary because if the $B$ allele fixes in large neighborhoods where it is deleterious,
we cannot approximate its dynamics via a branching process.
We also neglect the time for migration-selection equilibrium to be reached.
As discussed above, we also assume that migration to a new patch takes place over a number of generations;
if there are sufficient rare, long-distance migration events
that would move between patches in a single hop,
this would require a different analysis.



To test the robustness of our results to the various approximations we used,
we used individual-based simulations on a one-dimensional lattice of 501 demes,
with $\rho$ haploid individuals per deme,
dispersal to nearby demes with $\sigma=0.95$,
and run for 25,000 generations. 
More details are given below in section \secref{ss:simulations},
and the number of simulations used and parameter values are given in
supplementary tables \ref{stab:mutation_params} and
\ref{stab:migration_params}. 
To estimate the mean time until adaptation by mutation,
we used one centrally located patch of 99 demes
and a mutation rate of $\mu=10^{-5}$,
and to estimate the mean time until adaptation by migration,
we used two centrally located patches of 99 demes
separated by a varying number of demes,
with one patch initialized to have the $B$ allele at frequency 0.8.
In each case, we measured the ``time to adaptation'' as the amount of time until at least 100 $B$ alleles
were present in the previously unadapted patch.
Figure~\ref{fig:sim_times}
summarizes how the results compare to theory,
excluding parameter combinations 
that violate the assumptions discussed above, 
or where a majority of simulations did not adapt by 25,000 generations.
Supplementary figures \ref{sfig:sim_migration_times} and \ref{sfig:sim_mutation_times} 
show all times, 
and depictions of typical simulation runs are shown in supplementary figures
\ref{sfig:sims_1}, \ref{sfig:sims_2}, \ref{sfig:sims_3}, \ref{sfig:sims_4}, \ref{sfig:sims_5}, \ref{sfig:sims_6}, \ref{sfig:sims_7}, and \ref{sfig:sims_8}.

The agreement is reasonable for both.
The theoretical value $1/\mutrate$ underestimates the mean time to mutation
by a factor of around 2 that increases slightly with $s_m$.
This is to be expected for two reasons:
First, we compute the time to reach 100 $B$ alleles, while theory predicts
the time until the progenitor of those 100 $B$ alleles arose.
Second, the expression for $\mutrate$ neglects effects near the boundary of the patch,
and the larger $s_m$ is, 
the harder it is for mutations that arise near the edge of the patch to establish.
The theoretical values $1/\migrate$ again has a margin of error of a factor of about 2.
% with the exception of the weakest disadvantage off the patch, $s_m=10^{-4}$,
% which adapt much more quickly than predicted by theory.
% However, these simulations fall outside the region where we expect the model to do well,
% since $R < \sigma/\sqrt{s_m} \approx 100$.
Supplementary figure \ref{sfig:sims_8} shows simulations at more parameter values.
% since $s_m$ is so small, the second patch is reached by migrants before the migration-selection equilibrium is reached,
% and at migration-selection equilibrium, the two patches act as a single meta-patch.


\begin{figure}[ht!]
  \begin{center}
      % made in analyze-sims.R
      \includegraphics{times-predicted-observed}
  \end{center}
  \caption{
      \textbf{(left)} 
      Time until adaptation by mutation in an isolated patch for different combinations
      of selection coefficient on patches ($s_m$), 
      population density ($\rho$),
      mutation rate ($\mu$),
      and distance between patches ($R$),
      as observed in simulations (vertical axis),
      and as predicted from equation~\eqref{eqn:mutrate} (horizontal axis).
      In all simulations, $s_p=0.01$.
      \textbf{(right)} 
      Time until adaptation by migration from an already occupied patch to a new patch,
      as observed in simulations (vertical axis),
      and as predicted from equation~\eqref{eqn:migrate}, with $C=5$ (horizontal axis).
      For each,
      parameter combinations and numbers of simulations are in supplemental tables \ref{stab:mutation_params} and \ref{stab:migration_params};
      shown are the median (points) and interquartile range (lines)
      of the time until the patch had at least 100 adapted individuals,
      for each unique parameter combination.
      Only parameter combinations to which the theory here is applicable were used
      (precisely, if
      $R > \sigma/\sqrt{s_m}$, 
      patch width larger than $\sigma/\sqrt{s_m} \atan(\sqrt{s_m/s_p})$, 
      and $s_m > 2 \sigma \rho$);
      the remaining combinations are shown in figures
      S\ref{sfig:sim_migration_times} and S\ref{sfig:sim_mutation_times}.
  }   \label{fig:sim_times}
\end{figure}




%%%%%%%%%
\subsection[Probability of Parallel Adaptation]{The probability of parallel adaptation between patches.} 
\label{ss:probparallel}

We now turn to the question of whether the separated patches adapt by parallel genetic changes 
or by the spread of a migrant allele between the patches.
As only a single mutation is required for individuals to adapt to the
poor habitat patch, subsequent mutations that arise after an allele becomes established in the patch gain no selective benefit. 
Similarly, an allele introduced into a patch by migration will not be favored by selection to spread, 
if the patch has already been colonized. 
Therefore, mutations selectively exclude each other from patches, over short time scales, 
and will only slowly mix together over longer time scales by drift and migration,
an assumption we also made in \citep{ralph2010parallel}. 
In reality, different mutations will likely not be truly selectively equivalent,
in which case this mixing occurs over a time-scale dictated by the difference in selection coefficients.

We assume that once a $B$ allele is introduced (by migration or mutation) 
it becomes established in the poor habitat patch rapidly
if it escapes loss by demographic stochasticity.
Under this approximation, and the ``selective exclusion'' just discussed,
after one of the patches becomes colonized by a single $B$ allele, 
the other patch will become colonized by migration or mutation, but not by both. 
As such, the question of how the second patch adapts
simply comes down to whether a new mutation or a migrant allele is the first to become established in the second patch. 
To work out how adaptation is likely to proceed, 
we can compare expressions \eqref{eqn:mutrate} and \eqref{eqn:migrate} above
for the effective rates of adaptation by new mutation and by migration.
We work in one dimension, as the square root term appearing for two dimensions is relatively unimportant.
% (as seen in figure~\ref{fig:sim_occupation_freqs} by the fact that the $d=2$ curve is nearly straight).
\linelabel{rr:sqrt_term}

%%felt like we needed a better intro to this paragraph
We first consider the order of magnitude that our parameters need to
be on in order for adaptation via mutation or migration to dominate.
Let $\gamma = \min(1,s_m/p_e)$
and $w = A/A'$ 
-- since $A$ is the area of the not-yet-adapted patch and $A'$ is the area of its closest $\sigma/\sqrt{2s_m}$ strip,
$w$ is approximately the width of the patch in units of $\sigma/\sqrt{2s_m}$.
Effective migration and mutation rates will be on the same order if 
$A \mu \approx 2 A' \gamma s_m \exp(- R \sqrt{2 s_m} / \sigma)$,
where $R$ is the distance between the patches.
In other words, the migrational analogue of ``mutational influx'' ($\mu \rho A$) 
is $2 \rho A' \gamma s_m \exp(- R \sqrt{2 s_m} / \sigma)$,
which depends fairly strongly on the selective disadvantage $s_m$ between patches.
Equivalently, the rates are roughly equal if
$R/\sigma = \log(2 \gamma s_m/(w \mu))/\sqrt{2s_m}$,
which gives the critical gap size past which adaptation will be mostly parallel
in terms of selection coefficients, patch width, and mutation rate.

% cdist <- function (sm,z) { (-log(z)+log(2*sm))/sqrt(2*sm) }
% cdist(c(.05,.001),z=1e-5)
% cdist(c(.05,.001),z=1e-4)
% cdist(c(.05,.001),z=1e-3)
% cdist(c(.05,.001),z=1e-2)
% cdist(c(.05,.001),z=1)
% ss <- exp(seq(log(1e-6),log(1e-2),length.out=100))
% plot( ss, cdist(sm=.001,z=ss), type='l', log='x', xlab='mutation influx', ylab='R/sigma' )
% lines( ss, cdist(sm=.05,z=ss), col='red')
% abline(h=0,lty=2)

If we take 
$\mu=10^{-5}$,
the patch width to be ten times the length scale $\sigma/\sqrt{2s_m}$
so $w=10$ (and $-\log(w\mu)\approx 9.2$),
and $s_m>p_e$ so that $\gamma=1$,
then adaptation is mostly parallel for patches separated by gaps larger than $R/\sigma > (9.2+\log(2s_m))/\sqrt{2s_m}$.
If the selective pressure is fairly strong (say, $s_m=.05$),
then for convergence to be likely,
the distance between patches must be at least 21.8 times the dispersal distance (i.e., $R/\sigma > 21.8$).
If $s_m$ is much smaller, say $s_m = .001$, 
the required distance increases to $R/\sigma > 67$.

If the mutation rate is higher -- say, $\mu=10^{-3}$ --
the required separation between patches is only reduced to $R/\sigma > 7.3$ with $s_m=.05$.
If $s_m=.001$, then with this higher mutational influx this calculation predicts that mutation will \emph{always} be faster than migration
-- however, this should be taken with caution,
since as discussed above, this model does not hold if $R$ is of the same order as $\sigma$
or if $s_m$ is small enough that local drift is more important.
\linelabel{rr:separated_patches}

% One way to visualize this is by a phase diagram.
% In figure \ref{fig:phase_diagram},
% we give a contour plot for the time until adaptation by migration ($\Tmig = 1/\migrate$) and by new mutation ($\Tmut=1/\mutrate$)
% as a function of various parameters. 
% This gives us an idea of not only the parameter space where parallel adaptation is likely ($\Tmig \ll \Tmut$),
% but also where adaptation by any means takes a very long time ($\Tmig$ and $\Tmut$ both very large),
% an important check in applications of these results.
% 
% \begin{figure}[ht]
%   \begin{center}
%     \includegraphics{phase-diagram-log}
%   \end{center}
%   \caption{
%   Shaded contour plots of
%   mean time until adaptation in a new patch occurs by migration ($\Tmig$, blue) or by mutation ($\Tmut$, red),
%   as a function of population density ($\rho$), 
%   distance to an already adapted patch in units of dispersal distance ($R/\sigma$),
%   strength of deleterious selection ($s_m$),
%   and mutation rate scaled by patch area ($\mu A$).
%   The right-hand plot shows that $\migrate$ is relatively insensitive to $s_m$ (thanks to the square root).
%   The black, dotted line shows values where $\Tmig=\Tmut$.
%   Default parameter values were $A=100$, $\rho=10$, $\mu=10^{-8}$, $s_p=0.1$, $s_m=0.01$, $R/\sigma=20$, and $\xi^2=1$.
%   Note that population density $\rho$ and mutation influx $A\mu$ are shown on a log scale.
%   \label{fig:phase_diagram}
%   }
% \end{figure}

We can go beyond these rough calculations to find
the probability of parallel adaptation if we are willing to take our approximations at face value. 
Doing so, the time until the first of the two patches is colonized by the $B$
allele will be approximately exponentially distributed with mean $1/(2
\mutrate)$.
Following this, the time until the second patch is subsequently colonized 
(via either migration or new mutation) 
will be exponentially distributed with mean $1/(\mutrate+\migrate)$.
Both these rates scale linearly with population density ($\rho$) 
and the probability of establishment of a rare allele ($p_e\approx 2 s_b/\xi^2$, for $p_e>s_m$),
so that the overall time to adaptation is robustly predicted to increase with offspring variance ($\xi^2$)
and decrease with population density and beneficial selection coefficient.
Furthermore, the probability that the second adaptation is a new mutation,
i.e., the probability of parallel adaptation, 
with $\gamma = \min(1,s_m/p_e)$,
is 
\begin{equation} \label{eqn:parallel_prob}
    \frac{\mutrate}{\mutrate+\migrate(R)} = \frac{ w \mu / (2s_m) }{ w \mu / (2s_m) + C \gamma \left(R \sqrt{2 s_m} /\sigma \right)^{-(d-1)/2}\exp\left(- \sqrt{2 s_m} R / \sigma \right) },  
\end{equation}
so that the probability of parallel mutation should increase
approximately logistically with the distance between the patches, on the spatial scale $\sigma/\sqrt{2 s_m}$. \revpoint{2}{13}

We tested this using the same simulations as figure \ref{fig:sim_times},
by using the empirical distributions of the respective times to adaptation
to estimate the probability,
for each parameter combination,
that a new, successful mutation appears 
before a successful migrant arrives from another, already adapted patch.
The results are compared to \eqref{eqn:parallel_prob}
in figure~\ref{fig:sim_probs}.


\begin{figure}[ht!]
  \begin{center}
      % made in analyze-sims.R
      \includegraphics{prob-mutation-compared}
  \end{center}
  \caption{
      Probability that two patches 
      evolve by independent mutations (i.e., convergently),
      as a function of their geographic separation ($R$)
      and the strength of selection between patches ($s_m$).
      The background shades of grey and contours
      give the probability predicted by theory (equation~\eqref{eqn:parallel_prob}, with $C=5$).
      Points correspond to parameter combinations of the simulations in figure \ref{fig:sim_times},
      with the shade of grey and the red numeric label indicating the same probability
      estimated from these simulations (see text for details).
      The two densities, $\rho=100$ and $\rho=1000$ would lie directly on top of each other, so the $\rho=1000$ points have been shifted vertically.
  }   \label{fig:sim_probs}
\end{figure}


%%%%%%%%%%%%
\subsubsection{Multiple patches}

Suppose that instead of patches there are many,
each patch adapting through either migration or mutation.
If the times between successive establishments are the result of many nearly-independent attempts
with small probability of success,
the process of occupation is well approximated by a Markov chain,
whose state records the labeling of patches according to the colonizing allele
(or none, for those that haven't yet adapted).

If not yet adapted, a focal patch of area $A$ will acquire the adaptation by new mutation at rate $2 s_p \mu \rho A/\xi^2$ (equation \eqref{eqn:mutrate}).
Suppose that patches $1, \ldots, k$ are already adapted,
and that these patches are at distances $R_1, \ldots, R_k$ away from this unadapted patch.
The total rate of adaptation through migration, from equation \eqref{eqn:migrate}
if patches do not interfere with each other, \linelabel{rr:interference}
is
\begin{equation} \label{eqn:total_migrate}
    \sum_{i=1}^k \migrate(R_i) = C \rho A' s_m \min(s_m,p_e) \sum_{i=1}^{k} \left(R_i \sqrt{2 s_m} /\sigma \right)^{-(d-1)/2} \exp\left(- R_i \sqrt{2 s_m} /\sigma\right).
\end{equation}
If the patch adapts through migration, the probability the adapted allele 
comes from patch $i$ is 
\begin{align}   \label{eqn:migprob}
  \frac{
      \left(R_i \right)^{-(d-1)/2} \exp\left(- R_i \sqrt{2 s_m} 
      /\sigma\right)
  } {
      \sum_{j=1}^{k}  \left(R_j \right)^{-(d-1)/2} \exp\left(- R_j \sqrt{2 s_m}
    /\sigma\right) 
} .
\end{align}

Since the compound parameter $s_p \rho$ is common to both migration and mutation rates,
it functions only as a time scaling of the process, 
and therefore has no effect on the final configuration, namely, 
the sets of patches sharing the genetic basis of their adaptive response.
We can also rescale time by the typical patch size, and introduce a parameter, say $a_k = A_k/\bar A$,
making the properties (other than the time scaling) of the discrete model independent of the \emph{numerical sizes} of the demes themselves.
This is complementary to the results of \cite{softsweepsII}, who showed that multiple mutations are likely to arise \emph{within} a panmictic deme
if the population-scaled mutation rate $2 N \mu$ is greater than $1$.

We can say more if we suppose that all patches have the same area and are the same distance from each other
(i.e., the ``island model'').
The resulting process is a continuous-time version of the ``Chinese restaurant process''
described in \citet{aldous1985exchangeability}, % and \citet{pitman1995partitions},
and so the final partition of types across demes has the Ewens
sampling distribution with parameter $w \mu / (4 s_m \exp (-\sqrt{2 s_m}R/\sigma))$.
Now we can compute most properties we might want about the process.
For instance, the proportion of demes that shares the same origin as a randomly sampled deme
is approximately Beta distributed -- see \cite{donnelly1989continuity}.
The high connectedness of the discrete deme island model means that the expected number of distinct alleles
grows with the $\log$ of the number of demes.
This strongly contrasts with the continuous spatial model 
where the local nature of dispersal means that doubling the species range will double the number of mutations expected. 


\begin{figure}[ht]
  \begin{center}
      \includegraphics{sim-transit}
  \end{center}
  \caption{
  \textbf{Adaptation by migration:}
  An initially unadapted patch (10 demes at the left end of the range) is colonized around generation 5000;
  a representative lineage is shown in the new patch, which follows the original migrant ``trunk'' lineage
  between the patches and coalesces with a lineage in the unadapted patch.
  In the left panel, as in Figure~\ref{fig:sim_snapshots}, color and contours show allele frequency across space and time;
  other simulation parameters are as in Figures~\ref{fig:sim_occupation_freqs}B and \ref{fig:sim_snapshots}.
  The right panel shows (in red) the extent of the shared haplotype between the two lineages around the selected locus (at position 0),
  which decreases as time progresses. 
  \label{fig:lineagesmotion}
  }
\end{figure}




%%%%%%%%%
\subsection[Haplotypes Shared Between Patches]{Length of the hitchhiking haplotype}
\label{ss:haplotype_length}


If a patch adapts through new mutation or a rare migrant lineage, the
genomic region surrounding the selected site will hitchhike along with it \citep{maynardsmith1974hitchhiking},
so that at least initially, all adapted individuals within a patch
share a fairly long stretch of haplotype. 
Pairs of adapted individuals within a patch will initially share a haplotype of mean genetic length of
about $\log(\rho A s_p)/s_p$ around the selected site, 
as long as the patch is reasonably well mixed by dispersal
(otherwise see \citet{barton2013genetic}).
This association gets slowly whittled down by recombination during interbreeding at the edge of the patch,
but there will always be longer LD nearby to the selected site \citep{barton1979geneflow}.

When an already adapted patch colonizes another through migration,
 the newly colonized patch will inherit a long piece of haplotype around the selected site from the originating patch.
The genetic length of this haplotype is roughly inversely proportional to the
number of generations the allele spends ``in transit'', 
because while very rare, the allele will be mostly in heterozygotes,
and so each generation provides another opportunity for a different haplotype to recombine closer to the transiting $B$ allele.
A large, linked haplotype may still arrive and fix in the new patch,
in which case the haplotype has literally hitchhiked across geography!
Figure~\ref{fig:lineagesmotion} shows a simulation of such an event,
including the lineage that founds an adaptive population on a second patch,
and the length of the haplotype shared between this lineage and one in the original patch.
The time that the lineage is outside the region where the $B$ allele is common (dark red in Figure~\ref{fig:lineagesmotion}), 
the haplotype that accompanies it is broken down rapidly. 
After the lineage establishes on the patch, the rate of decay of the haplotype is slowed significantly, 
since most others with which it recombines have similar haplotypic backgrounds. 
% We first discuss the transit of the lineage, and its accompanying haplotype, and return
% to how haplotypes are whittled down within established patches in the next section.


Above we argued that a good model for this transit is a single Brownian trunk lineage
surrounded by a cloud of close relatives of random size $K$
whose chance of surviving until $t$ generations is $1-k_e(t)$.
Consider a single such family, and let $\tau$ be the (random) time at which it first hits the new patch,
with $\tau = \infty$ if it never does. 
We assume that the length of the hitchhiking segment depends only on 
the time spent by the trunk lineage of the first successful migrant ``in transit'' between the patches,
i.e., $\tau$ conditioned on $\tau < \infty$.
In so doing, we neglect 
recombination between $B$ alleles (since they are at low density in transit),
and the possibility that more than one successful migrant family is in transit at once
(so that faster migrants would be more likely to have arrived first).
% These factors should have a minor impact if the rest of our framework fits.

Since each generation spent in transit provides an opportunity for recombination,
if recombination is Poisson, the length of the haplotype (in Morgans)
initially shared between populations on each side of the selected locus is exponentially distributed
with mean $\tau$. 
Therefore, if $L$ is the length of hitchhiking segment on, say, the right of
the selected locus, then
\begin{equation} \label{eqn:haplen_cdf2}
\P\{L>\ell\} = \E[e^{-\ell \tau}|\tau<\infty] .
%  &= \exp\left\{{-\frac{R}{\sigma}\left(\sqrt{2(\ell+s_m)} - \sqrt{2s_m}\right)}\right\} .
\end{equation}

Computing this depends on $1-k_e(t)$, 
the probability a family survives until $t$.
Since $1-k_e(t) \simeq e^{-s_m t} / \E[K]$,
we approximate the lifetime distribution of a family by an exponential with mean $1/s_m$.
% i.e., that it has exponential tails.
% This suggests one further approximation, 
% that the random lifetime of a family is exponentially distributed with mean $1/s_m$;
% equation \eqref{eqn:mean_K} shows this introduces a constant into the equations below. 
\linelabel{rr:EK}
We can then use standard results on hitting times of $d$-dimensional Brownian motion
that is killed at rate $s_m$ (see \citet{borodin2002handbook} 2.2.0.1 and 4.2.0.1).
In particular, if the patch is circular with radius $w$ and lies at distance
$R$ from the already adapted patch, then 
\begin{align}
  \E[e^{-\ell \tau}] &=
    \begin{cases}
      e^{- R \sqrt{2(s_m+\ell)}/\sigma} \qquad & d=1 \\
      \frac{ K_0( (R+w)\sqrt{2(s_m+\ell)}/\sigma) }{ K_0( w\sqrt{2(s_m+\ell)}/\sigma) } \qquad & d=2  ,
    \end{cases} \label{eqn:borodinresult}
    % &= \left( R/w +1 \right)^{-\frac{d-1}{2}} e^{- R \sqrt{2(s_m+\ell)}/\sigma}  (1+O(1/w)) .
\end{align}
where $K_0$ is a modified Bessel function of the second kind.
We are interested in lineages that manage to reach the patch before being killed,
i.e., having $\tau < \infty$,
which occurs with probability
$\P\{\tau < \infty\} = \lim_{\ell \to 0} \E \left[\exp(-\ell \tau) \right]$. 

To keep the expressions simple, in the remainder of this section we only compute quantities for one dimension.
By Bayes' rule, 
\begin{equation} \label{eqn:haplen_cdf}
    % \E[e^{-\ell \tau}|\tau<\infty]  =
    \P\{ L > \ell \}
    = 
    \exp\left\{{-\frac{R}{\sigma}\left(\sqrt{2(\ell+s_m)} - \sqrt{2s_m}\right)}\right\} .
\end{equation}
This can be differentiated to find that the expected transit time is
\begin{equation} 
  \E[\tau \given \tau<\infty ] = (R\sqrt{2s_m}/\sigma)\times 1/(2s_m) \label{eqn:mean_tau}
\end{equation}
and that $\var[\tau \given \tau<\infty ] = (R\sqrt{2s_m}/\sigma) \times 1/(2s_m)^2$.
(A saddle point approximation provides an alternative route to these expressions.) \revpoint{1}{4}

% \simeq (R/w+1)^{-(d-1)/2} \exp(- R \sqrt{2s_m}/\sigma)$.

% Above, we approximated this by the time $\tau$ it takes the trunk lineage to hit the new patch;
% now that we are studying a successful migrant family means we need to condition on the establishment occurring, i.e., on $\tau < \infty$.
% By the above arguments, and equation~\eqref{eqn:estab_prob}, we have that the conditional probability density is
% \begin{align}
%   \P\{\tau\in dt | \tau<\infty \} &= \frac{\P\{\tau \in dt\}}{\P\{\tau<\infty\}} \\
%   &= \frac{R}{\sigma^3 t^{3/2}\sqrt{2\pi}} \exp\left(-\frac{R^2}{2t\sigma^2} -s_m t + \sqrt{s_m} \frac{R}{\sigma} \right) .
% \end{align}
% \plr{more dimensions}

The form of equation \eqref{eqn:haplen_cdf} 
implies that if $Y$ is an exponential random variable with rate $R\sqrt{2}/\sigma$,
then $L$ has the same distribution as $(Y + \sqrt{s_m})^2 - s_m$.
Furthermore, the expected length of shared hitchhiking haplotype is
\begin{equation} \label{eqn:haplotype_length}
\E[L] = \sigma \sqrt{2s_m}/R + \sigma^2/R^2
\end{equation}
and $\var[L] = 2s_m\sigma^2/R^2 + 4 \sigma^3 \sqrt{2s_m}/R^3 + 5 \sigma^4 / R^4$.
For two dimensions, asymptotics of Bessel functions show that $L$ has the same tail behavior,
but how other properties might change is unclear. 

As a rule of thumb, equation~\eqref{eqn:mean_tau} means that families who successfully establish
move at speed $\sqrt{2s_m} \sigma$ towards the new patch 
-- if $s_m$, the strength of the selection against them, is smaller, the need to move quickly is less imperative.
Then, equation~\eqref{eqn:haplotype_length} means that the haplotype length
is roughly the length one would expect given the mean transit time,
so more weakly deleterious transiting alleles arrive with them shorter haplotypes.
Note that we have become used to seeing $\sigma$ divided by $\sqrt{2s_m}$, rather than multiplied;
it appears here because $\sigma/\sqrt{2s_m}$ is a length scale, 
and to convert it to a speed this is divided by $1/2s_m$. 

% In Appendix~\ref{apx:limiting_dist_of_HH_haplotype}, 
% we also give some expressions for the limiting distribution of the length of shared haplotype.

% 
% \section{Limiting distribution of the transit time and hitchhiking haplotype}
% \label{apx:limiting_dist_of_HH_haplotype}
% 
% In the above we already assumed that the inter-patch distance $R$ is large relative to the migration-selection cline width $\sigma/\sqrt{s_m}$;
% also taking the large-$R$ limit, obtains a more intuitive description of the process
% via a central limit theorem for $\tau$.
% Define $\eta_\tau$ to be the deviation of $\tau$ from its mean, normalized by its standard deviation,
% so that $\tau = R/\sigma(2s_m)^{-1/2} + \eta_\tau \sqrt{R/\sigma} (2s_m)^{-3/4}$.
% Plugging this into equation~\eqref{eqn:haplen_cdf},
% after some algebra (and using that $\sqrt{R+\epsilon} = \sqrt{R} + \frac{\epsilon}{2\sqrt{R}} - \frac{\epsilon^2}{2\sqrt{R}^3} + O(\epsilon^3)$) we obtain that
% \begin{align}
%   \E[e^{-\ell \eta}] &= \exp\left\{ \frac{\ell^2}{2} + O\left(\frac{1}{R}\right) \right\}.
% \end{align}
% Since the Laplace transform uniquely determines a distribution \citep{durrett1996probability},
% this implies that $\eta_\tau$ converges in distribution to a standard Gaussian random variable as $R \to \infty$.
% In other words, we now know that the distribution of $\tau$ about its mean is approximately Gaussian.
% Since the standard deviation of $\tau$ is of lower order than its mean,
% a similar calculation shows that for large $R$, the haplotype halflength $L$ is approximately exponentially distributed, 
% with mean $\sqrt{2s_m} \sigma/R$.



% \begin{align}
%   \E[e^{-\ell \eta}] &= \E\left[ \exp\left\{-\ell\left(\tau - \frac{R}{\sigma\sqrt{2s_m}}\right)\frac{(2s_m)^{3/4}\sqrt{\sigma}}{\sqrt{R}} \right\} \right] \\
%   &= \E\left[ \exp\left\{ - \left(\frac{\ell (2s_m)^{3/4} \sqrt{\sigma}}{\sqrt{R}}\right) \tau \right\} \right] \exp\left( (2s_m)^{1/4} \sqrt{\frac{R}{\sigma}} \right) \\
% \end{align}




%%%%%%%%%%
\subsection{Applications} 
\label{ss:applications}
% FOR REFERENCE:
% On Wed, May 28, 2014 at 10:20 AM, Michael Nachman <mnachman@berkeley.edu> wrote:
% > Graham:
% >
% > The density is probably in the range of 10-40 animals per hectare (10,000
% > m^s).  We built enclosures of 0.25 hectare and found that they could support
% > about 10 animals with food provided ad libitum, so I think this is an upper
% > density (i.e., 40/hectare).  As for dispersal distance, I'm less sure.
% > Studies on closely related species (genus Perognathus) in similar habitats
% > suggest distances of 100's of feet.  Attached are some old papers that tried
% > to estimate dispersal distances in Perognathus.
% > best,
% > Michael
%
% 
% On Mon, Aug 31, 2015 at 1:42 PM, Michael Nachman <mnachman@berkeley.edu> wrote:
% > Peter:
% >
% > I think this is better, but I would even be a little more cautious, e.g.:
% > "SOME of the largest patches of
% > dark rock range from 10km to 100km wide and 50-400km from each other, and
% > melanic populations of C. intermedius have been found on many of these
% > (but not all) [Benson, Dice & Blossom]. However, patches of all sizes
% > occur across all scales in a heterogeneous manner, so it is not clear
% > to what degree these should be treated as isolated from each other."
% >
% >
% > Note that I changed the numbers a little.  For example, Armendaris and
% > Carrizozo are less than 100 km from each other, and both are big.

Coat color in the rock pocket mouse
\emph{Chaetodipus intermedius} is a classic example of local
adaptation 
\citep{benson1933concealing,DiceBlossom1937}.
These mice live on rocky outcrops throughout the southwestern United States and northern Mexico, 
and generally have a light pelage similar to the predominant rock color.
However, in some regions these mice live on darker substrates (e.g., old lava flows),
and in these patches have much darker pigmentation, 
presumably to avoid visual predators.
Some of the largest patches of dark rock range from 10km to 100km wide
and lie 50--400km from each other,
and dark-colored populations of \emph{C.\ intermedius} have been found on many of these.
(However, patches of all sizes occur across all scales in a heterogeneous manner.)
\revpoint{2}{7}
\citet{nachman2003genetic} demonstrated that on one of these flows (Pinacate), 
much of the change in coloration is due to an allele at the MC1R locus.
This dark allele differs from the light allele by four amino acid changes, 
and has a dominant or partially dominant effect depending on the measure of coat color.
The Pinacate allele is not present in a number of other populations with dark pelage,
suggesting these populations have adapted in parallel \citep{nachman2003genetic,hoekstra2003different}.
However, \citet{hoekstra2004local} reasoned that, elsewhere in the range, 
multiple close dark outcrops may share a dark phenotype whose genetic basis has been spread by migration
despite intervening light habitat. 

A key parameter above was the dispersal distance divided by the square root of strength of selection against the focal allele between patches, $\sigma/\sqrt{s_m}$. 
\citet{hoekstra2004ecological} studied the frequency of the dark MC1R allele and coat color phenotypes, 
at sites across the (dark) Pinacate lava flow and at two nearby sites in light-colored rock.
On the lava flow the dark MC1R allele is at 86\% frequency,
while at a site with light substrate 12km to the west (Tule), the frequency is 3\%.
The dark allele was entirely absent from Christmas pass, a site with light substrate 7km north of Tule, and 3km further from the lava flow.
In the other direction, the dark MC1R allele was at 34\% at a site with light substrate 10km to the east of the flow (O'Neill).
Note that such apparent asymmetry is expected,
since as noted above the migration--selection equilibrium can be highly stochastic.
These numbers give us a sense of a plausible range of the parameters.
Assuming the dark allele is at 50\% frequency at the edge of the lava flow, 
we can fit formula \eqref{eqn:eqfreq} to these frequencies
(similar to \citet{hoekstra2004ecological}).
Doing this for Tule we obtain $\sigma/\sqrt{s_m} \approx 3$km, 
and for O'Neill, $\sigma/\sqrt{s_m} \approx 30$km, giving us a range of cline widths. 
We also need estimate of $s_m$. 
Using $\sigma \approx 1$km \citep{french1968dispersal,allred1963range}, 
these cline widths imply that $s_m=1/9$ and $1/900$ are reasonable values.

\begin{figure}[ht]
  \begin{center}
    \includegraphics{Lava_flow_mice_prob_parallel}
  \end{center}
  \caption{
Top panel: The probability of parallel mutation as a function of the distance
between environmental patches for two different cline widths
and two different mutation rates (using equation \eqref{eqn:parallel_prob}). 
The parameters were chosen to match the example of \textit{Chaetodipus intermedius};
see text. Bottom panel: The initial genetic length of the haplotype
shared between patches due to adaptation via migration as a function
of the distance between environmental patches. As a rough rule of
thumb $1$cM is approximately $10^6$ bases in a number of mammalian species. 
  \label{fig:mice_prob_parallel}
  }
\end{figure}

The mutational target size $\mu$ for the trait is unclear. 
While the Pinacate dark haplotype differs from the light haplotype at four amino acid residues,
it is likely that not all of these changes are needed for a population to begin to adapt. 
Also, there a number of genes besides MC1R at which adaptive changes affecting pigmentation 
have been identified in closely related species and more broadly across vertebrates \citep{Hoekstra:06}.
To span a range of plausible values, we use a low mutation rate of $\mu= 10^{-8}$
(a single base pair),
and a high mutation rate $\mu= 10^{-5}$
(a kilobase).
Finally, we set $A=100\text{km}^2$ (roughly the size of the Pinacate
patch). In
Figure~\ref{fig:mice_prob_parallel} we show the dependence of the probability
of parallel mutation on the distance between lava flow patches using these parameters, 
showing that parallel mutation should become
likely over a scale of tens to a few hundred kilometers between patches. 

Given the large selection coefficient associated with the dark allele
on the dark substrate, we expect the initial haplotype associated with
either a new mutation or migrant allele to be large. 
Figure~\ref{fig:mice_prob_parallel} also shows how long the founding haplotype 
shared between populations is expected to be,
from equation \eqref{eqn:haplotype_length}.
The initial length can be quite long between geographically close patches (tens of kilometers). 
However, for the wider cline width ($\sigma/\sqrt{s_m} = 30$km), 
adaptation by migration can still be likely for patches 100km apart, 
but the shared basis may be hard to detect, as the length of shared haplotype can be quite short. 

%\gc{DELETE THIS: 
% Furthermore, given the relatively small radius of the patches of dark substrate (often $<10$km),
% we should expect the haplotype to be rapidly whittled down within
% patches after establishment (equation \ref{eqn:rec_rate_within}). We
% estimate that the effective recombination rate is no less than 50\% of the actual recombination rate.
% Indeed, if we take the population density to be roughly $\rho \approx 10$ per hectare 
% (M.~Nachman, personal communication),
% and the variance in offspring number to be $\xi^2 = 3$ (mostly to get round numbers),
% then the shared haplotype length within patches from equation \eqref{eqn:within_haplotype_length}
% is only expected to be on the order of hundreds or thousands of bases
% ($10^{-3}$--$10^{-4}$cM).
% This suggests that unless the migration has happened fairly recently
% then it will be difficult to identify shared haplotypes
% (how recently depends on the background level of LD in the population).}


%%%%%%% %%%%%%% %%%%%%%
\section*{Discussion} 
\label{ss:discussion}

This paper is an investigation into the basic question: 
What is the spatial resolution of convergent local adaptation?
In other words, 
over what spatial scale of environmental patchiness will the process
of adaptation develop independent solutions to evolutionary problems?
The answer to this depends most strongly on $\sigma/\sqrt{s_m}$, 
the dispersal distance divided by the square root of the strength of selection against the allele between patches. 
It depends much more weakly on the
selective benefit within the patches or (perhaps surprisingly) the population density, 
although these two factors will determine the time-scale over which adaptation will occur
(and note that population density could affect the selection coefficients). \linelabel{rr:popdensity}
This is in contrast with models of panmictic populations
\citep{softsweepsI, MesserPetrov, wilson2014selective} 
and geographically spread populations adapting to homogeneous selection pressures \citep{ralph2010parallel}, 
where the probability of multiple, independently arising adaptive alleles increases with the population size.
However, in all of these models the dependence on the beneficial selection
coefficient is absent or weak, due to the fact that selection both
aids establishment of new alleles and the spread of existing
alleles (but see \citet{wilson2014selective} for the complications of varying population sizes). 

We have also shown that
while weaker selection against alleles will make sharing 
of adaptations between patches easier, 
it also makes it harder to spot such sharing,
since the lucky alleles that manage to colonize new patches move slower,
and thus carry a shorter shared haplotype.
This issue is amplified by the fact that the length of haplotype
shared within patches decays over time, potentially making the
identification of shared adaptations to old selection pressures difficult.

Perhaps the most useful rule-of-thumb quantities we found were the following.
The effective rate of migration into an as-yet-unadapted patch from an already-adapted patch distance $R$ away 
-- the analogue of the mutational influx $\mu \rho A$ --
is $2 \rho A' s_m \exp(- R \sqrt{2 s_m} / \sigma)$.
Equivalently, the critical gap size between patches past which adaptation is likely independent
is $R = (\sigma/\sqrt{2s_m}) \log(2s_m/(w \mu))$.
Finally, successfully transiting migrant lineages move at rate $\sigma \sqrt{2s_m}$,
and so shared haplotype lengths between patches will be of order $R/(\sigma \sqrt{2s_m})$.

In developing the set of approximations in the paper we have ignored a
number of complicating factors. We now briefly discuss these.

\paragraph{Standing variation} 
We have focused on relative rates of adaptation,
since in applications where adaptation has occurred,
the question is whether adaptations in distinct patches
have appeared independently or not.
However, 
any adaptation that does occur may have to make use of standing variation,
if mutation rates are low.
The case of a panmictic population was studied by \citet{softsweepsI},
and we study the case of a continuous, spatial population in \citet{ralph2014standing}.
If parallelism in local adaptation of the sort we study here is due to standing variation
rather than new mutation,
then the dynamics of adaptation should not depend strongly on migration patterns
(but the initial spatial distribution of standing variation may).

\paragraph{Dominance}
We have mostly ignored the issue of dominance 
by dealing with essentially haploid models,
and appealing to the fact that the dynamics we study occur where the mutation is rare,
and hence mostly present only in heterozygotes. 
Our results should hold as a good approximation to dominant and partially dominant alleles
(with $s_m$ the selection against heterozygotes).
If, however, the mutation is recessive, then it is essentially neutral where rare,
and so would encounter much less resistance to spreading between patches.
The shape of the cline obtained is given by \citet{haldane1948theory}.
This is counteracted, however, by the increased difficulty with which the mutation would establish
once arriving in a new patch,
if the beneficial effect is also recessive. \linelabel{rr:dominance}
As such it is not clear what our intuition should be 
about the contribution of recessive alleles to adaptation via migration. 
Further work is needed to put empirical observations of local adaptation by recessive alleles 
in a theoretical context.
It is similarly unclear how the model should extend to a polygenic trait.

\paragraph{Decay of shared haplotypes} \linelabel{rr:haplotype_discussion}
To provide context for the results on shared haplotype length in
section \secref{ss:haplotype_length}
it is important to also understand
the process by which haplotypes are whittled down within patches.
The initial haplotype that sweeps within a patch 
will be dispersed over time by recombination.
Likewise, the haplotype that is shared between patches coadapted by migration
will also break down
(equation~\ref{eqn:haplotype_length}).
However, a long time after the initial sweep,
we may still expect to find individuals within the patch sharing longer haplotypes around the selected locus
than with individuals elsewhere,
since selection against migrants 
decreases mean coalescence times within the patch near the selected locus.  %\citep{bartonpaper,charlesworths}. 
The literature on clines (e.g., \citet{barton1979geneflow}) has important information,
but more work is needed
to provide robust estimates for these processes.
Questions about the genomic length-scale of signals of
sweeps shared by migration have also been addressed in discrete
population settings \citep{slatkin1998hitchhiking,kim2011hitchhiking}, reviewed in \citet{barton2000genetic}. 
This work has shown that the length of the shared swept haplotype is often significantly
shorter than the sweep within each patch, resulting in a pattern of
shoulders of elevated $F_{ST}$ between adapted populations some distance away from
the shared selected allele. It would be of interest to see how similar
patterns can arise in a continuous population setting, as a way of
uniting these results.

\paragraph{Long distance migration}
We have also ignored the possibility of very long distance migration,
instead focusing on local dispersal (hence Gaussian by the central limit theorem).
However, dispersal distributions can be very heavy tailed, 
with a small fraction of individuals moving very long distances indeed \citep{levin2003ecology,reynolds2009levy}.
In addition, over long time-scales, very rare chance events (mice carried off by hurricanes and the like; \citet{censky1998overwater,nathan2008mechanisms})
could play a role in spreading migrant alleles if adaptation by other means is sufficiently unlikely.
Such tail events could greatly increase the probability of shared adaptation above that predicted by our model. 
Furthermore, if adaptive alleles do move between distant patches via rare, long distance migration 
then they will be associated with a much longer shared haplotype than predicted by equation \eqref{eqn:haplotype_length}. 
As such, we view our results as a null model by which the contribution of long distribution migrants to adaptation could be empirically judged.   

\paragraph{Other geographic models}
We have studied circular patches of habitat at long distances from each other.
Real habitat geometry can be much more complex,
e.g., with archipelagos of patches of varying sizes, 
or patches connected by long, skinny corridors, for instance.
The work of \citet{cantrell1991diffusive} comes closest to a general theory of balanced polymorphisms in such habitats.
It is possible that our techniques could be applied in their much more general setting,
as both are based, fundamentally, on branching process approximations.
It is also interesting to think about the probability of convergent
adaptation to continuously varying environments, e.g. replicated
environmental clines.

\paragraph{Concluding thoughts}
The falling cost of population genomic sequencing means that we will soon have the
opportunity to study the interplay of adaptation with geography and ecology 
across many populations within a species. 
Our work suggests that even quite geographically close populations 
may be forced to locally adapt by repeated, convergent, \textit{de novo} mutation 
when migration is geographically limited
and selective pressures are divergent.
Thus, systems where populations have been repeatedly subject to 
strong local selection pressures may offer the opportunity 
to study highly replicated convergent adaptation within a similar
genetic background \citep{stern2013genetic}.  
Such empirical work will also strongly inform our understanding of the
ability of gene flow to keep ecologically similar populations evolving
in concert \citep{sexton2013genetic}.
Our results suggest that adaptation to shared environments is
certainly no guarantee of a shared genetic basis to adaptation, 
suggesting that rapid adaptation to a shared environment could potentially drive
speciation if the alleles that spread in each population fail to work
well together \citep{Kondrashov:03}. 

% discussion of applications. 

% discuss mixing of types; 
% Graph: from simulation showing local proportions in a single patch showing initial fixation of a single type and later mixing converging toward more than one type.  (and eventual loss of one?)

%%%%%
\section*{Materials \& Methods}

%%%%%%%
\subsection{Simulation methods}
\label{ss:simulations}

First we briefly describe the simulations we used for illustration and validation
(the \texttt{R} code used is provided as a supplement).
We simulated forward-time dynamics of the number of alleles of each type in a rectangular grid (either one- or two-dimensional) of demes with fixed size $N$.
Each generation, each individual independently chose to reproduce or not with a probability $r$ depending on her type and location in the grid;
locally beneficial alleles were more likely to reproduce.
Each extant individual then either remained in the same location with probability $1-m$
or else migrated a random number of steps 
in a uniformly chosen cardinal direction; 
for most simulations $m=0.2$ and
the probability of migrating $k$ steps was proportional to $2^{-k}$ for $1\le k \le 5$.
In 2D, diagonal steps were also used.
This gave us values of $\sigma=0.95$ deme spacings in 1D
and $\sigma=0.74$ in 2D.
\linelabel{rr:sim_details}
Once migrants were distributed, each deme was uniformly resampled back down to $N$ individuals.
(Although we described the simulation in terms of individuals,
we kept track only of total numbers in an equivalent way.)

The base probability of reproduction in each generation in simulations for type $b$ alleles was $r=0.3$;
this was then multiplied by $1+s$ to get the probability of reproduction for type $B$,
where the value of $s$ is either $s_m$ or $s_p$ depending on the individual's location.
This determines the values of $s_m$ and $s_p$ reported in the figures, 
and do not depend on the basic rate of reproduction. \revpoint{2}{5}
However, to obtain values for $s_p$ and $s_m$ when comparing theory to simulation,
we computed the rate of intrinsic growth, 
i.e., the $s$ so that the numbers of $B$ alleles when rare would change by $e^{st}$ after $t$ generations
in the absence of migration.
(The resulting values are close to the first notion of $s$,
but give better agreement with theory,
which uses the second definition.)

To sample lineages, we first simulated the population dynamics forwards in time,
then sampled lineages back through time
by, in each generation,
moving each lineage to a new deme with probability proportional to the reverse migration probability
weighted by the number of $B$ alleles in that deme in the previous time step.
If more than one lineage was found in a deme with $n$ alleles of type $B$,
then each lineage picked a label uniformly from $1 \ldots n$,
and those picking the same label coalesced.
Since reproduction is Poisson, this correctly samples from the distribution of lineages given the population dynamics.

\subsection[Probability of Establishment]{Numerical calculation of the probability of establishment}
\label{apx:establishment_sims}

When rare, 
copies of a new mutant allele are approximately independent and experience a uniform selective benefit;
and can therefore be treated as a branching process.
Furthermore, whether or not a new, beneficial mutation establishes or is lost to demographic stochasticity
is determined by this initial phase where it is rare.
Fortunately, the probability that a branching process dies out
can be found as a fixed point of the generating function of the process \citep{jagers1975branching}.
Therefore, we calculated explicitly the generating function for a spatial branching process
with nearest-neighbor migration on a one-dimensional lattice
and a Poisson number of offspring with mean $1+s$,
where $s$ could vary by location,
and iterated this forward to convergence to obtain $1-p(x)$, the probability a single mutation appearing at $x$ would fail to establish.
We considered two situations:
where $s$ is a step function,
and where it has a linear transition.
These solutions are shown, and parameters described, in figure~\ref{fig:prob_estab_calcs}.

Here (and at other parameter choices) we see that the probability of establishment $p(x)$ goes to the equilibrium value
(approximately $p_e = 2s/\xi^2$) within the patch;
the transition is fairly symmetrical about the edge of the patch, even if the edge of the patch is not sharp.
Additional experimentation indicated that the fit remains equally good for other parameter values,
even if migration can move further than one deme and offspring numbers are not Poisson.
This lends credence to our approximation that the integral of $p(x)$ over the entire range
is close to $p_e$ multiplied by the area of the patch.

\begin{figure}[ht!]
    \begin{center}
        % made in patchy-selection-plots.R
        \includegraphics{prob-establishment}
    \end{center}
    \caption{The probability of establishment of a new mutant allele as a function of distance from the edge of the region where it is beneficial,
    in an abrupt transition (left) and a gradual transition (right).
    The allele is deleterious to the left and beneficial to the right (with selection coefficients $\mp 0.05$ respectively);
    and the selection coefficient interpolates linearly between these in the central hatched region.
    The number of offspring is Poisson.
    The underlying landscape has demes every 15km and a migration rate of 0.5 between neighboring demes;
    the probability was found by numerically solving the equation for the probability of establishment of a multi-type branching process.
    Note that the approximation used in the text is here $p_e \approx 2 s_p / \xi^2 = 0.1$,
    agreeing with the value in the center of the patch.
    \label{fig:prob_estab_calcs}
    }
\end{figure}



%%%%%%%%%%%%
\subsection[Equilibrium Frequency]{The equilibrium frequency}
\label{apx:eqfreq}

For completeness, and clarity as to the scalings on the relevant parameters,
here we provide a derivation of the differential equations referred to above equation~\eqref{eqn:eqfreq},
and establish the asymptotics given in that equation.
One route to the ``equilibrium frequency'' of the allele outside the range where it is advantageous is as follows;
see \citet{slatkin1973geneflow} or \citet{barton1987establishment}
(or \citet{KPP1937book} or \citet{fisher1937wave} or \citet{haldane1948theory})
for other arguments in equivalent models,
and see \citet{etheridge2000introduction} and/or \citet{dawson1993measurevalued} for a general framework for the stochastic processes below.

Suppose that the population is composed of a finite number of small demes of equal size $N$ arranged in a regular grid,
and that selection (for or against) the allele is given by the function $s(x)$, with $x$ denoting the spatial location.
Each individual at location $x$ reproduces at random, exponentially distributed intervals,
producing a random number of offspring with distribution given by $X$
who then all migrate to a new location chosen randomly from the distribution given by $x+R$,
where they replace randomly chosen individuals.
If $x+R$ is outside of the range, then they perish.
Each individual's time until reproduction is exponentially distributed:
the reproduction rate is 1 if it carries the original allele, or is $1+s(x)$ if it carries the mutant allele.
Suppose that the number of offspring $X$ has mean $\mu$; the variance of $X$ will not enter into the formula
(but assume $X$ is well-behaved).
Also suppose that the migration displacement $R$ has mean zero and variance $\sigma^2$;
in more than one dimension, we mean that the components of the dispersal distance are uncorrelated
and each have variance $\sigma^2$.

Let $\Phi^N_t(x)$ be the proportion of mutant alleles present at location $x$ at time $t$,
and $\Phi_t(x)$ the process obtained by taking $N \to \infty$ (which we assume exists).
Denote by $\delta_x$ a single unit at location $x$, so that e.g.~$\Phi^N_t + \delta_x/N$
is the configuration after a mutant allele has been added to location $x$.
For $0\le \phi \le 1$, we also denote by $\bar X_\phi$ the random number of mutant alleles added if $X$ new offspring carrying mutant alleles
replace randomly chosen individuals in a deme where the mutant allele is at frequency $\phi$ (i.e.~hypergeometric with parameters $(X,N\phi,N(1-\phi))$);
similarly, $\widetilde X_\phi$ is the number lost if the new offspring do not carry the allele (i.e.~hypergeometric with parameters $(X,N(1-\phi),N\phi)$).
(We like to think of $\Phi^N_t$ as a measure, but it does not hurt to think of $\Phi^N$ as a vector;
we aren't providing the rigorous justification here.)
Then the above description implies that for any sufficiently nice function $f(\Phi)$ that
\begin{align} \label{eqn:discrete_generator}
  \begin{split} \frac{\partial}{\partial t} \E\left[ f(\Phi^N_t) \right] 
  &= N \sum_x \left\{ \E\left[ (1+s(x+R)) \Phi^N_t(x+R) \left( f\left(\Phi^N_t + \frac{\bar X_{\Phi_t(x+R)}}{N}\delta_{x}\right) - f(\Phi^N_t) \right) \right] \right. \\
     & \qquad  \qquad \left. {} + \E\left[ \left(1-\Phi^N_t(x+R)\right) \left( f\left(\Phi^N_t - \frac{\widetilde X_{\Phi_t(x+R)}}{N}\delta_{x}\right) - f(\Phi^N_t) \right) \right] \right\}  \end{split} \\
     &= \mu \sum_x \E\left[ \left(\partial_{\phi(x)} f(\Phi_t) \right) \left\{ \Phi_t(x+R) - \Phi_t(x) + s(x+R) \Phi_t(x+R) (1-\Phi_t(x)) \right\} \right] + O\left(\frac{1}{N}\right).
\end{align}
In the final expectation, $R$ and $\Phi$ are independent.
This follows by taking first-order terms in $1/N$ in the Taylor series for $f$, 
and the fact that $\E[\bar X_\phi] = \phi \mu$ and $\E[\widetilde X_\phi] = (1-\phi)\mu$.
We can see two things from this:
First, since this is a first-order differential operator, the limiting stochastic process $\Phi$ obtained as $N \to \infty$
is in fact deterministic (check by applying to $f(\Phi) = \Phi(x)^2$ to find the variance).
Second, if we want to rescale space as well to get the usual differential equation, 
we need to choose $\var[R]=\sigma^2$ and $s(x)$ to be of the same, small, order; 
this is another way of seeing that $\sigma/\sqrt{s}$ is the relevant length scale (as noted by \citet{slatkin1973geneflow}).
More concretely, suppose that the grid size is $\epsilon \to 0$, 
that $\var[R] = (\sigma \epsilon)^2$, and that the strength of selection is $s(x) \epsilon$,
suppose that $\Phi_t(x)$ is deterministic and twice differentiable,
and let $\xi(t,x) = \Phi_{t/\epsilon}(x)$;
then the previous equation with $f(\Phi) = \Phi(x)$ converges to the familiar form:
\begin{align}
  \label{eqn:derived_fkpp}
  \partial_t \xi(t,x) = \mu \left( \frac{\sigma^2}{2} \sum_{k=1}^d \partial_{x_k}^2 \xi(t,x) + s(x) \xi(t,x) (1-\xi(t,x)) \right) .
\end{align}
Here we have taken first the population size $N \to \infty$ and then the grid size $\epsilon \to 0$;
we could alternatively take both limits together, but not if $\epsilon$ goes to zero too much faster than $N$ grows.
One reason for this is that at finite $N$,
the process $\Phi_t$ is an irreducible finite-state Markov chain with absorbing states at 0 and 1;
therefore, the inevitable outcome is extinction of one type or another,
which is not the regime we want to study.

In one dimension, we are done (and discuss exact solutions in the Supporting Information);
in higher dimensions, we are more interested in the mean frequency at a given distance $r$ from a patch.
If we take a radially symmetric patch centered at the origin (so $s$ only depends on $r$), 
and let $\xi(t,r)$ denote the mean occupation frequency at distance $r$,
then the polar form of the Laplacian in $d$ dimensions gives us that \eqref{eqn:derived_fkpp} is
\begin{align}
  \label{eqn:radial_fkpp}
  \partial_t \xi(t,r) = \mu \left( \frac{\sigma^2}{2} \partial_{r}^2 \xi(t,r) + \sigma^2\frac{d-1}{2r} \partial_r \xi(t,r) + s(r) \xi(t,r) (1-\xi(t,r)) \right) .
\end{align}


%%%%%%%%%%%
\subsection[Asymptotics]{Asymptotic solution for the equilibrium frequency} 
\label{apx:asymptotics}

A radially symmetric equilibrium frequency $\xi(t,x)=q(|x|)$,
with $s(r) = -s < 0$ for all $r>r_0$,
solves for $r_0 < r < \infty$,
\begin{align} \label{eqn:radial_eqfreq_diffeq}
    \partial_r^2 q(r) + \frac{d-1}{r} \partial_r q(r) - \frac{2s}{\sigma^2} q(r) (1-q(r)) = 0 \\
    \lim_{r \to \infty} q(r) = \lim_{r \to \infty} \partial_r q(r) = 0 \qquad 
    0 < q(r) < 1 
\end{align}
Since $q(r) \to 0$ as $r \to \infty$, so $q(r) (1-q(r)) \approx q(r)$,
it can be shown 
% theory in \citet{wasow1965asymptotic} implies 
that the true equilibrium frequency $q$ is close, for large $r$, to the solution to
\begin{align}
    \partial_r^2 u(r) + \frac{d-1}{r} \partial_r u(r) - \frac{2s}{\sigma^2} u(r) = 0  \label{eqn:bessel} \\
    \lim_{r \to \infty} u(r) = \lim_{r \to \infty} \partial_r u(r) = 0 \qquad  .
\end{align}
This has general solution given by a modified Bessel function:
using \citet{gradshteyn2007table} 8.494.9,
the general solution is
\begin{align}
    u(r) = C' (r-r_1)^{(2-d)/2} K_{(2-d)/2} \left( (r-r_1) \sqrt{2s}/\sigma \right) ,
\end{align}
where $C'$ and $r_1$ are chosen to match boundary conditions.
Asymptotics of Bessel functions (\citet{gradshteyn2007table}, 8.451.6) then imply that
\begin{align}
    q(r) \approx C r^{(1-d)/2} \exp \left( -r \sqrt{2s}/\sigma \right) + O(1/r),
\end{align}
where $C$ is a different constant.



%%%%%%%%% %%%%%%%%%%%%%%
\subsection[Hitting and Occupation]{Hitting and occupation}
\label{ss:hitting_occupation}

Here we make a more precise argument to back up expression \eqref{eqn:migrate} for the migration rate.
The argument made above in section \secref{ss:heuristics} applies to general migration mechanisms,
since it relies only on a decomposition of the migrant families upon hitting the new patch;
but it is also imprecise in subtle ways
that are difficult to formalize.
Here we take a somewhat different tack,
supposing that it suffices to model the spatial movement of a migrant family
by following only the motion of the ``trunk''
(i.e., the red line in figure \ref{fig:branching_decomp}),
and supposing this motion is Brownian, with variance $\sigma^2$ per generation.
(Recall $\sigma$ is the dispersal distance.)
We then use facts about Brownian motion and branching processes, 
to compute more precise versions of equations \eqref{eqn:gestalt_q} and \eqref{eqn:gestalt_migrate}.
\linelabel{rr:moved_argument}

We are approximating the dynamics of the focal allele in the region further away than $r_0$ from the patch
as the sum of independent migrant families, each of whose dynamics are
given by a spatial branching process (as depicted in Figure \ref{fig:branching_decomp}).
Call $B(r_0)$ the region closer than $r_0$ to the patch, and $\partial B(r_0)$ its boundary.
Denote by $\gamma(x)$ the mean rate of outflux of migrant families from a point $x \in \partial B(r_0)$,
i.e., the time-averaged density of individuals near a point $x$ in $\partial B(r_0)$ 
that are the founders of new migrant families.
Expressions \eqref{eqn:gestalt_q} and \eqref{eqn:gestalt_migrate} are a simple product of the ``outflux of families'',
when in fact they should be an integral of $\gamma(x)$ over possible locations. \linelabel{rr:awkward}
However,
it will turn out that the integrand of equation \eqref{eqn:gestalt_migrate} 
is well-approximated by a constant multiple of the integrand of \eqref{eqn:gestalt_q}.

First consider equation \eqref{eqn:gestalt_q} for the equilibrium frequency.
Suppose that $Z$ is one such spatial branching process as above in section \secref{ss:migrant_math}, 
started at time 0 with a single individual at $x$,
and write $S$ for the new patch.
The \emph{mean occupation measure} of $Z$ in the region $S$,
which we denote $u(x,S)$,
% and denote by $Z_t(A)$ is the number of descendant individuals alive at time $t$ living in the spatial region $A$.
% The \emph{mean occupation density} of $Z$ at location $y$
can be thought of informally \linelabel{rr:density}
as the expected total number of offspring of a family beginning at $x$ that ever live in $S$.
% It is defined to be the density of $\E[\int_0^t Z_t(\cdot) dt]$,
% i.e., the function $u(x,y)$ that, integrated over a set of locations $S$, 
% gives the expected total number of individuals ever living in $S$:
% \begin{align} \label{eqn:occupation_density_defn}
%     \E\left[ \int_0^\infty Z_t(S) dt \right] = \int_S u(x,y) dy .
% \end{align}
% % Said another way, $u(x,y)$ is the expected number of offspring of a progenitor at $x$ that will ever be observed near $y$.
% Since reproduction and spatial motion are independent,
% $u(x,y)$ is the sum across all generations of the expected number of individuals alive in that generation,
% multiplied by the probability density that spatial motion takes a lineage from $x$ to $y$,
% i.e.,
% \begin{align}
%     u(x,y) = \int_0^\infty e^{-s_m t} p_t(x,y) dt .
% \end{align}
Let $q(S) = \int_S q(y) dy$ denote the total equilibrium frequency in $S$.
This is decomposed in expression \eqref{eqn:gestalt_q} 
as the sum of mean occupation densities of a constant outflux of branching processes
from $\partial B(r_0)$:
\begin{align} \label{eqn:occupation_integral}
    \rho q(S) &= \int_{\partial B(r_0)} \gamma(x) u(x,S) dx  . 
\end{align}

Now we will decompose $u(x,S)$ under the assumption that 
the marginal distribution of the spatial motion of a single lineage is Brownian.
Let $B_t$ be a Brownian motion with variance $\sigma^2$, 
and $\tau_\dagger$ an independent Exponential($s_m$) time.
The mean occupation time of $Z$ spent in a region $S$ is,
\begin{align}
    u(x,S) &= \E\left[\int_0^\infty Z_t(S) dt \right] \\
           &= \int_0^\infty \E[Z_t] p_t(x,S) dt \\
        &= \int_0^\infty e^{-s_m t} p_t(x,S) dt \\
        &= \int_0^\infty \P^x\{ \tau_\dagger > t \; \& \; B_t \in S \} dt \\
        &= \E^x\left[ \int_0^{\tau_\dagger} \one_S(B_t) dt \right]  ,
\end{align}
where $\P^x$ gives probabilities for Brownian motion begun at $x$ (i.e., $B_0=x$),
and likewise $\E^x$.
If we define
$\tau_S$ to be the hitting time of $S$ by the Brownian motion $B$,
and $\mu_S(x)$ to be the hitting distribution of $\partial S$ by $B_{\tau_S}$ conditioned on $\tau_S < \tau_\dagger$,
by the strong Markov property this is equal to
\begin{align}
    u(x,S) % &= \E^x[ \int_0^{\tau_\dagger} \one_S(B_t) dt ] \\
           &= \P^x\{ \tau_S < \tau_\dagger \}  \E^{\mu_S(x)}\left[ \int_0^{\tau_\dagger} \one_S(B_t) dt \right] \\
           &= \P^x\{ \tau_S < \tau_\dagger \}  g(A)  ,
\end{align}
where now $\E^\mu$ denotes expectations for Brownian motion for which the distribution of $B_0$ is $\mu$, 
% provided $S$ is not pathological.
and $g(A)$ is defined to be the latter expectation,
which does not depend on $x$ if $S$ is circular (with area $A$).

This form we can now compare to 
the expression \eqref{eqn:gestalt_migrate} for the outflux of successful migrants.
Consider the probability that a migrant family beginning with a single individual at $x$ will ever establish in the new patch.
% The probability that this family reaches $A$ at time $t$ is the probability that it survives until $t$, $(1-k_e(t))$,
% multiplied by the probability that the family reaches the patch, given survival.
% If $t$ is large then as in the previous section, $(1-k_e(t)) \approx e^{-s_m t}/\E[K]$.
It would be possible to analyze this probability directly,
as in \citet{barton1987establishment};
but here we take a simpler route, 
approximating this by the chance that the trunk hits the new patch,
multiplied by the chance that at least one member of the family escapes demographic stochasticity
and successfully establishes in the new patch.
Write $h(x,S)$ for the probability that the Brownian trunk hits the patch $S$
before the family dies out,
and $f(S)$ for the chance that the family manages to establish in the new patch,
given that it successfully arrives.
(This is approximately independent of $x$.)
As for $q(S)$ above, expression \eqref{eqn:gestalt_migrate} is properly an integral against the outflux $\gamma(x)$:
\begin{align}\label{eqn:migrate_integral}
  \migrate(S) &= \int_{\partial B(r_0)} \gamma(x) h(x,S) f(S) dx .
\end{align}


If we make the approximation that $Z$ hits the new patch only if the trunk of $Z$ does,
and recall that $1-k_e(t)$ is the chance that $Z$ survives for $t$ generations,
then 
\begin{align}
    h(x,S) &\approx \int_0^\infty (1-k_e(t)) \P^x\{ \tau_S \in dt \} \\
           &= \int_0^\infty e^{s_m t} (1-k_e(t)) e^{- s_m t } \P^x\{ \tau_S \in dt \} \\
           &=\int_0^\infty e^{s_m t} (1-k_e(t)) \P^x\{ \tau_S \in dt \, \& \, t < \tau_\dagger \} \\
           &\approx \frac{1}{\E[K]} \int_0^\infty \P^x\{ \tau_S \in dt \, \& \, t < \tau_\dagger \} \\
    &= \frac{1}{\E[K]} \P^x[ \tau_S < \tau_\dagger ] .
\end{align}
Therefore, we have that
\begin{align} \label{eqn:boils_downto}
    h(x,S) \E[K] 
    \approx \P^x[ \tau_S < \tau_\dagger ] 
    = u(x,S) / g(A) .
\end{align}

Since the integrands in expressions \eqref{eqn:occupation_integral} and \eqref{eqn:migrate_integral} only differ by a factor
that (at least asymptotically) does not depend on the distance between $x$ and $S$,
we can obtain the migration rate by multiplying the equilibrium frequency by this factor.
This factor will not depend on $A$, 
because although $f(A)$ and $g(A)$ in principle depend on the patch size (and geometry),
the dependence is very weak.
For instance, the width of a circular patch only very weakly affects the chance 
of establishment of a new migrant that appears on its edge,
as long as the patch is large enough.

By equations~\eqref{eqn:occupation_integral} and \eqref{eqn:migrate_integral},
\begin{align}
    \rho q(S)  &= \int_{\partial B(r_0)} \gamma(x) u(x,S) dx  \\
          &\approx g(A) \E[K] \int_{\partial B(r_0)} \gamma(x) h(x,S) dx \\
          &= \frac{ g(A) \E[K] }{ f(A) } \migrate(S) . \label{eqn:q_migrate_relation}
\end{align}
Once we show that $g(A) \approx 1/(2 s_m)$, 
we will have arrived at the result, equation \eqref{eqn:ratio_K}.


The function $g(A)$ is the expected amount of time that a Brownian motion begun on the edge of a disk of area $A$
is expected to spend inside the disk before $\tau_\dagger$.
This is integral of the Green function for the Bessel process of the appropriate order, 
so using \citet{borodin2002handbook}, % (Appendix 1),
and letting $w$ be the width of the patch,
in $d=1$,
\begin{align} \label{eqn:gA_1D}
  \begin{split}
    g(A) % &= \E^w[ \int_0^\tau \one_A(B_t) dt ] 
    % &= \int_0^{2w} G_{s_m}(0,y) dy \\
      &= \int_0^{2w/\sigma} \frac{ e^{- y \sqrt{2s_m}/\sigma }}{\sqrt{2s_m}} dy \\
      &= (1-e^{-2w\sqrt{2s_m}/\sigma})/(2s_m)
  \end{split}
\end{align}
and in $d=2$, by \citet{gradshteyn2007table} 5.56.2,
\begin{align}
    g(A) % &= \E^w[ \int_0^\tau \one_A(B_t) dt ] 
    % &= \int_0^{2w} G_{s_m}(0,y) dy \\
  &= \int_0^{2w/\sigma} 2 y K_0(y \sqrt{2s_m}) dy \\
  &= \frac{1}{s_m}\int_0^{2w\sqrt{2s_m}/\sigma} y K_0(y) dy \\
  &= \frac{1}{2s_m}\left(1- \frac{2w\sqrt{2s_m}}{\sigma} K_1(2w\sqrt{2s_m}/\sigma) \right)  ,
\end{align}
where $K_0$ and $K_1$ are modified Bessel functions of the second kind.
In either case, $g(A) \approx 1/(2s_m)$, which is the approximation we use in the main text.


%%%%%%% %%%%%%%%%%%%
\subsection[The Integral $q(S)$]{The integral $q(S)$}
\label{apx:qS}

In the development above
we need to approximate the integral of equation \eqref{eqn:eqfreq}
across the area occupied by the new patch,
$q(S) = \int_S q(x) dx$.
The precise answer depends on the shape and orientation of the patches;
but we aim for a usable approximation,
primarily in terms of the shortest distance from the old patch to the new patch, denoted $R$,
and assuming $R$ is large enough we can take equation \eqref{eqn:eqfreq} as an equality.

Since $q(R)$ decreases as $|x|$ increases,
\begin{align}
    q(S) \le A q(R) % C \left( \frac{R\sqrt{2s_m}}{\sigma} \right)^{(d-1)/2} e^{-R\sqrt{2s_m}/\sigma} ,
\end{align}
where $A$ is the area of $S$.
This will be a good approximation if $S$ is small.

In one dimension, if $S$ has length $\ell$,
\begin{align}
    q(S) &= \int_R^{R+\ell} C e^{-u\sqrt{2s_m}/\sigma} du \\
         &= C e^{-R\sqrt{2s_m}/\sigma} \frac{\sigma}{\sqrt{2s_m}} \left( 1 - e^{-\ell\sqrt{2s_m}/\sigma} \right) \\
         &\le C e^{-R\sqrt{2s_m}/\sigma} \frac{\sigma}{\sqrt{2s_m}} .
\end{align}

In two dimensions, 
suppose that $T$ is a rectangle enclosing $S$ with one axis aligned towards the original patch and transverse width $w$.
Then
\begin{align}
    q(S) &\le \int_T q(x) dx \\
        &\le w \int_R^\infty q(r) r dr \\
        &= w C \int_R^\infty \left( \frac{r\sqrt{2s_m}}{\sigma} \right)^{-1/2} e^{-r\sqrt{2s_m}/\sigma} dr \\
        &\le w \frac{\sigma}{\sqrt{2s_m}} C \sqrt{\pi} \left( \frac{R\sqrt{2s_m}}{\sigma} \right)^{-1/2} e^{-R\sqrt{2s_m}/\sigma} dr .
\end{align}
If we absorb $\sqrt{\pi}$ into the constant $C$,
this is of the form $(\text{width}) \times (\sigma/\sqrt{2s_m}) \times q(R)$,
i.e., roughly the area, after replacing the length by $\sigma/\sqrt{2s_m}$.

In both cases,
the upper bound is $q(S) \le A' q(x)$,
where $A'$ is the area of the parts of $S$ that are no more than $R+\sigma/\sqrt{2s_m}$ away from the original patch.
Lower bounds could be obtained along similar lines.


% %%%%%%% %%%%%%%%%%%%
% \subsection[Outflux of Families]{The outflux of families}
% \label{apx:outflux}
% 
% It is of interest to have an expression for the ``outflux of families'',
% $\gamma(r_0)$, 
% defined to be the mean density of $B$ alleles at distance $r_0$
% whose lineage traces back to the original patch without venturing further away than $r_0$.
% By expression \eqref{eqn:occupation_integral},
% taking $S$ to be a small area located at $y$,
% \[
%     \gamma(r_0) = \frac{\rho q(y)}{\int_{B(r_0)} u(x,y) dx}
% \]
% for any $y$.  
% Take $\hat x = (1,0)$ and $y=r \hat x$.  Then
% \begin{align}
%   \frac{ q(y) }{ u(x,y) } &\simeq C' \frac{ \|y\|^{-(d-1)/2} e^{-\|y\| \sqrt{2s_m}/\sigma } }{ \|y-x\|^{-(d-1)/2} e^{-\|y-x\| \sqrt{2s_m}/\sigma } } \\
%   &= C' \| \hat x - x/r \|^{(d-1)/2} e^{ r \sqrt{2s_m}/\sigma  (\|\hat x - x/r\| - 1)} .
% \end{align}
% Now let $x = (r_0 \cos \theta, r_0 \sin \theta)$,
% so that the expression in the exponent is to, first order in $1/r$,
% \begin{align}
%     r (\|\hat x - x/r\| - 1) &= r \left( \sqrt{ \left( 1 - \frac{r_0}{r} \cos \theta \right)^2 + \frac{r_0^2}{r^2} \sin^2 \theta } - 1 \right ) \\
%             &= - r_0 \cos \theta + O(1/r) .
% \end{align}
% This implies that, still with $r$, $r_0$, and $\theta$ defined as above,
% \begin{align}
% \frac{q(y)}{u(x,y)} \longrightarrow C' e^{-r_0 \frac{\sqrt{2s_m}}{\sigma} \cos \theta} \qquad \text{ as } r \to \infty .
% \end{align}
% In one dimension we are done (since $\cos \theta=1$).
% In two dimensions,
% by bounded convergence and \citet{gradshteyn2007table} (3.339),
% \begin{align}
%   \int_{B(r_0)} \frac{q(y)}{u(x,y)} dx &\to \gamma(r_0) \quad \text{as } y \to \infty\\
%   &= C' \int_0^{2 \pi} e^{-r_0 \frac{\sqrt{2s_m}}{\sigma} \cos \theta} d\theta \\
%   &= 2 \pi C' I_0(-r_0\sqrt{2 s_m}/\sigma),
% \end{align}
% where $I_0$ is a modified Bessel function of order zero of the first kind.


%%%%%%%%%%%


\section*{Acknowledgements}
We would like to heartily thank Joachim Hermisson, Richard Neher, and an anonymous reviewer
for a number of insightful suggestions,
and catching several errors present in earlier drafts. \revpoint{3}{4}
We would also like to acknowledge Michael Nachman, Steve Evans, and Yaniv Brandvain for useful discussions, 
as well as Gideon Bradburd, and Simon Aeschbacher and the rest of the Coop lab
for helpful comments and discussion of the manuscript.
\ifsubmissionversion
% PLoS puts this somewhere else
\else
This work was supported by grants from the
National Science Foundation under Grant No.\ 1262645 to PLR and GC and by the
National Institute of General Medical Sciences of the National
Institutes of Health under award numbers NIH RO1GM83098 and
RO1GM107374 to GC.
\fi

\bibliography{standing_patches_refs}


\ifsubmissionversion
\clearpage
\renewcommand*\listfigurename{Figure Legends}
\listoffigures
\fi

\clearpage

% endfloat uses different names
\renewcommand{\thesfigure}{S\arabic{sfigure}}
\renewcommand{\thetable}{S\arabic{table}}
\ifsubmissionversion
  \renewcommand{\thepostsfigure}{S\arabic{postsfigure}}
  % \renewcommand{\thefigure}{S\arabic{figure}}
  % \renewcommand{\thepostfigure}{S\arabic{postfig}}
  % \setcounter{postfig}{0}
  \renewcommand{\theposttable}{S\arabic{posttable}}
  \setcounter{posttable}{0}
\fi

%% list of supplementary figures
\ifsubmissionversion
  \renewcommand*\listsfigurename{Supplementary Figure Legends}
  \listofsfigures
  \renewcommand*\listtablename{Supplementary Table Legends}
  \listoftables
  \pagebreak
\fi

\appendix
\section*{Supporting Information}

\setcounter{secnumdepth}{2}

%%%%%%%
\subsection*{The exact solution to the equilibrium frequency}
\label{apx:elliptic_integrals}


Here we describe how to solve \eqref{eqn:radial_eqfreq_diffeq} exactly in one dimension,
i.e., the Fisher-KPP equation with piecewise constant selection.
Let $p(t,x)$ be the proportion of the allele under spatially varying selection $s(x)$ in one dimension,
so that 
\[
\partial_t p(t,x) = \frac{\mu}{2} \sigma^2 \partial_x^2 p(t,x) + \mu s(x) p(t,x) (1-p(t,x)) .
\]
The stable distribution $\phi(x) = \lim_{t\to\infty} p(t,x)$ then solves
\begin{align} \label{eqn:definingphi}
    \partial_x^2 \phi(x) = - 2 s(x) \phi(x) (1-\phi(x)) /\sigma^2,
\end{align}
with appropriate boundary conditions.
First rescale space by $\sigma/\sqrt{2}$ so the $\sigma^2/2$ term disappears.
If we now assume that $s(x)$ is piecewise constant,
$s(x) = s_i$ for $x \in [x_i,x_{i+1})$, with $x_0=-\infty$ and $x_{n+2}=\infty$,
then the equation is integrable: if we multiply through by $2\partial_x \phi(x)$ and integrate, then we get that
\begin{align} \label{eqn:conservation}
    ( \partial_x \phi(x) )^2  &= - \int^{x} 2 s(x) \phi(x) (1-\phi(x)) \partial_x \phi(x) dx \\
        &= - s_i \phi(x)^2 \left( 1 - \frac{2}{3} \phi(x) \right) - K_i \quad \mbox{for } x \in [x_i,x_{i+1}) ,
\end{align}
if we define
\begin{align} 
  K_i = - s_i \phi(x_i)^2 \left( 1 - \frac{2}{3} \phi(x_i) \right) - ( \partial_x \phi(x_i) )^2  .
\end{align}
For ease of reference, we define
\[
        V_i(\phi) = s_i \phi^2 \left( 1 - \frac{2}{3} \phi \right) + K_i .
\]
Note that $V_i'(0)=V_i'(1)=0$, that $V_i(0)=K_i$ and $V_i(1) = K_i+s_i/3$.  
We will always have that $V(\phi) \le 0$.
(We have then that $\partial_x^2 \phi = - \partial_\phi V(\phi)$, the equation of motion of a particle in potential $V$.
Also note that this implies ``conservation of energy'', i.e., $( \partial_x \phi(x) )^2 + V(\phi(x))$ is constant.)
Rearranging, we get that $dx = d\phi / \sqrt{-V(\phi)}$, so where $\phi(x)$ is monotone, the inverse is
\[
    x(\phi) = x(\phi^*) \pm \int_{\phi^*}^\phi \frac{ d\psi }{ \sqrt{ -V(\psi) } } .
\]
For each $i$ then define the elliptic function
\begin{align}  \label{eqn:elliptic_function}
    F_i(\phi) = \int_{\phi_i^*}^\phi \frac{ d\psi }{ \sqrt{ -V_i(\psi) } } ,
\end{align}
where take the positive branch of the square root, and $\phi_i^*$ will be chosen later.
Then we have that
$x(\phi) - x(\phi_0) = \pm( F(\phi) - F(\phi_0))$,
or for an appropriate $x_0$,
\[
    \phi(x) = F^{-1}\left( F(\phi(x_0)) \pm (x - x_0) \right).
\]

We clearly want $\lim_{x \to \infty} \partial_x \phi(\pm x) = 0$, 
and $\lim_{x \to \infty} \phi(\pm x)$ to be zero or one depending on the sign of $s_0$ and $s_{n+1}$.
Since $(\partial_x \phi(x))^2 = -V(x)$, this implies that if $s_0<0$, then $K_0 = 0$,
while if $s_0>0$ then $K_0 = s_0/6$; and likewise for $K_{n+1}$.

We also require that $\phi(x)$ and $\phi'(x)$ are continuous.
Continuity of $\phi'(x)$ is equivalent to $V_i(x_{i+1}) = V_{i+1}(x_{i+1})$,
which we can rearrange to find an equation for $K_{i+1}$ in terms of $K_i$ and $\phi(x_{i+1})$:
\[
    K_{i+1} - K_i = (s_i - s_{i+1}) \phi(x_{i+1})^2 (1-2\phi(x_{i+1})/3) .
\]
What about the frequency at the points the selection changes, $\phi(x_i)$?  
Well, if $\phi(x)$ is monotone on $[x_i,x_{i+1})$ then we can without loss of generality take $\phi_i^*=0$ or $1$ depending on the sign of $s_i$.
Otherwise, let $\phi_i^*$ be the (unique) root of $V_i$ in $[x_i,x_{i+1})$, so $V_i(\phi_i^*)=0$.
In this case, $\phi_i^*$ is the maximum or minimum of $\phi$ in the interval:
if $s>0$, then $\phi_i = \max\{ \phi(x) : x \in [x_i,x_{i+1})\}$. 
Recall we defined $F_i$ using $\phi_i^*$; now using the fact that $\phi$ is monotone with the opposite sign on either side of $\phi_i^*$,
$x_{i+1} - x_i = F_i(\phi(x_{i+1})) - F_i(\phi_i^*) + F_i(\phi(x_{i})) - F_i(\phi_i^*)$,
and that $F(\phi_i^*) = 0$,
we know that the length of the $i$th stretch is
\[
    x_{i+1} - x_i = \pm \left( F_i(\phi(x_i)) + F_i(\phi(x_{i+1})) \right).
\]
Note that all the $\pm{}$'s are easily relatable to the signs of $s_i$.
If we knew $\phi(x_1)$ and $\phi'(x_1)$, then we'd be able to solve the equations for $\phi(x_i)$ and $K_i$ recursively upwards.
In some cases, such as \citet{slatkin1973geneflow}, we can infer $\phi(0)$ and $\partial_x \phi(0)$ by spatial symmetry.
In other cases, we are only given $\phi(-\infty)$ and $\phi(\infty)$, and have to work inwards from the ends.


\subsection*{Doing the integrals}
\label{apx:some_integrals}

An important ingredient in the above method is the integral \eqref{eqn:elliptic_function},
to which a method for solving the Korteweg-de Vries equation can be applied \citep{NEQwiki}.
Recall that $V_i(\phi) =  s_i \phi^2(1-2\phi/3) + K_i$, with $K_i = V_i(0)$ chosen to match $V$ at the boundaries;
since we're just working within an interval with $s$ constant, we can rescale space by $\sqrt{2 s_i}/\sigma$,
so that $s$ and $\sigma$ drop from the equation.
We then want to integrate
\[
    \int \frac{ d\phi }{ \sqrt{-V(\phi)} } = 
         \int \frac{ d\phi }{ \sqrt{ \phi^2 (1-2\phi/3) + K } } ,
\]
over a domain where $V$ is always negative.
Let $\phi^2(1-2\phi/3)+K = (\alpha-\phi)(\phi-\beta)(\phi-\gamma)$,
and change variables first to $y^2=(\alpha-\phi)$, 
and then to $x = y/\sqrt{\alpha-\beta}$, so that
\begin{align*}
    \frac{ d\phi }{ \sqrt{ \phi^2 (1-2\phi/3) + K } } 
        = \frac{ - 2 dy }{ \sqrt{ (\alpha-\beta-y^2) (\alpha-\gamma-y^2) } } \\
        = \frac{ - 2 dx }{ \sqrt{\alpha-\gamma} \sqrt{ (1-x^2) (1-S^2 x^2) } } ,
\end{align*}
with $S^2 = (\alpha-\beta)/(\alpha-\gamma)$.
Now Jacobi's incomplete elliptic integral of the first kind is defined by
\[
    F(x;k) = \int_0^x \frac{dt}{\sqrt{ (1-t^2)(1-k^2t^2) }} ,
\]
and the Jacobian elliptic function $\sn(x;k)$ is the inverse: $F(\sn(x;k);k) = x$.
As $k \to 0$, $\sn(x;k) \to \sin(x)$, while as $k \to 1$, $\sn(x;k) \to \sinh(x)$.

% Presumably, this gets the same answer as \citep{slatkin1973geneflow}, who finds that in the solvable cases,
% for $x$ in the last interval (which ends in $\infty$), that
% \begin{align*}
%     \phi(x) &= \frac{3}{2}\left( 1 - \tanh^2 ( x \sqrt{2s} / 2 \sigma + C ) \right) \\
%         &\simeq C' \exp\left( - \frac{ x \sqrt{2s} }{ \sigma } \right) .
% \end{align*}



%%%%%%%
\subsection*{More information about the simulations}
\label{apx:parameter_tables}

Here we provide additional details regarding the simulations we used
to validate the theory in figures \ref{fig:sim_times} and \ref{fig:sim_probs}.

%%%
% Tables

% table of parameters, produced by analyze-sims.R
\input{mutation-parameters-table.tex}

% table of parameters, produced by analyze-sims.R
\input{migration-parameters-table.tex}

%%%
% Supplementary Figures

\begin{sfigure}
  \begin{center}
    \includegraphics{mutation-times-predicted}
  \end{center}
  \caption{
    The same data shown in the left panel of figure~\ref{fig:sim_times},
    but all times shown (not just the interquartile ranges),
    and including those parameter values at which most of the simulations did not adapt by 25,000 generations.
  } \label{sfig:sim_migration_times}
\end{sfigure}

\begin{sfigure}
  \begin{center}
    \includegraphics{migration-time-predicted}
  \end{center}
  \caption{
    The same data shown in the right panel of figure~\ref{fig:sim_times},
    but all times shown (not just the interquartile ranges),
    and including those parameter values at which most of the simulations did not adapt by 25,000 generations.
    The upper panel has the predicted time to adaptation on the horizontal axis as in figure~\ref{fig:sim_times},
    and the lower panel has, for comparison, the raw distance between patches 
    (which predicts time to adaptation, but not as well).
  } \label{sfig:sim_mutation_times}
\end{sfigure}

% example trace plots

% sm=0.1
\begin{sfigure}
  \begin{center}
    \includegraphics{example-mutation-sims/18885-r1-501-sb0_01-sm-0_1-N1200-pophistory-run}
    \includegraphics{example-mutation-sims/56325-r1-501-sb0_01-sm-0_1-N600-pophistory-run}
    \includegraphics{example-mutation-sims/28432-r1-501-sb0_01-sm-0_1-N50-pophistory-run}
  \end{center}
  \caption{
    Randomly chosen simulations of adaptation by new mutation
    with $s_m=0.1$, $\sigma\approx 1$, and $\rho$ varying.
    On the left of each is a space-time heatmap of the local frequency of $B$ alleles;
    and on the right are twenty-five curves showing the frequencies of $B$ at evenly spaced time points
    (i.e., each line represents a vertical slice through the plot on the left);
    dotted black lines indicate the patches where $B$ is advantageous.
  } \label{sfig:sims_1}
\end{sfigure}

% sm=0.01
\begin{sfigure}
  \begin{center}
    \includegraphics{example-mutation-sims/43099-r1-501-sb0_01-sm-0_01-N1200-pophistory-run}
    \includegraphics{example-mutation-sims/69787-r1-501-sb0_01-sm-0_01-N600-pophistory-run}
    \includegraphics{example-mutation-sims/11821-r1-501-sb0_01-sm-0_01-N50-pophistory-run}
  \end{center}
  \caption{
    Randomly chosen simulations of adaptation by new mutation
    with $s_m=0.01$, $\sigma\approx 1$, and $\rho$ varying.
    On the left of each is a space-time heatmap of the local frequency of $B$ alleles;
    and on the right are twenty-five curves showing the frequencies of $B$ at evenly spaced time points
    (i.e., each line represents a vertical slice through the plot on the left);
    dotted black lines indicate the patches where $B$ is advantageous.
  } \label{sfig:sims_2}
\end{sfigure}

% sm=0.001
\begin{sfigure}
  \begin{center}
    \includegraphics{example-mutation-sims/59611-r1-501-sb0_01-sm-0_001-N1200-pophistory-run}
    \includegraphics{example-mutation-sims/93713-r1-501-sb0_01-sm-0_001-N600-pophistory-run}
    \includegraphics{example-mutation-sims/29850-r1-501-sb0_01-sm-0_001-N50-pophistory-run}
  \end{center}
  \caption{
    Randomly chosen simulations of adaptation by new mutation
    with $s_m=0.001$, $\sigma\approx 1$, and $\rho$ varying.
    On the left of each is a space-time heatmap of the local frequency of $B$ alleles;
    and on the right are twenty-five curves showing the frequencies of $B$ at evenly spaced time points
    (i.e., each line represents a vertical slice through the plot on the left);
    dotted black lines indicate the patches where $B$ is advantageous.
  } \label{sfig:sims_3}
\end{sfigure}

% sm=0.0001
\begin{sfigure}
  \begin{center}
    \includegraphics{example-mutation-sims/15449-r1-501-sb0_01-sm-1e-04-N1600-pophistory-run}
    \includegraphics{example-mutation-sims/5582-r1-501-sb0_01-sm-1e-04-N400-pophistory-run}
    \includegraphics{example-mutation-sims/24639-r1-501-sb0_01-sm-1e-04-N25-pophistory-run}
  \end{center}
  \caption{
    Randomly chosen simulations of adaptation by new mutation
    with $s_m=0.001$, $\sigma\approx 1$, and $\rho$ varying.
    On the left of each is a space-time heatmap of the local frequency of $B$ alleles;
    and on the right are twenty-five curves showing the frequencies of $B$ at evenly spaced time points
    (i.e., each line represents a vertical slice through the plot on the left);
    dotted black lines indicate the patches where $B$ is advantageous.
  } \label{sfig:sims_4}
\end{sfigure}


% sm=0.1, R=80
\begin{sfigure}
  \begin{center}
    \includegraphics{example-migration-sims/89826-r1-501-sb0_01-sm-0_1-N4000-pophistory-run}
    \includegraphics{example-migration-sims/76178-r1-501-sb0_01-sm-0_1-N1000-pophistory-run}
    \includegraphics{example-migration-sims/95095-r1-501-sb0_01-sm-0_1-N100-pophistory-run}
  \end{center}
  \caption{
    Randomly chosen simulations of adaptation by migration
    with $s_m=0.1$, $R=80$, $\sigma\approx 1$, and $\rho$ varying.
    On the left of each is a space-time heatmap of the local frequency of $B$ alleles;
    and on the right are twenty-five curves showing the frequencies of $B$ at evenly spaced time points
    (i.e., each line represents a vertical slice through the plot on the left);
    dotted black lines indicate the patches where $B$ is advantageous.
  } \label{sfig:sims_5}
\end{sfigure}

% sm=0.01, R=80
\begin{sfigure}
  \begin{center}
    \includegraphics{example-migration-sims/42080-r1-501-sb0_01-sm-0_01-N4000-pophistory-run}
    \includegraphics{example-migration-sims/82698-r1-501-sb0_01-sm-0_01-N1000-pophistory-run}
    \includegraphics{example-migration-sims/97545-r1-501-sb0_01-sm-0_01-N100-pophistory-run}
  \end{center}
  \caption{
    Randomly chosen simulations of adaptation by migration
    with $s_m=0.01$, $R=80$, $\sigma\approx 1$, and $\rho$ varying.
    On the left of each is a space-time heatmap of the local frequency of $B$ alleles;
    and on the right are twenty-five curves showing the frequencies of $B$ at evenly spaced time points
    (i.e., each line represents a vertical slice through the plot on the left);
    dotted black lines indicate the patches where $B$ is advantageous.
  } \label{sfig:sims_6}
\end{sfigure}

% sm=0.001, R=80
\begin{sfigure}
  \begin{center}
    \includegraphics{example-migration-sims/88079-r1-501-sb0_01-sm-0_001-N4000-pophistory-run}
    \includegraphics{example-migration-sims/3464-r1-501-sb0_01-sm-0_001-N1000-pophistory-run}
    \includegraphics{example-migration-sims/37774-r1-501-sb0_01-sm-0_001-N100-pophistory-run}
  \end{center}
  \caption{
    Randomly chosen simulations of adaptation by migration
    with $s_m=0.001$, $R=80$, $\sigma\approx 1$, and $\rho$ varying.
    On the left of each is a space-time heatmap of the local frequency of $B$ alleles;
    and on the right are twenty-five curves showing the frequencies of $B$ at evenly spaced time points
    (i.e., each line represents a vertical slice through the plot on the left);
    dotted black lines indicate the patches where $B$ is advantageous.
  } \label{sfig:sims_7}
\end{sfigure}

% sm=0.0001, R=80
\begin{sfigure}
  \begin{center}
    \includegraphics{example-migration-sims/26044-r1-501-sb0_01-sm-1e-04-N4000-pophistory-run}
    \includegraphics{example-migration-sims/85750-r1-501-sb0_01-sm-1e-04-N1000-pophistory-run}
    \includegraphics{example-migration-sims/4111-r1-501-sb0_01-sm-1e-04-N100-pophistory-run}
  \end{center}
  \caption{
    Randomly chosen simulations of adaptation by migration
    with $s_m=0.0001$, $R=80$, $\sigma\approx 1$, and $\rho$ varying.
    On the left of each is a space-time heatmap of the local frequency of $B$ alleles;
    and on the right are twenty-five curves showing the frequencies of $B$ at evenly spaced time points
    (i.e., each line represents a vertical slice through the plot on the left);
    dotted black lines indicate the patches where $B$ is advantageous.
  } \label{sfig:sims_8}
\end{sfigure}






%%%% Reviews
\includereviews


\end{document} 

%%%%%%%% END %%%%%%%%%%%%%%%
%%%%%%%% %%% %%%%%%%%%%%%%%%


\begin{figure}[ht!]
    \begin{center}
        % made in patchy-sim-writeup-plots.R
        % \includegraphics{example-equilibrium}
    \end{center}
    \label{fig:example_equil}
    \caption{
    Snapshots of a simulation of allelic frequency dynamics around a patch where the allele is beneficial:
    the focal allele is benefical in the middle region and deleterious outside.
    The dynamics were simulated for 10000 generations across demes with 1000 individuals in each
    and random dispersal to nearby demes with a mean dispersal distance of $\sigma \approx 2$ demes.
    Colored lines show allele frequencies across space in 50 uniformly spaced generations;
    the solid line shows the mean across all generations.
    Below is the same plot, except on a log scale.
    }
\end{figure}

\section*{Temporary work: hitting probs}

Now consider the probability that a migrant family from $x$ will ever establish in the new patch,
which we assume has relatively small area $A$ and is centered at position $y$.
The probability that this family reaches $A$ by time $t$ is $(1-k_e(t))$
multiplied by the probability that the family reaches the patch, given survival.
If $t$ is large then the previous section argues that $(1-k_e(t)) \approx e^{-s_m t}/\E[K]$,
and the probability of reaching the patch by $t$ approximately the chance that a Brownian path
will have hit the patch by then.
If $w = A^{1/d}$ is the width of the patch, then 
this is the chance that a $d$-dimensional Bessel process begun at $\|y-x\|$ hits $w$ by time $t$.
Let $\tau_w$ be the hitting time of $w$,
so, integrating by parts, and using Borodin \& Salminen (2.2.0.1 and 4.2.0.1)
\begin{align}
    \int_0^\infty (1-k_e(t)) \P^x\{ \tau_w \le t \} dt  
    &\approx \int_0^\infty e^{-s_m t} \P^x\{ \tau_w \le t \} dt  / \E[K] \\
    &= \frac{1}{s_m \E[K]} \int_0^\infty e^{-s_m t} \P^x\{ \tau_w \in dt \} \\
    &= \frac{1}{s_m \E[K]} \E^x[e^{-s_m \tau_w}] \\
    &= 
    \frac{1}{s_m \E[K]} \times
    \begin{cases}
        e^{-(x-w) \sqrt{2s_m}} \qquad & d=1 \\
        \frac{ K_0(x\sqrt{2s_m}) }{ K_0(w\sqrt{2s_m}) } \qquad & d=2 
    \end{cases}
\end{align}
Note that the $d=2$ case
\begin{align}
    \frac{ K_0(x\sqrt{2s_m}) }{ K_0(w\sqrt{2s_m}) } = \sqrt{\frac{w}{x}} e^{-(x-w)\sqrt{2s_m}} ( 1 + O(1/w) ).
\end{align}



\section{Patchy selection} 
\label{ss:discretedemes}

We have been working in a continuous model of geographic space; 
in this section we move to a discrete model of isolated populations exchanging rare migrants,
but in section \ref{ss:patchyspace} discuss how a model with {\em patchy} selection can be approximated by such a discrete model.

Suppose now that we have a set of $L$ discrete populations of individuals that may reproduce or migrate,
and that migration and mutation are rare, so that the waiting times until either event is approximately exponential.
Suppose that for each pair of populations, labeled $i$ and $j$, the mean number of migrants that travel from $i$ to $j$ per generation
is $M_{ij}$ and that the mean number of new mutations appearing in population $i$ per generation is $\mu_i$.

{\tt Include cartoon here.}

If we suppose that fixation occurs on a faster time scale than mutation or migration,
so that it is very unlikely that two different migrants or mutants begin to fix locally in the same population,
then we can treat the process of extinction or local fixation as instantaneous 
(a limit previous studied by \cite{Slatkin:81}).
In this limit we may also ignore swamping effects of inmigration of nonmutant alleles.

In this case, we have the following picture.
The population at first starts out with no selected alleles. 
At some later time point suppose that there are $T$ different alleles present in the population, 
that the $k^\mathrm{th}$ allele is occupying demes $\{i^k_1, \ldots, i^k_{n_k}\}$, for $1\le k \le T$.
Let $I$ denote the set of all adapted demes, 
and let $J = \{j_1, \ldots, j_\ell\}$ denote the demes that have not yet adapted, with $\ell = L - \sum_k n_k$.
Then we are assuming that two types of event can happen which could change the state:
either at rate $\mu_j p_f$, an unadapted deme $j$ produces a new mutant, which fixes in this deme;
or at rate $M_{ij} p_f$, an adapted deme $i$ sends a migrant to unadapted deme $j$, which fixes in that deme.
Therefore, under these assumptions, the probability of fixation $p_f$ (and hence the selection coefficient)
only enter as a time scaling: for instance,
if we define $R = \sum_{j \in J} \left( \mu_j + \sum_{i \in I} M_{ij} \right)$,
then the total rate of such events is $p_f R$.
The probability that the next event results in, say, adaptation of deme $j$
is 
\[
 \left( \mu_j + \sum_{i \in I} M_{ij} \right) / R,
\]
and the probability that deme $j$ adapts through migration from deme $i$, given that deme $j$ is the next to adapt, is
$M_ij / R$,
while the probability that it adapts through a new mutation is $\mu_j / R$.

The main point here is that the final pattern,
and hence the likely extent of parallel adaptation,
is independent of the strength of selection.
This is in contrast to the continuous case, however, this observation is dependent on the assumption of weak migration
and fails to hold if different migrant lineages interact.
This can be viewed as a continuous-time Markov process XXX.
Explicit solutions are not available in the general case,
but if we specialize to a completely symmetric model (i.e., the ``island'' model),
then the mathematical picture is a pretty one.

\subsection{Symmetric patches}

Indeed, suppose that migration and mutation are symmetric: $M_{ij}=m$ and $\mu_j = \mu$ for all $j$ and $i\neq j$.
Suppose at some time the $k^\mathrm{th}$ allele is occupying $n_k$ demes, 
with $n = \sum_k n_k < L$.
Then at rate $(L-n) N \mu p_f + n (L-n) N m p_f$,
either a migration or mutation event happens.
With probability $\mu/(\mu + n m)$, it is a mutation,
and a new deme is randomly chosen and assigned a new allele.
With probability $n_k m / (\mu + n m)$,
the $k^\mathrm{th}$ type sends a successful migrant to a randomly chosen new island,
which is assigned the $k^\mathrm{th}$ allele.

This process is a continuous-time version of the ``Chinese restaurant process''
described in \citet{aldous1985exchangeability} and \citet{pitman1995partitions},
and so the final partition of types across demes has the Ewens sampling distribution with parameter $\mu/m$.
Again note that the selection coefficient, which is implicit in the probability $p_f$,
does not enter into the final distribution.

Now we can compute most properties we might want about the process.
For instance, the expected number of distinct mutational origins is
\begin{equation}
    \E\left[ \mbox{ \# of independent mutations } \right] = 
            1 + \frac{\mu}{m} \sum_{k=2}^L \frac{1}{k+(\mu/m)}. \label{discrete_expected}
\end{equation}
The probability that there is only a single successful mutation is
\begin{equation}
\P \left\{ \mbox{only one mutational origin} \right\} = 
            \prod_{k=1}^{L-1} \frac{ k }{ k+(\mu/m)} .
\end{equation}
More generally, suppose we sample a single deme, and are interested in the total number of demes $S$
(including the sampled one) that share the same fixed mutation as our sampled deme.
Then $S$ has distribution
\begin{equation} \label{eqn:Sdistrn}
\P\{ S=s \} = \frac{ (s-1)! (\mu/m)^{L-s} }{ \prod_{k=1}^{L-1} (k+(\mu/m)) } .
\end{equation}
Furthermore, if $L$ is large, then $S/L$ has probability density approximately
\begin{equation} \label{eqn:betadistrn}
   (\mu/m) (1-x)^{(\mu/m)-1}
\end{equation}
namely, is a $\mathrm{Beta}((\mu/m), 1)$ distribution--- see \cite{donnelly1989continuity} and \cite{perman1992sizebiased}.

% In Figure \ref{fig:discbyratio} we plot mean number of types and size (number of demes occupied) of a sampled type,
% for a model with 20 interconnected demes, across different parameter values.  
% This indicates that we need the population-scaled mutation and migration rates to be within a factor of about 100 of each other for parallel adaptation to leave an interesting pattern. If migration is faster, then a single type is likely to take over, while if migration is weaker, each deme is likely to come up with its own type.

The high connectedness of the discrete deme island model means that the expected number of distinct alleles, (Equation \eqref{discrete_expected}), 
grows with the $\log$ of the number of demes. This strongly contrasts with the continuous spatial model where the local nature of dispersal means that doubling the species range will double the number of mutations expected. 
Furthermore, while in the continuous model the areas occupied by distinct mutations are similar in area, in this strong-selection discrete island model, 
only a few types tend to dominate even as the number of demes (and thus, distinct types) grows large (see Equations \eqref{eqn:Sdistrn} and \eqref{eqn:betadistrn}).

A more general model would allow migration only along edges of some graph connecting the demes.
In the low migration limit such a model still produces a partition distribution independent of the selection coefficient,
but it does not in general have the Ewens distribution.
Another extension would be to include the time alleles need to achieve an intermediate frequency,
along the lines of \citet{Navarro:03}, which would reintroduce dependence on the selection coefficient.

\texttt{XXX omit this? XXX}
We have treated the time during which the selected allele is at intermediate frequency in a deme (about $\log(\sigma^2 \rho)/s_p$ generations) as negligible, 
which ignores migration events that occur while the mutation is spreading through the population. 
we now sketch of a more realistic approximation.
For each occupied--unoccupied pair of demes, if the number of mutant alleles in the occupied deme is $n(t)>0$,
then at time $t$, migrants from the occupied deme pass the selected allele to the unoccupied deme at rate $2 n(t) m p_f(s_p)$,
while newly arisen mutants arise and fix in the unoccupied deme at constant rate $2 N \mu p_f(s_p)$.
The selected type should grow approximately as $n(t)=\exp(s(t-t_0))$.
Of course, to follow this model through, we will have to account for multiple types within a single deme and other complications,
but we can at least make a few observations.
One is that the final distribution of types is no longer independent of $s_p$, which enters through the dynamics $n(t)$.
However, it is clear that increasing the selection coefficient $s$ will decrease the number of types (independent mutational origins),
while on the other hand, increasing the deme size $N$ will increase the number of types,
and that the case presented above is at the extreme (see also \cite{Navarro:03} for deterministic approximations). 


%%%%%%% %%%%%%%%%%
\subsection[Haplotypes Shared Within a Patch]{Length of the shared haplotype within a patch}
\label{ss:patch_haplotype}

The other key issue in thinking about the population genetic signal of adaptation in spatial patches 
is the process by which haplotypes are whittled down within patches, 
which plays into a number of aspects of haplotype sharing. 
The initial haplotype that sweeps within a patch 
will be dispersed over time by recombination.
Likewise, the haplotype that is shared between patches coadapted by migration
(equation~\ref{eqn:haplotype_length}) will also break down. 
However, a long time after the initial sweep,
we may still expect to find individuals within the patch sharing longer haplotypes around the selected locus
than with individuals elsewhere,
since selection against migrants 
decreases mean coalescence times within the patch near the selected locus.  %\citep{bartonpaper,charlesworths}. 
We first develop a simple approximation to the rate 
at which recombination breaks down haplotypes within patches. 

Refer again to Figure~\ref{fig:lineagesmotion}. 
When a lineage is in a region with high frequency of $B$ alleles (dark red), 
it is very unlikely to recombine off the $B$ background. 
However, when it wanders out of the patch,
into regions with lower frequency of $B$ alleles,
when it recombines it will
usually be on to the background of the $b$ allele.

A detailed treatment of this is well beyond the scope of this paper,
but we can get a good idea of the genomic scale of these effects through heuristic arguments.
To understand the probability of a recombination moving our lineage
off of the B background we need to understand what fraction of the
time our lineage spends in geographic regions where $b$ is common. 
We will follow the lineage of a single $B$ allele
in a patch at migration-selection balance, 
i.e., with the local proportion of $B$ alleles near $x$ equal to $q(x)$,
and suppose that the lineage stays close enough to the patch 
that fluctuations about $q(x)$ are small. 
(Note that this assumption differs from
previous sections, but here our focus is on a typical lineage 
in the patch rather than the rare lineages that make it far from the patch).
If we suppose that type $B$ individuals at location $y$ reproduce at rate $1+s(y)$,
and the chance their offspring move to $x$ is $m(y,x)$ (with $\sum_x m(y,x)=1$),
then the total rate of influx of $B$ alleles from $y$ into $x$ is $\rho q(y) (1+s(y)) m(y,x)$.
As argued by \citet{hudson1988coalescent}, 
this implies that the chance that a $B$ allele at $x$
has been inherited from a parent at $y$ in a time interval $dt$
is equal to this influx divided by the local abundance of $B$ alleles,
i.e., $dt \times q(y) (1+s(y)) m(y,x) / q(x)$.
This therefore describes the movement of a lineage backwards in time.
After the same large-population, continuous-space limit that underlies the rest of this paper (section~\secref{apx:eqfreq}),
if the assumption that fluctuations in $q(x)$ are small,
this motion should converge to a diffusion process (as assumed by e.g., \citet{hallatschek2008surfing}),
notwithstanding some technical difficulties (e.g., discontinuity of $s(x)$).
This diffusion, when at $x$, 
is pulled toward regions with more $B$ alleles
by a mean displacement term $(1+s(x))\frac{d}{d\,x} \log( q(x) (1+s(x)) )$,
where $s(x)=s_p$ if $x$ is in the patch and $s(x)=s_m$ otherwise, 
and moves at speed $\sqrt{1+s(x)}$ (which is nearly constant, since $s(x)$ is small).
In It\=o differential notation, 
the location $X_t$ of a $B$ lineage $t$ generations ago satisfies
\begin{align}
    d X_t &= (1+s(X_t)) \frac{d}{dx} \log( q(X_t) (1+s(X_t)) ) dt + \sigma \sqrt{1+s(X_t)} dB_t,
\end{align}
where $B_t$ is a standard Brownian motion.
Furthermore, it is easy to verify that the stationary distribution of this process has density
\begin{align}
  \pi(x) = \frac{ q(x)^2 (1+s(x)) }{ \int_y q(y)^2 (1+s(y)) dy } . \label{eqn:lineagestatdist}
\end{align}
A lineage at a locus at genetic distance $R$ from a $B$ allele
moves as a $B$ lineage until it recombines onto a $b$ background,
which occurs when at $x$ with rate $r (1-q(x)) (1+s(x)/2)$.
The Feynman-Kac formula could then be used to write down differential equations
solved by various moments of the haplotype length as a function of the sampling location of the lineage;
instead, we only use the above as a guide for rough approximation.

Based on this picture of
the lineage bouncing around in the patch,
we propose a simple heuristic to describe the probability of recombination. 
If recombination is on a slower time scale than mixing within the patch,
then the time until the last recombination is the integral of $r (1-q(x)) (1+s(x)/2)$ 
against the stationary distribution (equation~\ref{eqn:lineagestatdist}).
Within the patch, $q(x) \approx 1$,
so recombination off the $B$ background is unlikely to occur,
and so the rate of recombination depends to a first approximation 
on the proportion of the patch in which $q(x)$ is intermediate,
i.e., where both $B$ and $b$ are present.
This region where $q(x)$ is intermediate has width of order $\sigma/\sqrt{s}$, where $s = \min(s_p,s_m)$.
As we increase the size of a patch, the area on which $q(x)$ is intermediate scales with the perimeter of the patch.
% Therefore, the proportion of the time a lineage at stationarity is found in the intermediate regions
% multiplied by $q(x)$ in these intermediate regions, where we assume $q(x) \approx 1/2$. 
The per generation rate of recombination of a locus linked to the $b$ background is thus
reduced by a factor roughly equal to the proportion of the patch lying within distance $\sigma/\sqrt{s}$ of the boundary,
so the effective recombination rate is
% \plr{Note to ourselves: If a $d$-dimensional object with area A has radius $r$ (in any sense)
% and you shrink it to have radius $r_0$,
% then the resulting object has area $A \times (r_0/r)^d$;
% here we have $r_0 = r - \sigma/\sqrt{s}$,
% in $d=1$ the radius is $A/2$ and in $d=2$ it is $\sqrt{A/\pi}$.}
\begin{align}
r_\text{eff} &=
    \begin{cases}
    r \frac{ \sigma /\sqrt{s} }{ \sigma/\sqrt{s} + A }    \qquad & d=1 \\
   r \frac{\pi  \sigma /\sqrt{s}  \left( \sigma /\sqrt{s}+
  \sqrt{A/\pi} \right)}{A+\pi \sigma/\sqrt{s} \left( \sigma/\sqrt{s} +
  \sqrt{A/\pi} \right) }    \qquad & d=2 .
\end{cases}
\label{eqn:rec_rate_within}   %(\sigma/\sqrt{s})A^{-1/d}
\end{align}
It follows that $T$ generations after the founding of a patch we expect a
haplotype of genetic length $1/(r_\text{eff}T)$ Morgans to remain linked to
the $B$ allele in a typical lineage. 
Note that the length of haplotype
decays at a rate roughly proportional to $\sqrt{s}/ \sigma$, as the narrower the cline 
the less opportunity for recombination to move our lineage off the $B$ background. 
 

 
Finally we can consider the expected length of haplotype shared within a patch. 
To do this, we compare the rate of recombination between $B$ and $b$ haplotypes (equation~\ref{eqn:rec_rate_within})
to the rate of coalescence of $B$ haplotypes within a patch. 
We will assume that mixing within the patch is fast compared to the rate of drift, 
so that the rate of coalescence of $B$ alleles within a patch is 
$\xi^2/(A\rho)$,
the ``variance effective population size'' of the patch
(this bypasses the well-known difficulties to spatial coalescence \citep{felsenstein1975pain,barton2002neutral}).
The rate of recombination and coalescence will be on the same order if
$\xi^2 /(A\rho) \approx r (\sigma/\sqrt{s}) A^{-1/d}$.
Hence the genetic length of the haplotype shared between pairs of
individuals within the same patches at equilibrium is on the order of
\begin{align} \label{eqn:within_haplotype_length} 
  r = \frac{\sqrt{s} \xi^2}{\sigma \rho A^{(d-1)/d}}  .
\end{align}
This does not depend on $A$ in one dimension,
but decreases with $\sqrt{A}$ in two dimensions.
In applications, this can be compared to equation~\eqref{eqn:haplotype_length}
for the length of shared haplotype between patches.


