\documentclass{article}
\usepackage{fullpage}
\usepackage{amsmath,amssymb}
\usepackage{natbib}
\usepackage{marginnote}
\usepackage{graphicx}
\usepackage{hyperref}
\usepackage{latexml} % for \iflatexml

\newcommand{\var}{\mathop{\mbox{Var}}}
\renewcommand{\P}{\mathbb{P}}
\newcommand{\E}{\mathbb{E}}
\newcommand{\R}{\mathbb{R}}

\bibliographystyle{plain}

\begin{document}

\section{Introduction}

Recall questions of origin of adaptations

Reference on parallel adapation, standing variation, etc.:
 coop,
 pennings and hermisson,

Summarize previous paper

Question assumptions of no standing variation and homogenous selection

Stick to constant-speed waves.

\section{General model}

outline here assumptions common to both models.

\section{Standing variation}

\subsection{The set-up}

% selection pressure changes at $t=0$ from $1-s_d$ to $1+s_b$ -- homogeneous

For now we ignore geographic hetergeneity, and assume that the species range
is a homogenous, one- or two-dimensional region. 
There are two selective classes -- the {\em neutral} type, and the {\em mutated} type.
We assume that different mutations are distinguishable --
either as selectively equivalent mutations, or by linked variation.
For a long time in the past, the mutated type has been at a selective disadvantage (long enough to be at selection-mutation equilibrium),
but at a certain time the selective regime changes, so that the mutated type has a selective advantage and quickly spreads to fixation.
After fixation, alleles are either descended
from families of mutants present as standing variation when the selective regime changed,
or from new mutants arising since that time.

For concreteness, suppose that before time $t=0$,
the mutant type has fitness $1-s_d$ relative to the neutral type
(i.e.\ it produces on average $1-s_d$ times the number of offspring per generation),
and that after time $t=0$,
the mutant type has fitness $1+s_b$,
where $s_b>0$ will usually be assumed to be small, and $0<s_d<1$.
We assume that diploid fitness is additive, or at least that the important dynamics are determined by the haploid fitness.
The number of offspring has finite variance, which to keep the formulas simple we assume to be equal to one.
(If the mutation is recessive, XXX more complicated things happen.)
As for the other parameters,
suppose that each offspring of a neutral parent is of the mutant type with probability $\mu$,
and that the mean squared distance between parent and child is $\sigma^2$ (the {\em dispersal variance}).
The species occupies an area with mean density
$\rho$ haploid individuals (or chromosomes) per unit area.
\marginnote{XXX or do diploid? below is haploid.}

% Branching-process-sized point mass approx to local groups present at $t=0$
% -- depends on local population density large enough relative to migration (to be "branching")
% -- and migration not too large (since we approximate as a point mass).
% Find mean density of clusters of successful standing mutants. 
% Is greater-than-one size of clusters likely to make a difference? 

We make use of the commonly used approximation that neglects competition between relatives,
treating the offspring of a new mutant that appears in an area not already occupied by the mutated type
as a branching process.
After $t=0$, the offspring of a single mutant form (approximately) a branching process with growth rate $s_b$,
so each new mutant establishes locally with probability $p_s \approx 2s_b$.
As in \cite{ralphcoop2010}, each new offspring has a very small probability of being a mutant and establishing locally,
so the collections of times and locations at which mutants appear and establish locally 
is well--approximated by a Poisson process in space and time.
This the rate of this Poisson process, 
the rate per unit area at which new mutants appear and establish locally after $t=0$ in areas not already occupied by the mutant type
is approximately $\lambda = 2 \mu \rho s_b$.

Before $t=0$, on the other hand, the mutation is deleterious, so the genetic descendents of each new mutation are (with high probability) doomed to extinction,
but may persist for some time.
Since the times and locations of mutants before $t=0$ also well-approximated by a Poisson process with rate $\mu \rho$,
the locations of all mutant families extant at $t=0$ whose descendents are destined to fix locally is also,
by the Mapping theorem, a Poisson process with rate we define to be $\lambda_0$.
If we assume that the descendants of at most only a few members of any extant mutant family at $t=0$ will survive,
and that these progenitors are near to each other in space, we can then treat each such family as a single mutant,
equivalent to the mutants arising later.
(The approximation will be good if the log of the size of an extant families is small relative to the establishment time,
and the spatial distribution of the family is small relative to the spread between them.)
To find $\lambda_0$, consider a mutation that arose at time $-T<0$, and let $Z_s$ be the number of its descendants at time $s-T$.
At time $t=0$, there are $Z_T$ individuals present with the mutation,
and each has probability $p_s \approx 2s_b$ of establishing, approximately independently.
Therefore, the probability that some descendants of this mutation establish and fix locally is $1-(1-p_s)^{Z_T}$,
so defining $\zeta(u,t) = \E[u^{Z_t} | Z_0=1 ]$ to be the generating function of $Z_T$,
the rate is
\begin{align*}
    \mu \rho \int_0^\infty \left( 1- \zeta(1-p_s,T) \right) dT .
\end{align*}
Now for $s_b$ small, using $p_s \approx 2s_b$ and that $\E[Z_T]=(1-s_d)^T$,
we know that $\zeta(1-2s_b,T) \approx 1-2s_b \E[Z_T] = 1-2s_b (1-s_d)^T$,
resulting in the approximation
\begin{align}
    \lambda_0 &= \mu \rho \int_0^\infty 2s_b (1-s_d)^{T} dT \\
        &= \frac{ 2 \mu \rho s_b }{ -\log(1-s_d) } .
\end{align}

XXX More discussion on whether this is a good approximation?  
Compute typical size and radius of a standing cluster?

recall PPP computations; draw cones of opportunity for new mutations 

\subsection{Characteristic parameters}

% a characteristic length of regions and a proportion of haplotypes (or of space?) that are from standing variation

In \citet{ralphcoop2010}, studying the model without standing variation,
we defined a {\em characteristic length} which gave the spatial scale across with mutants with distinct origins would establish.
This was proportional to the mean distance between neighboring established mutants,
but had the advantage of being easier to calculate.
Furthermore, the time scale over which adaptation occurred could be found by dividing the characteristic length 
by the speed at which the mutants spread.
We will define the characteristic length for this model,
as well as a similar compound parameter describing the relative importance of standing variation to the process of adaptation.
(Note that a similar approach, summarizing with a {\em characteristic length} was taken by \citep{slatkin1973geneflow} for a different problem.)

Suppose we fix our attention on a particular new mutation that happens to be the first to occur in some region.
As it spreads, it will cover a distance $L$ in time $L/v$, where $v = \sigma \sqrt{2s_b}$ is the speed of the wave of advance.
The number of other mutations appearing in the circle it has covered up until this time is Poisson with mean
$\lambda_0 \pi L^2 + \lambda \pi L^3 /v$ in two dimensions
(and $\lambda_0 2 L + \lambda \pi L^2 /v$ in one dimension).
% $\lambda_0 \omega_d L^d + \lambda \omega_d L^{d+1} /v$ in $d$ dimensions
Therefore, if we define $\chi$ to be the unique positive solution to
\[
    \lambda_0 \pi \chi^2 + \lambda \pi \chi^3 /v = 1,
\]
then $\chi$ gives the distance spread unobstructed by the descendants of a new mutant
before it is expected that one other successful mutation would have arisen in the area covered so far.
The explicit formula for $\chi$ is cumbersome; here we omit it.
(This is in two dimensions; in one dimension the characteristic length is $\chi = \frac{ \sqrt{ 4 \lambda_0^2 + 2\lambda v } - 2 \lambda_0 }{ 4 \lambda v }$.)
This can be rewritten 
\begin{equation} \label{eqn:defines_chi}
%    \left( \frac{ - \chi \log(1-s_d) }{ 2 s_b \sigma } \right)^2 + \left( \frac{ - \chi \log(1-s_d) }{ 2 s_b \sigma } \right)^3 
%        = \frac{ - \log(1-s_d)^3 }{ 4 s_b^2 \sigma^2 \pi \rho \mu } .
   \frac{\sqrt{2s_b} }{-\log(1-s_d) } \chi^2 + \frac{1}{\sigma} \chi^3 = \frac{1}{\sqrt{2s_b} \rho\mu\pi}
\end{equation}
From this we see that $\chi$ decreases with $\rho$ and $\mu$.
Furthermore, for large $\sigma$, the characteristic length is close to the value obtained just from standing variation:
$\chi = \sqrt( -\log(s_d) / 2 s_b \rho \mu \pi ) + O(1/\sigma)$.
On the other hand, if the mutant allele is highly deleterious before $t=0$,
then the characteristic length is close to the value from \citet{ralphcoop2010}:
$\chi = ( \sigma / \rho \mu \pi \sqrt{2 s_b} )^{1/3} + O(1/\log(1-s_d))$.


By the above calculation, we know that the relevant mutations occur about distance $\chi$ apart, 
and occur within the first $\chi/v$ generations.
Said another way, if we look in a circular region of space of radius $\chi$ over $\chi/v$ generations,
we expect to find one mutational origin;
we denote the probability that this mutation is from standing variation by $\pi_0$.
Therefore, $\pi_0$ is roughly the proportion of establishing mutations that come from standing variation,
and it is an elementary consequence of the Poisson process that
\begin{align} \label{eqn:pizero}
    \pi_0 = \lambda_0 \pi \chi^2 .
\end{align}
\marginnote{put in value of $\chi$ computed above.}
(In one dimension, $\pi_0 = 2 \lambda_0 \chi$.)

\subsection{Results} 

The quantity that is perhaps easiest to compute is the mean time until adaptation.
Fix some geographic location, and let $\tau\ge0$ be the time at which the point is reached by the advantageous mutations.
Then, as shown in Figure XX,
$\tau > t$ if and only if the cone with point at $(x,t)$ and slope $v$ extending back to $t=0$ is empty of successful mutations.
Since we assume these are a Poisson process, 
\[
    \P\{ \tau > t \} = \exp\left( - \lambda_0 \pi v^2 t^2 - \lambda \pi v^2 t^3 / 3 \right) ,
\]
and so we have
\begin{align}
    \E[\tau] % &= \int_0^\infty \P\{ \tau > t \} dt \\
        &= \int_0^\infty \exp\left( - \lambda_0 \pi v^2 t^2 - \lambda \pi v^2 t^3 / 3 \right) dt.
\end{align}
For means of evaluting this integral, see Appendix \ref{apx:integrals}.

% proportion of haplotypes (and of space) that are from standing variation 

We have defined $\lambda_0$ to be the mean density of standing variants that reach local fixation.
Define $\nu_+$ to be the mean density of new mutants whose offspring fix locally.
Since the probability that a mutant arising at $(x,t)$ is lucky enough to be born in a location not already occupied by mutants
is $\P\{ \tau > t \}$,
we can see  $\nu_+ = \int_0^\infty \lambda \P\{\tau>t\} dt$, and hence
$\nu_+ = \lambda \E[\tau] $.
Using that $\lambda_0 = \lambda / \log(1/(1-s_d))$, this gives us that
\begin{equation}
    \nu_+ = (\mbox{mean density of new patches}) = \frac{1}{1-\log(1-s_d) \E[\tau]} .
\end{equation}
Since the mean proportion of patches that come from standing variation is $\lambda_0 / (\lambda_0 + \nu_+)$,
this gives us the exact quantity we approximated by $\pi_0$ in \eqref{eqn:pizero}.
There are $\lambda_0 + \nu_+$ patches per unit area, so
the typical patch (i.e.\ with distribution given by the Palm measure) occupies area $1/(\lambda_0 + \nu_+)$.

We can also find the mean proportion of space covered by standing variants.
If by time $t$ a geographic location has not yet been reached by the mutation,
the probability it will be reached by time $t+dt$ 
by a standing variant is approximately $2 \lambda_0 \pi v^2 t dt$, 
and the probability it is reached by a new variant is $\lambda \pi v^2 t^2 dt$.
Therefore, as is standard for competing exponentials,
the probability a given location is reached first by a standing variant,
and therefore the mean proportion of space covered  by standing variants,
is
\begin{equation}
    z_0 = \int_0^\infty {2 \lambda_0 \pi v^2 t} \; \exp \left( - \lambda_0 \pi v^2 t^2 - \lambda \pi v^2 t^3 / 3 \right) dt .
\end{equation}
\marginnote{put in this value in terms of Gaussian CDF?}
To evaluate this integral, again see Appendix \ref{apx:integrals}.

Furthermore, if we define $a_0$ to be the mean area occupied by a typical standing variant, 
then $a_0$ is given by the proportion of the range occupied by standing variants over the mean density of unique standing variants,
i.e.\ $a_0 = z_0 / \lambda_0$.
We can compare this to the corresponding quantity $a_+$ for a new variant through the formula
$a_0 / a_+ = z_0 / (1-z_0)$.

\begin{figure}[ht]
\begin{center}
\includegraphics[width=1.0\textwidth]{proportion-by-sigma.eps}
\caption{ %
{\bf Mean proportion of patches arising from standing variation.} The paramters given are for the sickle-cell example.  XXX say more.
}
\label{fig:cartoon}
\end{center}

\end{figure}

\marginnote{do mean (and variance?) of distance between haplotypes?}

influence of migration on things -- plot these versus migration 


\section{Local establishment}

Include stuff from ``comparison to panmixia'' here.


\section{Patchy selection} 

\subsection{Model} 

In this section, we suppose that instead of a uniform habitat,
that the selective pressure is {\em patchy}, so that the new alleles are only advantageous in isolated locations, 
and are disadvantageous in the intervening region.
As before, we assume that the new allele has fitness $1+s_b$ relative to the unmutated type in the ``good'' patches,
and assume that it has fitness $1+s_m$ in the intervening areas, with $s_m<0$.
To make things tractable, we assume that the distances between the patches are large relative to migration distance $\sigma$,
and that the disadvantage $s_m$ ensures that on the relevant timescale,
the allele is very unlikely to fix in the regions where it is at a disadvantage.
With these assumptions as justification, we will move to a model of discrete demes exchanging migrants,
where ``migration'' occurs through the rare, transient passage of the new allele through regions where it is disadvantageous.

equal deme sizes?  general migration rates?

Describe simple discrete model, for general migration rates.

\subsection{Mutational influx}

Consider first a single, isolated patch of area $A$ in which no new allele has appeared.
\citet{barton1987} (or \citet{slatkin1981}) has studied this situation (in one dimension)
and found that:
mutations appearing further than a few multiples of $\sigma/\sqrt{2s_m}$ away from the patch have vanishingly small probability of establishment;
mutations appearing within the patch have probability strictly less but approaching that $2s_b$ of establishment;
and if the patch width is less than $2 \tan^{-1} (\sqrt{s_m})$, then local fixation does not occur (rather, is not stable).
Let $p(x)$ be the probability that a new allele arising at location $x$ fixes.
The function $p(x)$ can be found in terms of Jacobi elliptic functions,
XXX as shown in the Appendix,
but the expressions are particularly unhelpful.
The total successful mutational influx per generation is then given by $\int \rho \mu p(x) dx$,
and depends in a complicated way on the relationship of patch width and selection coefficients,
but still scales linearly with the population mutational influx $\rho \mu$.
Therefore, we define $p_f = \int p(x) dx$, and in a pinch will approximate this by $2 s_b A$,
which will be a good approximation if the width $\sqrt{A}$ is large relative to $\sigma/\sqrt{2s_m}$.

XXX We are interested in the case where a single mutation only takes over? XXX

\subsection{Migration rates}

Now suppose that there are two patches of area $A$ of good environment separated by distance $R$,
and that the allele has begun to fix in the first, but has not yet appeared in the second.
As above, any copy of the new allele that appears at location $x$ sufficiently near to the second patch 
has probability approximately $p(x)$ of fixing locally in this patch,
whether the allele is a new mutation or is descended from the occupants of the first patch.
There will be a small number of alleles descended from those of the first patch present in the regions where they are disadvantageous,
held at mutation--selection balance;
the equilibrium frequency may again be expressed in terms of Jacobi elliptic functions (Appendix XXX),
although the equations are again not particularly helpful.
However, we do know that if $q(r)$ is the expected proportion present at mutation--selection balance at distance $r$ from the first patch
in the absence of the second patch, that for large $r$ the proportion decays exponentially, $q(r) \approx C \exp( -|s_m r / \sigma|)$,
where $C$ is a constant depending on the structure of the populations and the selection coefficients.

Combining this with the calculations above allows us to get a rough sense of the scale of migration between separated patches,
supposing that patches are large enough for local fixation to be stable.
If the alleles were not advantageous in the second patch,
the average numbers of mutant alleles present near the second patch would be around $\rho A \exp( - |s_m R/\sigma| )$;
so the waiting time until some migrant allele fixes in the second patch would be something like $T_m = ( 2 s_b \rho A \exp( - |s_m R/\sigma|)^{-1}$.
If $T_m$ is large, then we would expect successful migrants to be rare,
so that if the second patch adapts through migration, it is likely to come from only a single migrant.
In the converse situation, XXX more than one migrant? XXX


Get migration and mutation rates from a homogeneous continuous model:
The migration rate from patch $i$ to a patch at distance $r$ is $C_i \exp(-\gamma r s /\sigma)$,
where $C_i$ depends on the size of the patch and (to a lesser degree?) the local structural details of the selective difference.
The mutation rate is more or less (size of patch) times $\mu$; 
more accurately we can integrate using Barton assuming step selection.

\subsection{Results} 

Pull out results from other paper on discrete model:
if migration rates are equal then we get Chinese Restaurant/Ewens;
in any case is independent of selection.

when are patches effectively isolated?   
This occurs when the effective migration rate is much smaller than the mutation rate, namely...

\section{Applications} 

Look at some human parameters? 

\section{Discussion} 

if populations and mutation rates are large enough, parallel adaptation is likely, aided by geography. 
recall others' work. 

patchy selection helps this happen; 
effective migration rate looks like what; 
strength of (positive?) selection suprisingly does not affect results (?) 

standing variation will be important under these circumstances; 
migration rate (and so geographic structure) will not affect numbers of types if this is true; 
however migration always affcts spatial patterns. 

discussion of applications. 

\bibliography{standing_patches_refs}

\appendix

\section{Integrals appearing in the text}
    \label{apx:integrals}

In the text at several points appear integrals of the form
\begin{align}
  \int_0^\infty t^c \exp \left( - \alpha t^d - \beta t^{d+1} \right) dt 
\end{align}
where $c$ is a positive integer, $d$ is the dimension, and $\alpha$ and $\beta$ are positive real numbers.
This could be evaluated through standard numerical methods; below we describe a power series expansion.
Changing variables to $u = \beta t^{d+1}$, this becomes
\begin{align}
    \left( (d+1)^{-1} \beta^{ (1-c)/(d+1) } \right) \int_0^\infty u^{(c+1-d)/(d+1)} \exp\left( - \alpha \beta^{-d/(d+1)} u^{d/(d+1)} - u \right) du ,
\end{align}
so it suffices to evaluate the function
\begin{equation}
    G(a,b,x) := \int_0^\infty  t^a \exp\left( -x t^b - t \right) dt ,
\end{equation}
in the case that $a=(c-d+1)/(d+1)$, $b=d/(d+1)$, and $x$ is a function of the demographic paramters.
Since $a$ and $b$ only depend on the dimension and the quantity being computed,
we are interested in $G$ as a function of $x$.
At least in the case $b=d/(d+1)$ it is possible to express $G$ as a finite sum of gamma functions,
but we proceed with a simpler method.
Note that $\partial_x G(a,b,x) = -G(a+b,b,x)$,
and that at $x=0$, the function $G$ is the gamma function $G(a,b,0) = \Gamma(a+1)$.
Therefore, a Taylor series for $G$ would be
\[
    G(a,b,x) = \sum_{n \ge 0} = \sum_{n \ge 0} \frac{(-x)^n}{n!} \Gamma(a+nb+1) .
\]
It is easy to check using Stirling's formula that $\limsup_{n \to \infty} ( x^n \Gamma(a+nb+1)/n! )^{1/n} = 0$
if $b<1$, so the sum converges.

\end{document} 
